---
title: "102-RAD-seq"
output: html_document
date: "2023-07-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggrepel)
library(viridis)
```
PRJDB3890    
Wang et al. (2016) data     
This data set includes the raw ddRAD-seq sequencing reads of Asian seabass, which were collected from six geographical populations consisting of 191 individuals from Southeast Asia to northern Australia and Papua New Guinea.     


```{r}
rad<-read_csv("meta/lates-rad-sra-info.txt")
sites<-rad %>% group_by(geo_loc_name_country, Lat_Lon) %>% summarize(Count=n())
sites
```
13.38 S 127.04 E West Australia
18.41 S 146.64 E East Australia   

can download like https://sra-pub-run-odp.s3.amazonaws.com/sra/DRR111125/DRR111125

putting in data/rad-raw

```{sh, eval=FALSE}
cat ../../meta/lates-rad-sra-info.txt | cut -d ',' -f 1 | while read line; do wget https://sra-pub-run-odp.s3.amazonaws.com/sra/$line/$line	; done;
```

Splitting, without readids for BWA

fastq-dump --outdir rad-split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip rad-raw/DRR111126

 `for f in rad-raw/DRR*; do echo $f; fastq-dump --outdir rad-split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f; done;  `   
 
This of course would take forever!

`for f in ../rad-raw/DRR*; do echo fastq-dump --outdir ./ --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f >> split-commands.sh; done;`

`module load parallel`
`srun -p high -t 1-00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 parallel -j 12 < split-commands.sh`   

Running 12 parallel tasks on one node, note the high node actually uses a Parallel (CPU) node with 512 GB RAM, 128 cores/256 threads, 2 TB /scratch

Getting the missing files
fastq-dump --outdir rad-split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip rad-raw/DRR111273.sralite.1
~/lates-wgs/data/rad-split$ ln -s DRR111273.sralite.1_pass_1.fastq.gz DRR111273_pass_1.fastq.gz
~/lates-wgs/data/rad-split$ ln -s DRR111273.sralite.1_pass_2.fastq.gz DRR111273_pass_2.fastq.gz

## Align    

`(base) maccamp@farm:~/lates-wgs/data/rad-split$ ls | grep pass_1 | perl -pe 's/.fastq.gz//g' > forward`     
`(base) maccamp@farm:~/lates-wgs/data/rad-split$ ls | grep pass_2 | perl -pe 's/.fastq.gz//g' > reverse`     
`(base) maccamp@farm:~/lates-wgs/data/rad-split$ ls | grep pass_1 | perl -pe 's/_pass_1.fastq.gz//g' > name`      

Should be able to join with rad df from the name

`(base) maccamp@farm:~/lates-wgs/data/rad-split$ paste forward reverse name  > to-align.txt`     
`bash ../../doAlign-rad-zipped.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz `
`bash ../../doAlign-rad-zipped.sh straggler.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz `

Visualize read counts.     

```{r}
files<-list.files(path="outputs/102", patter="*.stats", full.names = TRUE)
reads<-lapply(files, read.csv, header=FALSE, col.names=c("Run","Aligned","Filtered","Coverage"))
reads<-bind_rows(reads)
``` 

```{r}
meta<-left_join(rad,reads) %>% mutate(Location=paste0(geo_loc_name_country, "-",Lat_Lon)) %>%
  mutate(Pop=ifelse(Location=="Australia-13.38 S 127.04 E", "AUW",
             ifelse(Location=="Australia-18.41 S 146.64 E", "AUE",
         ifelse(Location=="Papua New Guinea-9.42 S 140.93 E", "PNG",
         ifelse(Location=="Indonesia-1.66 S 104.87 E", "INA",
         ifelse(Location=="Malaysia-3.35 N 103.64 E", "MAL",
         ifelse(Location=="Thailand-12.62 N 101.15 E","THA", geo_loc_name_country
         )))))))
meta <- meta %>% mutate(Path=paste0("bams/",Run,".sort.flt.bam"))
```

```{r}
min(meta$Filtered)

```

```{r}
ggplot(meta) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  scale_fill_viridis_d(option="turbo") +
  theme_bw() +
  ylab("Count") +
  xlab("Filtered Read Number") +
  theme(panel.grid=element_blank())
```


```{r}
ggplot(meta) +
  geom_histogram(aes(x=Coverage, fill=Pop)) +
  scale_fill_viridis_d(option="turbo") +
  theme_bw() +
  ylab("Count") +
  theme(panel.grid=element_blank())
```

## QC

Look at Chrom1 only and Oz+PNG

samples from paper    
```{r}
samples<-read_csv("outputs/105/samples", col_names = c("Isolate"))
```

```{r}
oz<-meta %>% filter(Pop %in% c("AUE","AUW","PNG"))  %>% filter(Isolate %in% samples$Isolate)

write_csv(oz %>% select(Isolate, Pop) %>% arrange(Pop, Isolate), file="meta/oz-meta.csv")
min(oz$Filtered)

oz %>% group_by(Pop) %>% summarize(Count=n())
write_tsv(oz %>% select(Path), file="bamlists/oz.bamlist", col_names = FALSE)


ggplot(oz) +
  geom_histogram(aes(x=Filtered, fill=Pop)) +
  scale_fill_viridis_d(option="turbo") +
  theme_bw() +
  ylab("Count") +
  xlab("Filtered Read Number") +
  theme(panel.grid=element_blank()) +
  xlim(0,max(oz$Filtered))
```
54 samples @ 75 percent missing
-r NC_066833.1 \
-rf genome/lates-lgs.txt

Wang et al reports

AUW 23
PNG 18
AUE 14

sample_183 is not published as a biosample and missing from AUW.   

-rf genome/lates-lgs.txt
```{sh, eval=FALSE}
srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/oz.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 41 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/102/oz-ibs > outputs/102/oz-ibs.out 2> outputs/102/oz-ibs.err &
# 10190 sites


srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/oz.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 49 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/102/oz-ibs-90 > outputs/102/oz-ibs.out 2> outputs/102/oz-ibs.err &
# 6770 sites

srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/oz.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 49 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.10 \
-out outputs/102/oz-ibs-90-maf1 > outputs/102/oz-ibs.out 2> outputs/102/oz-ibs.err &
#3809 sites 

srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/oz.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 49 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.20 \
-out outputs/102/oz-ibs-90-maf2 > outputs/102/oz-ibs.out 2> outputs/102/oz-ibs.err &
#2191 sites


#ngs admix    
srun -p high -t 01:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/102/oz-ibs.beagle.gz -K 2 -o outputs/102/oz-K2 &
srun -p high -t 01:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/102/oz-ibs.beagle.gz -K 3 -o outputs/102/oz-K3 &

## GLS
srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/oz.bamlist -rf genome/lates-lgs.txt \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 41 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-minMaf 0.05 \
-out outputs/102/oz-gl > outputs/102/oz-gl.out 2> outputs/102/oz-gl.err &

```


	-> Number of sites retained after filtering: 10237 on chrom1
	
	Genome-Wide 286267     
	

```{r}
pcm<-oz
m <- as.matrix(read.table("outputs/102/oz-ibs.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(pcm) 

text12<-covs %>% select(Run, Pop, V1, V2) %>%
  group_by(Pop) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Pop, shape=Pop), alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Pop), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d() +
  scale_shape_manual(values=c(21,22,23))
```



```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Pop) %>% mutate(Index=1:n())
df$Pop<-factor(df$Pop, levels=unique(df$Pop))

rdf<-df %>% dplyr::select(Pop, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Pop) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Pop,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1, alpha=0.8) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  theme(legend.position = "") +
  scale_x_continuous(breaks=labels$Position, labels=labels$Pop) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="magma")

}
```

```{r}
PlotAdmix("outputs/102/oz-K2.qopt", oz)
```

```{r}
PlotAdmix("outputs/102/oz-K3.qopt", oz)
```

Looks like Wang et al. Dropped some samples at some point. 

Call SNPs

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 49 \
-bam bamlists/oz.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/102/snps-oz  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/102/snps-oz.out 2> outputs/102/snps-oz.err &
```

50,391 variants 

As .plink file. Convert to VCF and then get differences in freqs with vcftools?    
AUE	14			
AUW	29	 		
PNG	18	

Are any SNPs different in PNG but not Oz?

```{r}
sub<-oz %>% filter(Isolate %in% samples$Isolate)
write_tsv(oz %>% select(Run), col_names = FALSE, file="bamlists/oz-sample-names.txt")
write_tsv(sub%>% filter(Pop %in% c("AUE","AUW")) %>% select(Run), file="outputs/102/oz.txt", col_names = FALSE)
write_tsv(sub %>% filter(Pop %in% c("PNG")) %>% select(Run), file="outputs/102/png.txt", col_names = FALSE)
write_tsv(sub%>% filter(Pop %in% c("AUW")) %>% select(Run), file="outputs/102/auw.txt", col_names = FALSE)
write_tsv(sub %>% filter(Pop %in% c("AUE","PNG")) %>% select(Run), file="outputs/102/aue+png.txt", col_names = FALSE)
write_tsv(sub %>% filter(Pop %in% c("AUE")) %>% select(Run), file="outputs/102/aue.txt", col_names = FALSE)

```

```{sh, eval=FALSE}
plink --tped outputs/102/snps-oz.tped --tfam outputs/102/snps-oz.tfam --out outputs/102/snps-oz.binary --recode --allow-extra-chr --noweb
plink --ped outputs/102/snps-oz.binary.ped --map outputs/102/snps-oz.binary.map --recode vcf --allow-extra-chr -out outputs/102/snps-oz-samples-recode

bcftools reheader --samples bamlists/oz-sample-names.txt -o outputs/102/oz.renamed.vcf outputs/102/snps-oz-samples-recode.vcf

#impute?

bcftools view -S outputs/102/oz.txt outputs/102/oz.renamed.vcf > outputs/102/oz-fish.vcf
bcftools view -S outputs/102/png.txt outputs/102/oz.renamed.vcf > outputs/102/png.vcf

vcftools --vcf outputs/102/oz-fish.vcf  --freq --out outputs/102/oz-fish
vcftools --vcf outputs/102/png.vcf --freq --out outputs/102/png-fish


```


```{r}
freq1<-read_tsv("outputs/102/oz-fish.frq", skip = 1, col_names = FALSE) %>% 
  rename(Chromosome=X1, Position=X2, NAllelesOzr=X3, NOz=X4, Major=X5, Minor=X6) %>%
  separate(Major, into=c("Major","MajorFrequencyOz"), sep=":") %>% separate(Minor, into=c("Minor","MinorFrequencyOz"), sep=":")
  
freq1$MajorFrequencyOz<-as.numeric(freq1$MajorFrequencyOz)
freq1$MinorFrequencyOz<-as.numeric(freq1$MinorFrequencyOz)

freq2<-read_tsv("outputs/102/png-fish.frq", skip = 1, col_names = FALSE) %>% 
  rename(Chromosome=X1, Position=X2, NAlleles=X3, N=X4, Major=X5, Minor=X6)%>%
  separate(Major, into=c("Major","MajorFrequency"), sep=":") %>% 
  separate(Minor, into=c("Minor","MinorFrequency"), sep=":")
freq2$MajorFrequency<-as.numeric(freq2$MajorFrequency)
freq2$MinorFrequency<-as.numeric(freq2$MinorFrequency)

comb<-left_join(freq2, freq1)

merge<-comb %>% mutate(AbsoluteDifference=abs(MajorFrequencyOz-MajorFrequency)) %>% 
  select(Chromosome, Position, Major, Minor, MajorFrequency, MinorFrequency, MajorFrequencyOz, MinorFrequencyOz, AbsoluteDifference, N, NOz) %>% 
  filter(N > 10, NOz > 10) %>% arrange(-AbsoluteDifference)

write_csv(merge,"outputs/102/oz-png-differences.csv")

merge
```

Local PCA in outputs/102/vcf   

```{sh, eval=FALSE}
cat lates-lgs.txt  | while read line; do bcftools view -Ob -r $line oz.renamed.vcf.gz > $line.bcf; done;
for f in *.bcf; do bcftools index $f; done;
```

```{r}
samples<-oz %>% select(Run) %>% rename(ID=Run)

population<-oz %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/102/vcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/102/vcf -t snp -s 20 -m 4 -I /Users/mac/github/lates-wgs/outputs/102/vcf/sample_info.tsv -j 1501

```{r}
mds<-read_csv("outputs/102/mds_coords.csv")
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```

```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2), alpha=0.75) +
  theme_bw()
```
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3), alpha=0.75) +
  theme_bw()
```

## Get samples from Wang et al.

```{r}
somesamples<-read_tsv("outputs/105/all-samples.txt", col_names = c("Isolate"))

wang<-meta %>% filter((Isolate %in% somesamples$Isolate))
wang %>% group_by(Pop) %>% summarize(Count=n())
```
There is a mismatch between what is reported and what there is. One AUW is dropped and not reported in the genepop file (sample_183)  

The wang df is 130 samples, expecting 131.   


```{r}
write_csv(wang, file="meta/wang.csv")

write_tsv(wang %>% select(Path), col_names = FALSE, file="bamlists/130.bamlist")

wang %>% group_by(Pop) %>% summarize(Count=n())
```

How does a single chrom look?

```{sh, eval=FALSE}
srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/130.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 98 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/102/wang-ibs > outputs/102/wang-ibs.out 2> outputs/102/wang-ibs.err &
```

11771 snps on chrom01    

```{r}
pcm<-wang
m <- as.matrix(read.table("outputs/102/wang-ibs.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(pcm) 

text12<-covs %>% select(Run, Pop, V1, V2) %>%
  group_by(Pop) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

covs$Population<-factor(covs$Pop, levels=c("AUW","AUE","PNG","INA","MAL","THA"))
ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Population), pch=21, alpha=0.75, cex=2) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Pop), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") 


ggsave("outputs/102/12pcs.jpeg")
```

Pretty much what we expected.    

Call snps for local PCA    

```{sh, eval=FALSE}
srun -t 36:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 117 \
-bam bamlists/130.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/102/snps-wang  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/102/snps-wang.out 2> outputs/102/snps-wang.err &
```

      
### What about the main Asian Seabass cluster?

```{r}
write_tsv(wang %>% filter(Pop %in% c("INA","MAL","THA")) %>% select(Path), col_names = FALSE, file="bamlists/76.bamlist")

```

```{sh, eval=FALSE}
srun -p high -t 3:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 16  \
-bam bamlists/76.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 57 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/102/76-ibs > outputs/102/76-ibs.out 2> outputs/102/76-ibs.err &
```

```{r}
pcm<-wang %>% filter(Pop %in% c("INA","MAL","THA"))
m <- as.matrix(read.table("outputs/102/76-ibs.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(pcm) 

text12<-covs %>% select(Run, Pop, V1, V2) %>%
  group_by(Pop) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Pop), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Pop), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") 
```

	-> Number of sites retained after filtering: 13000      

Yep.    


