---
title: "108-wgs-snps"
output: html_document
date: "2023-08-24"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

Following RADseq calls

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 53 -bam bamlists/58.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/108/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/108/snps-wgs.out 2> outputs/108/snps-wgs.err &
```


Now we need to split and calculate frequencies     

```{r}
library(tidyverse)
library(ggpubr)
```

Have 130 samples in outputs/106/renamed.vcf.gz 

Need SEA list and AUS+PNG lists to calculate freqs    


```{r}
m2<-read_csv("meta/58.csv")
m2$Region<-factor(m2$Region, levels=c("Northern Territory","Queensland","Papua New Guinea",
                                              "Indonesia","Philippines","Vietnam","Cambodia","Thailand",
                                              "India Eastern Coast","India Western Coast"))
m2 <-m2 %>% mutate(Lineage=ifelse(Region %in% c("Northern Territory","Queensland","Papua New Guinea"),"AUS+PNG",
                                                ifelse(Region %in% c("India Eastern Coast","India Western Coast"),"IND","SEA")))
m2 %>% group_by(Region) %>% summarize(Count=n(), Cov=mean(Coverage))
```


```{r}
sea<-meta %>% filter(Lineage %in% c("AUS+PNG","SEA"))
ind<-meta %>% filter(Pop %in% c("IND"))
```

```{r}
sea %>% select(Run) %>% write_tsv("outputs/108/sea.txt", col_names = FALSE)
ind %>% select(Run) %>% write_tsv("outputs/108/ind.txt", col_names = FALSE)
```


```{sh, eval=FALSE}
bcftools view -S outputs/108/sea.txt outputs/106/renamed.vcf.gz > outputs/108/sea.vcf
bcftools view -S outputs/108/ind.txt outputs/106/renamed.vcf.gz > outputs/108/ind.vcf

vcftools --vcf outputs/108/sea.vcf  --freq --out outputs/108/sea.vcf
vcftools --vcf outputs/108/ind.vcf  --freq --out outputs/108/ind.vcf
```


```{r}
freq1<-read_tsv("outputs/108/sea.vcf.frq", skip = 1, col_names = FALSE) %>% 
  rename(Chromosome=X1, Position=X2, NAllelesSEA=X3, NSEA=X4, Major=X5, Minor=X6) %>%
  separate(Major, into=c("Major","MajorFrequencySEA"), sep=":") %>% 
  separate(Minor, into=c("Minor","MinorFrequencySEA"), sep=":")
freq1$MajorFrequencySEA<-as.numeric(freq1$MajorFrequencySEA)
freq1$MinorFrequencySEA<-as.numeric(freq1$MinorFrequencySEA)

freq2<-read_tsv("outputs/108/ind.vcf.frq", skip = 1, col_names = FALSE) %>% 
  rename(Chromosome=X1, Position=X2, NAllelesIND=X3, NIND=X4, Major=X5, Minor=X6)%>%
  separate(Major, into=c("Major","MajorFrequencyIND"), sep=":") %>%
  separate(Minor, into=c("Minor","MinorFrequencyIND"), sep=":")
freq2$MajorFrequencyIND<-as.numeric(freq2$MajorFrequencyIND)
freq2$MinorFrequencyIND<-as.numeric(freq2$MinorFrequencyIND)

comb<-left_join(freq2, freq1)
```


```{r}
merge<-comb %>% mutate(AbsoluteDifference=abs(MajorFrequencyIND-MajorFrequencySEA)) %>% 
  select(Chromosome, Position, Major, Minor, MajorFrequencyIND, MinorFrequencyIND, MajorFrequencySEA, MinorFrequencySEA, AbsoluteDifference, NIND, NSEA) %>% arrange(-AbsoluteDifference)

# Sample sizes
#  filter(N > 30, NWinter > 20) 
write_csv(merge,"outputs/106/sea-aus-png-ind-freqs.csv")

diffs<-merge %>% filter(AbsoluteDifference>0.95)

write_csv(diffs,"Supplemental/Supplemental Table SX IND diagnostic loci.csv")

diffs %>% filter(AbsoluteDifference==1.00)
```

```{r}
nrow(diffs)
```


```{r}
dhisto<-ggplot(diffs) +
  geom_histogram(aes(x=Chromosome), stat="count") +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, vjust=1, hjust=1)) +
  ylab("Count\n") +
  xlab("\nChromosome") +
  theme(axis.title=element_text(size=14, face="bold")) +
  theme(axis.text.y=element_text(size=12)) +
  ggtitle("A") +
  theme(plot.title=element_text(size=20, face="bold"))

dhisto
```

```{r}
ahisto<-ggplot(diffs) +
geom_histogram(aes(x=AbsoluteDifference), bins=30) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(axis.text.x = element_text(angle = 45, vjust=1, hjust=1)) +
  ylab("Count\n") +
  xlab("\nAbsolute Difference") +
  theme(axis.title=element_text(size=14, face="bold")) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("B") +
  theme(plot.title=element_text(size=20, face="bold"))

ahisto
```


```{r}
ggarrange(dhisto,ahisto, ncol=1)
ggsave("outputs/108/histos.pdf", width=6, height=8)
```

