---
title: "1203-mtdna"
output: html_document
date: "2024-12-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

For some genomic analysis, we can use fragmentary de novo assemblies. We can also use the mtDNA to generate control region data because of _1_ unique andaman mtDNA, _2_ Wallacean fish, and _3_  better characterization of diversity in Australia.

let's start with the Wallacean fish and see if we can get a complete mitogenome.Options, use alignment to reference mtdna and pull out aligned/single aligned data and assemble. Otherwise, use all the data and see what shakes out.    

(base) maccamp@farm:~$ module load spades
Loading spades/3.15.5
spades.py --test

Seems to work.    
```{r}
sra<-read_csv("meta/lates-wgs-SraRunInfo.csv")
indo<-sra %>% select(Run, Experiment,SampleName,Region) %>% filter(Region=="Indonesia")
sra %>% select(Run, Experiment,SampleName,Region) %>% group_by(Region) %>% summarize(Count=n())
indo
```
ind data/raw
```{sh, eval=FALSE}
cat ../../meta/lates-wgs-SraRunInfo.csv | grep Indonesia | cut -d ',' -f 10 | while read line; do wget $line; done;
```

Getting SRR3165613.sralite.1

module load ncbi-toolkit/26_0_1 (added to bash_profile)     
module load sratoolkit/3.0.0       

 `for f in data/raw/*sralite.1; do echo $f; fastq-dump --outdir data/denovo --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f; done;  `    
```{r}
indo %>% mutate(Command=paste0("fastq-dump --outdir data/denovo --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip data/raw/", Run, ".sralite.1")) %>% select(Command) %>% write_tsv(file="1203.1-split.sh", col_names = FALSE)

```

module load parallel
module load ncbi-toolkit/26_0_1
module load sratoolkit/3.0.0       

```{sh, eval=FALSE}
srun -p bigmemm --time 6:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 1203.1-split.sh
```

files look like SRR3165613.sralite.1_pass_1.fastq.gz, SRR3165613.sralite.1_pass_2.fastq.gz

# Spades
module load spades

example command: bin/spades.py -1 left.fastq.gz -2 right.fastq.gz -o output_folder
spades examines a few kmers by default (k21, k33, k55 and k77) this should work for the vij et al. data, it is 100bp data

bangladesh is 150 bp, but shrug

```{sh, eval=FALSE}
#trialing bangladesh sample
srun -p bigmemm -t 24:00:00 --mem=240GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/bangladesh/split/SRR26358744_pass_1.fastq.gz  -2 data/bangladesh/split/SRR26358744_pass_2.fastq.gz -o data/denovo/SRR26358744 

#trialing indonesian sample
srun -p bigmemm -t 72:00:00 --mem=240GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/denovo/SRR3165613.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165613.sralite.1_pass_2.fastq.gz -o data/denovo/SRR3165613
```

Ran to completion

Did I get a complete mtDNA sequence?  

~/lates-wgs/genome/NC_007439.1.fasta

(base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165613$ blastn -query contigs.fasta  -db ../../../genome/NC_007439.1.fasta -outfmt 6
NODE_326_length_17913_cov_28.471721	NC_007439.1	98.649	15837	181	26	1363	17172	705	16535	0.0	28029
NODE_326_length_17913_cov_28.471721	NC_007439.1	94.026	703	29	9	17173	17874	1	691	0.0	1053
NODE_99499_length_1703_cov_1.860437	NC_007439.1	96.364	55	2	0	757	811	9802	9748	7.39e-21	91.6

(base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165613$ samtools faidx contigs.fasta 
(base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165613$ samtools faidx contigs.fasta NODE_326_length_17913_cov_28.471721 > SRR3165613.fasta

15747-17912: control region, neat!!!!
	D-loop	15747..17912	note	control region

base) ➜  SRR3165613 git:(main) ✗ samtools faidx NODE_326_length_17913_cov_28_471721.fa NODE_326_length_17913_cov_28.471721:15747-17912 
\>SRR3165613
ATATAATTATACATATATGTATTTACACCATGCATTTATGCTAACCAATAAGTGAATATA
TGGTGTAAATGCATATATATATATATGTATTAATCAACATCTGCTTGTAATCAAAGGACA
TATGTGCATTCAATGGTATTCGTAAATGTAATGTACGGTGACCTATAATTAATGTACTTT
AAACAATAATATTACATATTAATCATCAACAATAATTATTGGGCGTAGTGAGAGATCACC
AATTAGTAGGTACTCAGAGTGTTGACGGTTCTTGATAGTCAAGGACAAACACGGCGTGGG
GGTTACACAAATTGAACTATTACTGGCATTTGGTTCCTACCTCAGGTACATGGATGAGAA
AGATTCCACTGACGTTCCTTCATAAATGGCATAAGTTAATGGTGGATAACATGAATCCTC
GTTACCCACCAAGCCGAGCATTCTTTCCAGCGGGTAAGGGGTTTAATAATTTTTTTCTTA
TCATCTCATTTGACATCTCAGAGTGCACGCGCAGAGCATATAATATGGAAGTAGAACATA
AATCTTTTTATGGAAGAGAAATAGTCAGTATGAATGGTGAAAAGACTTGACCTGAAGAAT
TACATGATAGGATATCACGAGCATAACGTGCTAACCTTCTACTCCCAACATATCTATTAA
TCCACCCCGTTTTTATTACGCAACCCCCCCCCCCCCCCCCCACGAAAGTTCTTTTATCTC
ACCACACAAATTTTTCAACAATTATTCATACTTTGGGTGAGGTGTTAGTTTTTCGTCATA
ATTTTAGACACCCACACGCACACAAAGCCACACATTCATCCCGCGCACACATACACCACC
TCCATTCTCGGCATCACAAGGCCACCTCTTCTTCTCTCCAGCCCCCACCCTCCTCTCCCT
CTTCCTCTCTGTCTCTCTCTCTTTTCTTTCCTGCTCTCACTTCCCGCCTGCCCTTCTCCC
CCCTTTCCCATTATTTCATCTGTTCTTTTTTTTTTTTTCCTCTTGCCTCCTCATCTCACA
TCCTTTTCTCTCCTCACCTTTTTCCCGTTCCTTCACTTCACCCTGGCCCCCGTCCTCCCT
GTGCCCTCTGCTCCGTCCCAGCTGCATCTCCCTCCCTCGCTTTTTTCCCTCCCACTCCTT
AATCTTCCACCTCTCCTCCTCCTCCACCACCCCCTCCTCTGCTGTAAGTGGGCCAGGTGC
CTCCTGTGGCCTCCCCTACTCCATTTCTCATTTTTACCCTTCCTCTACTCCCTTTCTCAT
TTTTACCCTCTCCCTGTTCCCTTTCTCATTTTTATCCTTCCCCCTGACCCTAGCTCATAT
TTATACATCCTGTTAATGCAACAGCTCGCGTCAATGGACGACAGGTCCAGTTTTCCTGGT
AAAACACAAACATTTTTTGGCCTGATGCACCATTAGTTGCAAAGTTGAGCCGTCCAAGTT
AGAACAACTTCTCTCTTGAGGTTAGAGGGTGATGTCTCAAAGAGGGGAGTGGGTGATTTT
GAAATTAAAAGGCGTTTTCAGCACTGAAAGACTCATTCTTAGTATGTATCATTTCGTGAA
CTTTATAAATCCGCGATGAAAGTGCTGCTTTGTCATTGTTGCTGGACATTTTACGAGCTC
TCCTGCGTTAGTGTGGTTCCCTTTATGAACACGAGACGTGAAGCCAATCCAAATGTGAAC
AGAAGTGAGTAAGTTCACATCCCCTTTTCAGTGGTGAGGGGCAAATTTCCCCAACGGCCT
ACTCTCCACTCCTCCCAGTGTTGCATCTGCACAAGCACACACACACACACACACACACAC
ACACACCGAAACATGCGTATACATGGAAGTCTCATTGTTTCTGTCTTTTTGTTCTCCTCT
CGCTCTGTGAATGCACACGGTGAGCTGGCAGTGCGCTTTTCACCTCATCTCATCCCCTGT
ATTTTCTTCTTTTTTTTCCCCCCACACTCTTCCTCTTTTCACTCATTGTCATTCTCTTTG
CCTTTGTCTCTGGCTGGCTCTTTCCCTTGCCCTCTTCTTTTTTCTTTTTTTTTTTTTTTT
TTCTTCCCCTTTTCCTCACCCCCCCCCCCCCCCCCCCTTTTTTTTTTCCCCCCCCCCCCC
CCCCCCCACAATAACTTGATCTTAATCTAAAAATGCACTATTTACACTATTTAAAATATT
TTAAGA

1-702 aligns at 95.6% similarity to indonesian Dloop I4

What about SRR26358744
NODE_15157_length_12789_cov_301.085588	NC_007439.1	94.275	6445	357	10	1	6438	1360	7799	0.0	9847
NODE_15157_length_12789_cov_301.085588	NC_007439.1	94.326	6362	349	10	6435	12789	7716	1360	0.0	9738

NODE_15157_length_12789_cov_301.085588	NC_007439.1	94.275	6445	357	10	1	6438	1360	7799	0.0	9847
NODE_15157_length_12789_cov_301.085588	NC_007439.1	94.326	6362	349	10	6435	12789	7716	1360	0.0	9738
NODE_36637_length_3576_cov_302.675907	NC_007439.1	88.585	3110	341	10	475	3571	14867	11759	0.0	3764
NODE_36637_length_3576_cov_302.675907	NC_007439.1	88.911	496	44	10	1	486	15188	14694	5.83e-174	601
NODE_41264_length_2603_cov_345.231196	NC_007439.1	91.660	2602	213	4	1	2600	7723	10322	0.0	3600
NODE_47635_length_1582_cov_281.047841	NC_007439.1	90.840	1583	143	2	1	1582	11830	10249	0.0	2119
NODE_47721_length_1570_cov_6.014735	NC_007439.1	94.805	77	4	0	1	77	9592	9668	8.68e-30	121
NODE_48543_length_1461_cov_296.318642	NC_007439.1	89.986	1428	139	4	37	1461	16535	15109	0.0	1842
NODE_52295_length_1044_cov_253.304033	NC_007439.1	96.429	700	24	1	346	1044	737	1436	0.0	1153
NODE_57864_length_619_cov_273.313653	NC_007439.1	82.270	282	42	8	34	309	498	219	3.21e-65	237

is fragmentary



Can I annotate? https://mitofish.aori.u-tokyo.ac.jp/annotation/input/

now to run all of them....
```{r}
#trialing indonesian sample
indo %>% mutate(Command=paste0("srun -p bigmemm -t 72:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/denovo/",Run,".sralite.1_pass_1.fastq.gz -2 data/denovo/",Run,".sralite.1_pass_2.fastq.gz -o data/denovo/",Run," > data/denovo/",Run,".out 2> data/denovo/",Run,".err &")) %>% select(Command) %>%
  write_tsv("1203.2-spades.txt")
```

module load spades 

running

Let's gather the remaining data.
```{r}
sra<-read_csv("meta/lates-wgs-SraRunInfo.csv")
remainder<-sra %>% select(Run, Experiment,SampleName,Region) %>% filter(Region!="Indonesia")
remainder %>% select(Run, Experiment,SampleName,Region) %>% group_by(Region) %>% summarize(Count=n())
```

in data/raw

```{sh, eval=FALSE}
cat ../../meta/lates-wgs-SraRunInfo.csv | grep -v Indonesia | cut -d ',' -f 10 | while read line; do wget $line; done;
```

```{r}
remainder %>% mutate(Command=paste0("fastq-dump --outdir data/denovo --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip data/raw/", Run, ".sralite.1")) %>% select(Command) %>% write_tsv(file="1203.3-split.sh", col_names = FALSE)

```


```{sh, eval=FALSE}
srun -p bigmemm --time 48:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 1203.3-split.sh
```


Check these out!!
SRR3165614
blastn -query contigs.fasta  -db ../../../genome/NC_007439.1.fasta -outfmt 6
NODE_1316_length_16642_cov_57.354736	NC_007439.1	98.642	15837	182	26	760	16569	16535	705	0.0	28024
NODE_1316_length_16642_cov_57.354736	NC_007439.1	94.220	692	30	8	69	759	683	1	0.0	1048
SRR3165615
(base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165615$ blastn -query contigs.fasta  -db ../../../genome/NC_007439.1.fasta -outfmt 6
NODE_3541_length_16628_cov_58.673565	NC_007439.1	98.661	15837	179	26	74	15883	705	16535	0.0	28040
NODE_3541_length_16628_cov_58.673565	NC_007439.1	94.067	691	34	7	15884	16573	1	685	0.0	1042
SRR3165616

NODE_6878_length_16633_cov_82.739534	NC_007439.1	98.655	15837	180	26	749	16558	16535	705	0.0	28035
NODE_6878_length_16633_cov_82.739534	NC_007439.1	93.581	701	33	9	49	748	690	1	0.0	1035
SRR3165617

base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165617$ blastn -query contigs.fasta  -db ../../../genome/NC_007439.1.fasta -outfmt 6
NODE_7945_length_16800_cov_90.730964	NC_007439.1	98.661	15837	179	26	927	16736	16535	705	0.0	28040
NODE_7945_length_16800_cov_90.730964	NC_007439.1	94.118	697	33	8	231	926	690	1	0.0	1053

SRR3183267 needs more ram
SRR3183268 needs more ram
SRR3183269 needs more ram
SRR3183270 needs more ram 486.992GB
SRR3183271 needs more ram

SRR3165612 looks fragmentary
