---
title: "1204-taiwan"
output: html_document
date: "2024-12-16"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(dartR)
library(snpR)
library(grid)
library(ggpubr)
library(phytools)
library(tanggle)
library(phangorn)
library(viridis)
library(pcadapt)
library(ape)
library(ggrepel)
```

Have some new data from Taiwan!

drwxr-sr-x 2 maccamp  49 Dec 15 15:00 AGRF_CAGRF24090039-2_22VGLNLT3/
drwxr-sr-x 2 maccamp 272 Dec 15 15:00 AGRF_CAGRF24090039-2_22VGMKLT3/

Checksums done, seem fine.

Creating linked files like so
(base) maccamp@farm:~/lates-wgs/data/taiwan/separate$ ln -s ../AGRF_CAGRF24090039-2_22VGMKLT3/WJC* ./

This is pretty fast!!
```{sh, eval=FALSE}
cat separate/WJC9162_*R1.fastq.gz > WJC9162-R1.fastq.gz
cat separate/WJC9162_*R2.fastq.gz > WJC9162-R2.fastq.gz
```

Sample list
WJC9162
WJC9163
WJC9164
WJC9166
WJC9187
WJC9196
WJC9556
WJC9557
WJC9558
WJC9559
WJC9560
WJC9188
Mainstream6
Mainstream7
Mainstream8
Mainstream9
Mainstream10
Mainstream11
Mainstream12

```{sh, eval=FALSE}
cat separate/WJC9163_*R1.fastq.gz > WJC9163-R1.fastq.gz
cat separate/WJC9164_*R1.fastq.gz > WJC9164-R1.fastq.gz
cat separate/WJC9166_*R1.fastq.gz > WJC9166-R1.fastq.gz
cat separate/WJC9187_*R1.fastq.gz > WJC9187-R1.fastq.gz
cat separate/WJC9196_*R1.fastq.gz > WJC9196-R1.fastq.gz
cat separate/WJC9556_*R1.fastq.gz > WJC9556-R1.fastq.gz
cat separate/WJC9557_*R1.fastq.gz > WJC9557-R1.fastq.gz
cat separate/WJC9558_*R1.fastq.gz > WJC9558-R1.fastq.gz
cat separate/WJC9559_*R1.fastq.gz > WJC9559-R1.fastq.gz
cat separate/WJC9560_*R1.fastq.gz > WJC9560-R1.fastq.gz
cat separate/WJC9188_*R1.fastq.gz > WJC9188-R1.fastq.gz
cat separate/Mainstream6_*R1.fastq.gz > Mainstream6-R1.fastq.gz
cat separate/Mainstream7_*R1.fastq.gz > Mainstream7-R1.fastq.gz
cat separate/Mainstream8_*R1.fastq.gz > Mainstream8-R1.fastq.gz
cat separate/Mainstream9_*R1.fastq.gz > Mainstream9-R1.fastq.gz
cat separate/Mainstream10_*R1.fastq.gz > Mainstream10-R1.fastq.gz
cat separate/Mainstream11_*R1.fastq.gz > Mainstream11-R1.fastq.gz
cat separate/Mainstream12_*R1.fastq.gz > Mainstream12-R1.fastq.gz


cat separate/WJC9163_*R2.fastq.gz > WJC9163-R2.fastq.gz
cat separate/WJC9164_*R2.fastq.gz > WJC9164-R2.fastq.gz
cat separate/WJC9166_*R2.fastq.gz > WJC9166-R2.fastq.gz
cat separate/WJC9187_*R2.fastq.gz > WJC9187-R2.fastq.gz
cat separate/WJC9196_*R2.fastq.gz > WJC9196-R2.fastq.gz
cat separate/WJC9556_*R2.fastq.gz > WJC9556-R2.fastq.gz
cat separate/WJC9557_*R2.fastq.gz > WJC9557-R2.fastq.gz
cat separate/WJC9558_*R2.fastq.gz > WJC9558-R2.fastq.gz
cat separate/WJC9559_*R2.fastq.gz > WJC9559-R2.fastq.gz
cat separate/WJC9560_*R2.fastq.gz > WJC9560-R2.fastq.gz
cat separate/WJC9188_*R2.fastq.gz > WJC9188-R2.fastq.gz
cat separate/Mainstream6_*R2.fastq.gz > Mainstream6-R2.fastq.gz
cat separate/Mainstream7_*R2.fastq.gz > Mainstream7-R2.fastq.gz
cat separate/Mainstream8_*R2.fastq.gz > Mainstream8-R2.fastq.gz
cat separate/Mainstream9_*R2.fastq.gz > Mainstream9-R2.fastq.gz
cat separate/Mainstream10_*R2.fastq.gz > Mainstream10-R2.fastq.gz
cat separate/Mainstream11_*R2.fastq.gz > Mainstream11-R2.fastq.gz
cat separate/Mainstream12_*R2.fastq.gz > Mainstream12-R2.fastq.gz

```

Two other data sets where we have WGS data
10 lakdiva
5 Mainstream Fish
10 Mainstream Fish

Previously I trimmed the dat then aligned, then downsampled to ~7x Hmmm, why not just downsample existing bams?

Need to align all our fish in hatcheries
Need to align all our fish in taiwan

bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Running aligns, then need to compute coverage and edit the meta file


Then, something like this to try

```{r}
m105<-read_csv("meta/105-Indo-split.csv") %>% mutate(NewPath=ifelse(DataDir %in% c("data/hatcheries","data/lakdiva", "data/taiwan"), paste0("data/downsample/", Run, ".reduced.bam"), Path))

down<-m105 %>%  filter(DataDir %in% c("data/hatcheries","data/lakdiva", "data/taiwan")) %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", DataDir, "/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
 down %>% select(Command) %>% write_tsv(file="1204.1-downsample.sh", col_names = FALSE)

```

Don't have oceanpick20 and oceanpick8

WJC9557 didn't seem to have worked

module load parallel
srun -p high --time 24:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 1204.1-downsample.sh

Checking for complete oceanpick 20 and oceanpick 8. Done!

We have downsampled numerous fish!!

# Compile meta for downsampled fish

```{r}
m104 <- m105 %>% filter(Run != "WJC9557")
write_csv(m104, file="meta/meta-104.csv")
m103<-filter(m104, Lineage !="japonicus")
write_csv(m103, "meta/m103.csv")
```

Conduct basic PCA of one chrom.

```{r}
m103$Region<-factor(m103$Region, levels=c("Queensland","Northern Territory","Papua New Guinea","Mainstream","OceanPick",
                                           "Indonesia-K","Indonesia-SU","Indonesia-SJ","Philippines","Taiwan",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","Sri Lanka","India Western Coast"))

m103 %>% select(NewPath) %>% write_tsv("bamlists/103.bamlist", col_names = FALSE)
meta<-m103

```



using first chrom

```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  \
-bam bamlists/103.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 93 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1204/103-ibs-90 > outputs/1204/103-ibs-90.out 2> outputs/1204/103-ibs-90.err &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/103-ibs-90.beagle.gz -K 2 -o outputs/1204/103-K2 &
srun -p high -t 07:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/103-ibs-90.beagle.gz -K 3 -o outputs/1204/103-K3 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/103-ibs-90.beagle.gz -K 4 -o outputs/1204/103-K4 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/103-ibs-90.beagle.gz -K 5 -o outputs/1204/103-K5 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/103-ibs-90.beagle.gz -K 6 -o outputs/1204/103-K6 &

# no inv
srun -p high -t 24:00:00 --cpus-per-task=6 --mem=32G --nodes=1 $HOME/angsd/angsd -P 6  \
-bam bamlists/103.bamlist -rf meta/no-inv-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 93 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1204/103-noinv-ibs-90 > outputs/1204/103-noinv-ibs-90.out 2> outputs/1204/103-noinv-ibs-90.err &

```

	-> Number of sites retained after filtering: 173003 


```{r}
m <- as.matrix(read.table("outputs/1204/103-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1204/lates-c-genome-wide-pca-12.jpeg")
```
For figure  

```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta) 

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region, shape=Lineage), size=3, alpha=0.75) +
  geom_text_repel(data=covs %>% filter(Region %in% c("Taiwan","Thailand","Phillipines")), aes(x=V1, y=V2, label=Region), max.overlaps = Inf) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "none")

pc12
```

```{r}
text13<-covs  %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region, shape=Lineage), size=3, alpha=0.75) +
 # geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.4))
ggsave("outputs/1204/combined-wgspcs.pdf", width=11, height=5)
ggsave("outputs/1204/combined-wgspcs.jpeg", width=11, height=5)
```


## Admixture Plots.  

Plot as one figure. 

Get set of cols.    
```{r}
cs<-viridis(6, option = "A", alpha = 0.75)
csdf<-cs %>% as_tibble() %>% rename(Color=value) %>% mutate(x=1:n())

ggplot(csdf) +
  geom_point(aes(x=x), fill=csdf$Color, y=1, pch=21, cex=5)+
ylim(0,2)
```

First labels     

```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```

```{r}
PlotAdmix("outputs/1204/103-K2.qopt", meta)
``` 

```{r}
PlotAdmix("outputs/1204/103-K3.qopt", meta)
```

```{r}
PlotAdmix("outputs/1204/103-K4.qopt", meta)
```

```{r}
PlotAdmix("outputs/1204/103-K5.qopt", meta)
```


```{r}
PlotAdmix("outputs/1204/103-K6.qopt", meta)
```

# Dropping hatcheries/Australia

```{r}
m64<-m103 %>% filter(!Lineage %in% c("AUS+NG","Mainstream","OceanPick") )
m64 %>% select(NewPath) %>% write_tsv("bamlists/64.bamlist", col_names = FALSE)

meta<-m64
```

```{sh, eval=FALSE}
srun -p high -t 4:00:00 --cpus-per-task=6 --mem=32G --nodes=1 $HOME/angsd/angsd -P 6  \
-bam bamlists/64.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 58 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1204/64-ibs-90 > outputs/1204/64-ibs-90.out 2> outputs/1204/64-ibs-90.err &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/64-ibs-90.beagle.gz -K 2 -o outputs/1204/64-K2 &
srun -p high -t 07:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/64-ibs-90.beagle.gz -K 3 -o outputs/1204/64-K3 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/64-ibs-90.beagle.gz -K 4 -o outputs/1204/64-K4 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/64-ibs-90.beagle.gz -K 5 -o outputs/1204/64-K5 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1204/64-ibs-90.beagle.gz -K 6 -o outputs/1204/64-K6 &

srun -p high -t 24:00:00 --cpus-per-task=6 --mem=32G --nodes=1 $HOME/angsd/angsd -P 6  \
-bam bamlists/64.bamlist -rf meta/no-inv-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 58 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1204/64-noinv-ibs-90 > outputs/1204/64-noinv-ibs-90.out 2> outputs/1204/64-noinv-ibs-90.err &
```



```{r}
m <- as.matrix(read.table("outputs/1204/64-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1204/lates-taiwan-genome-wide-pca-12.jpeg")
```
For figure  

```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta) 

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region, shape=Lineage), size=3, alpha=0.75) +
  geom_text_repel(data=covs %>% filter(Region %in% c("Taiwan","Thailand","Phillipines")), aes(x=V1, y=V2, label=Region), max.overlaps = Inf) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "none")

pc12
```

```{r}
text13<-covs  %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region, shape=Lineage), size=3, alpha=0.75) +
 # geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.4))
ggsave("outputs/1204/combined-wgspcs-IND-SEA.pdf", width=11, height=5)
ggsave("outputs/1204/combined-wgspcs-IND-SEA.jpeg", width=11, height=5)
```


admix

```{r}
PlotAdmix("outputs/1204/64-K2.qopt", meta)
``` 

```{r}
PlotAdmix("outputs/1204/64-K2.qopt", meta)
``` 

```{r}
PlotAdmix("outputs/1204/64-K3.qopt", meta)
``` 

```{r}
PlotAdmix("outputs/1204/64-K4.qopt", meta)
``` 

```{r}
PlotAdmix("outputs/1204/64-K5.qopt", meta)
``` 
```{r}
PlotAdmix("outputs/1204/64-K6.qopt", meta)
``` 
# Generate calls

```{r}
m103 %>% select(Run) %>% write_tsv("bamlists/103-sample-names.txt", col_names = FALSE)
```

Generating across non-inversion regions first.

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 93 -bam bamlists/103.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/no-inv.txt -out outputs/1204/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1204/snps-wgs.out 2> outputs/1204/snps-wgs.err &

```


inversion genotypes from all three regions

NC_066835.1:1-17500000
NC_066837.1:5247652-
NC_066852.1:3500000-

```{sh, eval=FALSE}
srun -t 8:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 93 -bam bamlists/103.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-r NC_066835.1:1-17500000 -out outputs/1204/lg03  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1204/lg03.out 2> outputs/1204/lg03.err &

srun -t 8:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 93 -bam bamlists/103.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-r NC_066837.1:5247652- -out outputs/1204/lg05  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1204/lg05.out 2> outputs/1204/lg05.err &

srun -t 8:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 93 -bam bamlists/103.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-r NC_066852.1:3500000- -out outputs/1204/lg20  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1204/lg20.out 2> outputs/1204/lg20.err &

```



Creating a reduced sample list and a vcf

```{r}
m64<-m103 %>% filter(! Region %in% c("OceanPick","Mainstream","Northern Territory","Papua New Guinea","Queensland"))

m64 %>% select(Run) %>% write_tsv("bamlists/64-sample-names.txt", col_names = FALSE)

m41<-m103 %>% filter(! Region %in% c("OceanPick","Mainstream","Northern Territory","Papua New Guinea","Queensland","Bangladesh","Sri Lanka","India Eastern Coast","India Western Coast")) 

m41 %>% select(Run) %>% write_tsv("bamlists/41-sample-names.txt", col_names = FALSE)

```

```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz


#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/103-sample-names.txt -o outputs/1204/renamed.vcf.gz outputs/1204/plink.vcf.gz

bcftools view -S bamlists/64-sample-names.txt outputs/1204/renamed.vcf.gz > outputs/1204/64.vcf
bcftools view -S bamlists/41-sample-names.txt outputs/1204/renamed.vcf.gz > outputs/1204/41.vcf

#--nsites-per-win  removing SNPs with more than 5% missing data first -e'F_MISSING>=0.05'
bcftools +fill-tags outputs/1204/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned.vcf

bcftools +fill-tags outputs/1204/64.vcf  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-64.vcf

bcftools +fill-tags outputs/1204/41.vcf  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-41.vcf

#this command is no longer functional
#bcftools +prune -l 0.20 -w 10000 outputs/1204/renamed.vcf.gz > outputs/1204/pruned-02.vcf
bcftools +prune -m 0.20 -w 10000 outputs/1204/renamed.vcf.gz > outputs/1204/pruned-02.vcf

bcftools +fill-tags outputs/1204/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.4' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-maf05-04.vcf


bcftools +fill-tags outputs/1204/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.4' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-maf10-04.vcf

bcftools +fill-tags outputs/1204/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-maf01-45.vcf

bcftools +fill-tags outputs/1204/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.40' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1204/pruned-maf01-04.vcf
```

	-> Number of sites retained after filtering: 1850831 

Converting to phylip

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned.vcf;
conda deactivate;

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-64.vcf;
conda deactivate;

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-41.vcf;
conda deactivate;

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-maf05-04.vcf;
conda deactivate;

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-maf10-04.vcf;
conda deactivate;

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-maf01-45.vcf;
conda deactivate;
```{r}
dat<-read.dna(file="outputs/1204/pruned.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned.nex")

dat<-read.dna(file="outputs/1204/pruned-64.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned-64.nex")

dat<-read.dna(file="outputs/1204/pruned-41.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned-41.nex")


dat<-read.dna(file="outputs/1204/pruned-maf05-04.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned-maf05-04.nex")

dat<-read.dna(file="outputs/1204/pruned-maf10-04.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned-maf10-04.nex")

dat<-read.dna(file="outputs/1204/pruned-maf01-45.min4.phy")
write.nexus.data(dat, file="outputs/1204/pruned-maf01-45.nex")

```


```{r}
path_to_file <- "outputs/1204/pruned-maf01-45.vcf"
filename <- read.pcadapt(path_to_file, type = "vcf")
```



_1_ choose a K   

```{r}
x <- pcadapt(input = filename, K = 20) 
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```
```{r}
m<-m103
```

```{r}
plot(x, option = "scores", pop = factor(m$Region, levels=c("Northern Territory","Queensland",
                                                           "Papua New Guinea","Mainstream","OceanPick","Indonesia-K","Indonesia-SU",
                                                           "Indonesia-SJ",
                                                           
                                      "Philippines","Taiwan","Vietnam","Cambodia","Thailand","Bangladesh","Sri Lanka",
                                              "India Eastern Coast","India Western Coast"))) 
 
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
pcb<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Region), alpha=0.95) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) 

#labels13<-pcadata %>% filter(Region %in% c("Indonesia-K","Indonesia-SU")) %>% 
 # select(Region,V1, V3) %>% group_by(Region) %>% summarize(x=mean(V1), y=mean(V3)) 

pcc<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, shape=Lineage, fill=Region), alpha=0.9) +
#  geom_text_repel(data=labels13, aes(x=x, y=y, label=Region))+
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +

  xlab("PC 1") +
  ylab("PC 3") +
  ggtitle("C") +
  theme(plot.title = element_text(size=14, face="bold")) 

pcd<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V4, shape=Lineage, fill=Region), alpha=0.9) +
#  geom_text_repel(data=labels13, aes(x=x, y=y, label=Region))+
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 4") +
  ggtitle("D") +
  theme(plot.title = element_text(size=14, face="bold")) 
```

```{r}
blank <- grid.rect(gp=gpar(col="white"))
```



```{r}
a<-ggplotGrob(ggarrange(panela, pcb, blank, ncol=3, widths=c(1,1.4,.5)))
bc<-ggplotGrob(ggarrange(pcc, pcd, ncol=2, widths=c(1,1.4)))
ggarrange(a,bc, ncol=1, heights=c(1,1))

ggsave("outputs/1204/called-snp-pcas.pdf", width=12, height=10)
```

iqtree2 -s pruned.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -nt AUTO -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo


iqtree2 -s pruned-64.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -nt AUTO -s pruned-64.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo

iqtree2 -s pruned-maf05-04.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -nt AUTO -s pruned-maf05-04.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo


iqtree2 -s pruned-maf01-45.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -nt AUTO -s pruned-maf01-45.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo

Phylogeny has better resolution by letting low variants (MAF 0.01) be present
```{r}
tree<-read.tree("outputs/1204/pruned-maf01-45.min4.phy.varsites.phy.contree")
t1<-ggtree(tree) + geom_nodelab(aes(label=node))
t1
```

```{r}
#ind<-m %>% filter(Lineage=="IND")
#t2<-root.phylo(tree,outgroup = ind$Run)
t2<-midpoint.root(tree)
t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 25)

t<-t3

t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```

```{r}
u<-t %<+% m + #bind_rows(m58, rm) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=3) + 
  geom_tiplab(aes(label=paste0(Region, " ", label), x=x+0.01), align = FALSE, size=3) +
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.15) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale(x = 0)

u

ggsave("outputs/1204/ml-phylogeny.pdf", height=12, width=8)
```


Network

```{r}
net<-read.nexus.networx("outputs/1204/pruned-maf01-45-network.nex")
```

```{r}
g<-ggsplitnet(net)  
g$data<-left_join(g$data, m, by=c("label"="Run")) 

n<-g + 
  geom_tippoint(aes(shape=Lineage, fill=Region), cex=5, alpha=1.00) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22)))

n
ggsave("outputs/1204/wgs-network.jpeg", width=12, height=8)
ggsave("outputs/1204/wgs-network.pdf", width=12, height=8)

```


```{r}
tree<-read.tree("outputs/1204/pruned-64.min4.phy.varsites.phy.contree")
t1<-ggtree(tree) + geom_nodelab(aes(label=node))
t1
```

```{r}
#ind<-m %>% filter(Lineage=="IND")
#t2<-root.phylo(tree,outgroup = ind$Run)
t2<-midpoint.root(tree)
t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 25)

t<-t3

t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```

```{r}
u<-t %<+% m + #bind_rows(m58, rm) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=3) + 
  geom_tiplab(aes(label=paste0(Region, " ", label), x=x+0.01), align = FALSE, size=3) +
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.15) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale(x = 0)

u

ggsave("outputs/1204/ml-phylogeny.pdf", height=12, width=8)
```
Can I run an admixture analysis with SnpR

```{r}
ms<-m103
ms$Region<-gsub(" ","-", ms$Region)

ms$Region<-factor(ms$Region, levels=c("Queensland","Northern-Territory","Papua-New-Guinea","Mainstream","OceanPick",
                                           "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines","Taiwan",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh","Sri-Lanka",
                                              "India-Eastern-Coast","India-Western-Coast"))
```



```{r}
m<-ms
vcf <-read.vcfR("outputs/1204/pruned-maf10-04.vcf") # from recalled and pruned snps in 1200
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(m$Region)
```


```{r}
glx<-gl
glx <- gl.compliance.check(glx) 

gl3<-gl.filter.maf(glx, threshold = 0.1)
gl3<-gl.filter.allna(gl3)
gl3
```



```{r}
snp<-import.snpR.data(gl3)
```

```{r}
structure<-plot_structure(snp, facet=c("pop"), facet.order = c("Queensland","Northern-Territory","Papua-New-Guinea","Mainstream","OceanPick",
                                           "Indonesia-K","Indonesia-SU","Indonesia-SJ","Philippines","Taiwan",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh","Sri-Lanka",
                                              "India-Eastern-Coast","India-Western-Coast"),
               k=c(1,2,3,4,5,6))
```

This plot is influenced by MAF 0.01 vs 0.05 vs 0.1

```{r}
structure$plot

ggsave("outputs/1204/all-samples-genome-wide-admix-plot.pdf", height=10, width=16)
```

## 41 or 64 Samples

```{r}
path_to_file <- "outputs/1204/pruned-64.vcf"
filename <- read.pcadapt(path_to_file, type = "vcf")
```



_1_ choose a K   

```{r}
x <- pcadapt(input = filename, K = 20) 
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```
```{r}
m<-m64
```

```{r}
plot(x, option = "scores", pop = factor(m$Region, levels=c("Northern Territory","Queensland",
                                                           "Papua New Guinea","Mainstream","OceanPick","Indonesia-K","Indonesia-SU",
                                                           "Indonesia-SJ",
                                                           
                                      "Philippines","Taiwan","Vietnam","Cambodia","Thailand","Bangladesh","Sri Lanka",
                                              "India Eastern Coast","India Western Coast"))) 
 
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
pcb<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Region), alpha=0.95) +
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) 

#labels13<-pcadata %>% filter(Region %in% c("Indonesia-K","Indonesia-SU")) %>% 
 # select(Region,V1, V3) %>% group_by(Region) %>% summarize(x=mean(V1), y=mean(V3)) 

pcc<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, shape=Lineage, fill=Region), alpha=0.9) +
#  geom_text_repel(data=labels13, aes(x=x, y=y, label=Region))+
  scale_shape_manual(values=c(21,23,23,21,21,24)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +

  xlab("PC 1") +
  ylab("PC 3") +
  ggtitle("C") +
  theme(plot.title = element_text(size=14, face="bold")) 

pcb
pcc
```

```{r}
blank <- grid.rect(gp=gpar(col="white"))
```



```{r}
m<-m64
vcf <-read.vcfR("outputs/1204/pruned-64.vcf") # from recalled and pruned snps in 1200
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(gsub(" ","-",m$Region))
```


```{r}
glx<-gl
glx <- gl.compliance.check(glx) 

gl3<-gl.filter.maf(glx, threshold = 0.05)
gl3<-gl.filter.allna(gl3)
gl3
```



```{r}
snp<-import.snpR.data(gl3)
```

```{r}
structure<-plot_structure(snp, facet=c("pop"), facet.order = c(                            "Indonesia-K","Indonesia-SU","Indonesia-SJ","Philippines","Taiwan",
                                              "Vietnam","Cambodia","Thailand","Bangladesh",
                                              "India-Eastern-Coast","Sri-Lanka","India-Western-Coast"),
               k=c(2,3,4,5,6))
```


```{r}
structure$plot

ggsave("outputs/1204/taiwan-admixture-plot.pdf")
```

Filter out WJC fish

```{r}
t<-structure$plot$data %>% tibble() %>% filter(K=="K = 3") %>% filter(pop=="Taiwan") %>% filter(Cluster=="1" & Percentage > 0.8)

t
```


```{r}
structure$plot$data %>% tibble() %>% filter(K=="K = 3") %>% filter(pop=="Taiwan") %>% filter(Cluster=="3" & Percentage > 0.2)
```

```{r}
p<-structure$plot$data %>% tibble() %>% filter(K=="K = 3") %>% filter(pop=="Taiwan")

ggplot(p, aes(x=sampID, y=Percentage)) +
  geom_col(aes(fill=Cluster)) +
  scale_fill_viridis_d(option="H", alpha=0.9) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ylab("Ancestry Component\n") +
  xlab("\nSample ID") +
  theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1))

ggsave("outputs/1204/taiwanese-samples-structure-plot.pdf")
```
