---
title: "1300-broome"
output: html_document
date: "2025-02-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(RcppCNPy)
library(pcadapt)
library(ggrepel)
library(viridis)
library(vcfR)
library(dartR)
library(snpR)
```

downloading broome data

~/lates-wgs/data/broome

Also putting on barbera

Let's rename and conduct a basic analysis.

_1_ rename

_2_ align
bash ../../doAlign-zipped.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Have stats in meta114.

```{r}
m114<-read_csv("meta/m114.csv") 

```


_3_ Downsample.  

```{r}

down<-m114 %>% filter(Region=="Western Australia") %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/broome/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
 down %>% select(Command) %>% write_tsv(file="1300.1-downsample.sh", col_names = FALSE)
```

module load parallel
srun -p high --time 4:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < 1300.1-downsample.sh

_4_ Look at basic popgen in Australasia

```{r}
m40<-m114 %>% filter(Lineage %in% c("AUS","Mainstream")) %>% mutate(Individual=1:n())

m40 %>% select(NewPath) %>% write_tsv("bamlists/40.bamlist", col_names=FALSE)
m40 %>% select(Run) %>% write_tsv("bamlists/40-sample-names.txt", col_names=FALSE)

meta<-m40
```

Chrom01 then all chroms
```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/40.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1300/40-ibs-90 > outputs/1300/40-ibs-90.out 2> outputs/1300/40-ibs-90.err &

srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/40.bamlist -rf meta/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1300/40-ibs-90 > outputs/1300/40-ibs-90.out 2> outputs/1300/40-ibs-90.err &
```

	-> Number of sites retained after filtering: 75260 (chrom01)
	-> Number of sites retained after filtering: 1953593 


```{r}
m <- as.matrix(read.table("outputs/1300/40-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8))

ggsave("outputs/1300/Australasian-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("Genome-Wide PCA of Australasian L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1300/australasian-lates-pc12.jpeg")

text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("Genome-Wide PCA of Australasian L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

```

Testing for kin:
conda activate py27
(py27) maccamp@farm:~/lates-wgs$ python ../pcangsd/pcangsd.py -kinship  -beagle outputs/1300/40-ibs-90.beagle.gz -o outputs/1300/40-kin

```{sh,eval=FALSE}
conda activate py27
srun -p high -t 2:00:00 --mem=32G --nodes=1 --cpus-per-task=6 python ../pcangsd/pcangsd.py -threads 6 -kinship  -beagle outputs/1300/40-ibs-90.beagle.gz -o outputs/1300/40-kin

srun -p high -t 2:00:00 --mem=32G --nodes=1 --cpus-per-task=6 python ../pcangsd/pcangsd.py -threads 6  -beagle outputs/1300/40-ibs-90.beagle.gz -relate outputs/1300/40-kin.kinship.npy -relate_tole 0.0625 -o outputs/1300/40-kinremoved

conda deactivate
```

```{r}
kin<-npyLoad("outputs/1300/40-kin.kinship.npy")
#Convert upper half of matrix to data frame compatible 
kinrind<-which(upper.tri(kin, diag = TRUE), arr.ind = TRUE)
kin<-cbind(kinrind, kin[kinrind])

kindf<-as_tibble(kin) %>%
  rename(Ind1=row, Ind2=col, Kinship=V3) %>%
  filter(Ind1 != Ind2) %>%
  mutate(Mean=mean(Kinship))

meta<-m40
df<-kindf %>% left_join(meta, by=c("Ind1"="Individual")) %>% rename(Individual1=Run, Location1=Region) %>%
  left_join(meta, by=c("Ind2"="Individual")) %>% rename(Individual2=Run, Location2=Region)

df %>% filter(Kinship>0.0625) %>% select(Kinship, Individual1, Individual2)

```


### Generate calls

```{r}
m66<-m114 %>% filter(Lineage %in% c("AUS","Mainstream","SEA")) %>% filter(Region %in% c("Indonesia-K","Philippines","Indonesia-SJ","Taiwan","Northern Territory","Papua New Guinea","Western Australia","Queensland","Mainstream"))

m66 %>% select(NewPath) %>% write_tsv("bamlists/66.bamlist", col_names=FALSE)
m66 %>% select(Run) %>% write_tsv("bamlists/66-sample-names.txt", col_names=FALSE)

```


Doing it genome wide for fun! Including some SE Asian gene pools

```{sh, eval=FALSE}

srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 36 -bam bamlists/40.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/lates-lgs.txt -out outputs/1300/snps-40-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/1300/snps-40-01.out 2> outputs/1300/snps-40-01.err &
# 	-> Number of sites retained after filtering: 1247772 


srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=6 $HOME/angsd/angsd -nthreads 6 \
-minInd 59 -bam bamlists/66.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/lates-lgs.txt -out outputs/1300/snps-66-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/1300/snps-66-01.out 2> outputs/1300/snps-66-01.err &
```


```{sh, eval=FALSE}
plink --tped snps-40-01.tped --tfam snps-40-01.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

bcftools reheader --samples bamlists/40-sample-names.txt -o outputs/1300/renamed-01.vcf.gz outputs/1300/plink-01.vcf.gz

#Using random seed: 1740369372

bcftools +fill-tags outputs/1300/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1300/pruned-01-maf05-45.vcf

#Using random seed: 1740372212

bcftools +fill-tags outputs/1300/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1300/pruned-01-maf01-45.vcf

#Using random seed: 1740372420
bcftools +fill-tags outputs/1300/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1300/pruned-01-maf10-45.vcf
```

paste \
<(bcftools query -f '[%SAMPLE\t]\n' pruned-01-maf05-45.vcf | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' pruned-01-maf05-45.vcf | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2)

Does any fish have systematically high missing data?

<(bcftools query -f '[%SAMPLE\t]\n' pruned-01-maf05-45.vcf | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' pruned-01-maf05-45.vcf | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2)
SRR3165618	0.0174943
SRR3165619	0.0237966
SRR3165620	0.0341193
SRR3165621	0.0205368
SRR3165622	0.0181463
SRR3165623	0.0236879
SRR3165629	0.0156471
SRR3165630	0.0252092
SRR3165631	0.0324894
SRR3165632	0.0298816
SRR3165633	0.0283603
SRR3165634	0.0194502
SRR3165624	0.0142345
SRR3165625	0.0273824
SRR3165626	0.021732
SRR3165627	0.0203195
SRR3165628	0.0253178
Mainstream1	0.0141258
Mainstream2	0.0196675
Mainstream3	0.0304249
Mainstream4	0.0122786
Mainstream5	0.0155384
Mainstream6	0.0222753
Mainstream7	0.0210801
Mainstream8	0.0147778
Mainstream9	0.00641095
Mainstream10	0.0110833
Mainstream11	0.00988808
Mainstream12	0.010866
Broome01	0.0124959
Broome02	0.0139085
Broome03	0.0127132
Broome04	0.0135825
Broome05	0.0119526
Broome06	0.0133652
Broome07	0.0146691
Broome08	0.0119526
Broome09	0.0137998
Broome10	0.0129306
Fitzroy01	0.0136912

```{r}
path_to_file <- "outputs/1300/pruned-01-maf05-45.vcf"
filename <- read.pcadapt(path_to_file, type = "vcf")
```



_1_ choose a K   

```{r}
x <- pcadapt(input = filename, K = 10) 
var<-round(x$singular.values^2*100,2)
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```
```{r}
m<-m40
```

```{r}
plot(x, option = "scores", pop = factor(m$Region))
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
cols<-viridis(17,option = "H")
text12<-pcadata %>% group_by(Region) %>% summarize(x=mean(V1),y=mean(V2))

ggplot(pcadata) +
  geom_jitter(aes(x=V1, y=V2, shape=Lineage, fill=Locality), alpha=0.95, cex=2) +
  geom_label_repel(data=text12, aes(x=x, y=y, label=Region), alpha=0.8) +
 # geom_text_repel(data=pcadata %>% filter(Region %in% c("Taiwan")), aes(x=V1, y=V2,label=Run)) +
  scale_shape_manual(values=c(21,21)) + 
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
 # theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("PC1 ",var[1],"%")) +
  ylab(paste0("PC2 ",var[2],"%")) +
  scale_fill_viridis_d(option="magma")

ggsave("outputs/1300/called-snps-pca.pdf", width=6, height=5)
```
9,203 variants
```{sh, eval=FALSE}
plink --vcf pruned-01-maf05-45.vcf --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb

for K in 1 2 3 4 5 6 7 8; \
do ~/github/admixture/admixture --cv myplink2.ped $K | tee log${K}.out; done

```


### Scores
```{sh, eval=FALSE}
grep -h CV log*.out | cut -f 3,4 -d ' '  | perl -pe 's/://g' | perl -pe s'/\(|\)//g' > scores.txt 
```


```{r}
scores<-read_delim("outputs/1300/scores.txt", col_names = c("K","Cross-Validation Error")) %>% mutate(x=1:n())

ggplot(scores) +
  geom_line(aes(x=x, y=`Cross-Validation Error`)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=(1:nrow(scores)))
```


```{r, echo=FALSE}
PlotAdmixture<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```

```{r}
PlotAdmixture("outputs/1300/myplink2.2.Q", m40)
```

```{r}
PlotAdmixture("outputs/1300/myplink2.3.Q", m40)
```


```{r}
PlotAdmixture("outputs/1300/maf10/myplink2.3.Q", m40)
```


```{r}
PlotAdmixture("outputs/1300/myplink2.5.Q", m40)
```

Let's calculate individual heterozygosity.
```{r}
m<-m40
vcf <-read.vcfR("outputs/1300/pruned-01-maf05-45.vcf") 


gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(gsub(" ","-",m$Region))
```
```{r}
#glt<-gl[gl$pop %in% c("Taiwan","Thailand"),]

glx<-gl
glx <- gl.compliance.check(glx) 

gl3<-gl.filter.maf(glx, threshold = 0.05)
#gl3<-gl.filter.allna(gl3)
gl3<-glx
```


```{r}
snp<-import.snpR.data(gl3)

structure<-plot_structure(snp, facet = c("pop"),
               k=c(2,3,4,5))

structure$plot

ggsave("outputs/1300/snmf-admixture-plot.pdf")
```

```{r}
hdf<-gl.report.heterozygosity(gl3, method="ind")
```
```{r}
ggplot(hdf) +
  geom_col(aes(x=ind.name, y=Ho)) +
  theme(axis.text=element_text(angle=45, hjust=1, vjust=1))
```


## Filter to a couple pops and run new hybrids

```{r}
t<-m40 %>% mutate(NewPop=ifelse(Run %in% c("Broome01","Fitzroy01"), "WA-Hybrids", gsub(" ", "-",Region)))
gls<-gl
gls$pop<-as.factor(t$NewPop)
glt<-gls[gls$pop %in% c("Northern-Territory","Papua-New-Guinea","Queensland","Western-Australia","WA-Hybrids"),]
glt2 <- gl.compliance.check(glt) 

glt3<-gl.filter.maf(glt2, threshold = 0.2)
```
```{r}
snp2<-import.snpR.data(glt3)
plot_clusters(snp2, facets = c("pop"))
```
```{r}
hyb3 <- gl.nhybrids(glt3,
p0=c("Northern-Territory","Queensland","Papua-New-Guinea"),
p1=c("Western-Australia"),
method="rand",
nhyb.directory="~/github/newhybrids", # Specify as necessary
outpath="~/github/lates-wgs/outputs/1300",  # Specify as necessary, usually getwd() [= workspace]
BurnIn=100000,
sweeps=50000,
verbose=3,
plot=FALSE)
```


```{r}
out<-read_csv("outputs/1300/aa-PofZ.csv", col_names = c("Sequence ID","Region","NoName","P0","P1","F1","F2","0-BX","1-BX"), skip=1)

long<-out %>% pivot_longer(cols = 4:9, names_to = c("Class"), values_to = c("Prob"))


long$Region<-factor(long$Region, levels=c("Papua-New-Guinea","Northern-Territory","Queensland","WA-Hybrids","Western-Australia"))


long$Class<-factor(long$Class, levels=c("P0","P1","F1","F2","0-BX","1-BX"))
l2<-long %>% arrange(Region)
l2$`Sequence ID`<-as.factor(l2$`Sequence ID`)

ggplot(l2) + 
  geom_tile(aes(x=`Sequence ID`, y=Class, fill=Prob)) +
  scale_fill_viridis_c(direction=-1, option="plasma") +
  facet_grid(.~Region, scales="free_x", space="free" ) +
  theme_bw()+
  theme(panel.grid=element_blank()) +
  theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1)) +
  xlab("Sequence ID") 

ggsave("outputs/1300/wa-hybridization.pdf", width=12, height=6)
ggsave("outputs/1300/wa-hybridization.jpeg", width=12, height=6)

```
