---
title: "1300-broome"
output: html_document
date: "2025-02-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(RcppCNPy)
```

downloading broome data

~/lates-wgs/data/broome

Also putting on barbera

Let's rename and conduct a basic analysis.

_1_ rename

_2_ align
bash ../../doAlign-zipped.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Have stats in meta114.

```{r}
m114<-read_csv("meta/m114.csv") 

```


_3_ Downsample.  

```{r}

down<-m114 %>% filter(Region=="Western Australia") %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/broome/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
 down %>% select(Command) %>% write_tsv(file="1300.1-downsample.sh", col_names = FALSE)
```

module load parallel
srun -p high --time 4:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < 1300.1-downsample.sh

_4_ Look at basic popgen in Australasia

```{r}
m40<-m114 %>% filter(Lineage %in% c("AUS","Mainstream")) %>% mutate(Individual=1:n())

m40 %>% select(NewPath) %>% write_tsv("bamlists/40.bamlist", col_names=FALSE)
meta<-m40
```

Chrom01 then all chroms
```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/40.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1300/40-ibs-90 > outputs/1300/40-ibs-90.out 2> outputs/1300/40-ibs-90.err &

srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/40.bamlist -rf meta/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1300/40-ibs-90 > outputs/1300/40-ibs-90.out 2> outputs/1300/40-ibs-90.err &
```

	-> Number of sites retained after filtering: 75260 (chrom01)
	-> Number of sites retained after filtering: 1953593 


```{r}
m <- as.matrix(read.table("outputs/1300/40-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8))

ggsave("outputs/1300/Australasian-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("Genome-Wide PCA of Australasian L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1300/australasian-lates-pc12.jpeg")

text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("Genome-Wide PCA of Australasian L. calcarifer WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

```

Testing for kin:
conda activate py27
(py27) maccamp@farm:~/lates-wgs$ python ../pcangsd/pcangsd.py -kinship  -beagle outputs/1300/40-ibs-90.beagle.gz -o outputs/1300/40-kin

```{sh,eval=FALSE}
conda activate py27
srun -p high -t 2:00:00 --mem=32G --nodes=1 --cpus-per-task=6 python ../pcangsd/pcangsd.py -threads 6 -kinship  -beagle outputs/1300/40-ibs-90.beagle.gz -o outputs/1300/40-kin

srun -p high -t 2:00:00 --mem=32G --nodes=1 --cpus-per-task=6 python ../pcangsd/pcangsd.py -threads 6  -beagle outputs/1300/40-ibs-90.beagle.gz -relate outputs/1300/40-kin.kinship.npy -relate_tole 0.0625 -o outputs/1300/40-kinremoved

conda deactivate
```

```{r}
kin<-npyLoad("outputs/1300/40-kin.kinship.npy")
#Convert upper half of matrix to data frame compatible 
kinrind<-which(upper.tri(kin, diag = TRUE), arr.ind = TRUE)
kin<-cbind(kinrind, kin[kinrind])

kindf<-as_tibble(kin) %>%
  rename(Ind1=row, Ind2=col, Kinship=V3) %>%
  filter(Ind1 != Ind2) %>%
  mutate(Mean=mean(Kinship))

meta<-m40
df<-kindf %>% left_join(meta, by=c("Ind1"="Individual")) %>% rename(Individual1=Run, Location1=Region) %>%
  left_join(meta, by=c("Ind2"="Individual")) %>% rename(Individual2=Run, Location2=Region)

df %>% filter(Kinship>0.0625) %>% select(Kinship, Individual1, Individual2)

```


### Generate calls

```{r}
m66<-m114 %>% filter(Lineage %in% c("AUS","Mainstream","SEA")) %>% filter(Region %in% c("Indonesia-K","Philippines","Indonesia-SJ","Taiwan","Northern Territory","Papua New Guinea","Western Australia","Queensland","Mainstream"))

m66 %>% select(NewPath) %>% write_tsv("bamlists/66.bamlist", col_names=FALSE)
m66 %>% select(Run) %>% write_tsv("bamlists/66-sample-names.txt", col_names=FALSE)

```


Doing it genome wide for fun!

```{sh, eval=FALSE}

srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 36 -bam bamlists/40.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/lates-lgs.txt -out outputs/1204/snps-40-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/1300/snps-40-01.out 2> outputs/1300/snps-40-01.err &

srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 49 -bam bamlists/54.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/lates-lgs.txt -out outputs/1204/snps-54-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/1300/snps-54-01.out 2> outputs/1300/snps-54-01.err &
```



Include Southeast Asian fish

```{r}
m91<-m114 %>% filter(Lineage %in% c("AUS","Mainstream","SEA","OceanPick")) 
m91 %>% select(NewPath) %>% write_tsv("bamlists/91.bamlist", col_names=FALSE)
m91 %>% select(Run) %>% write_tsv("bamlists/91-sample-names.txt", col_names=FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/91.bamlist -rf meta/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 82 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1300/91-ibs-90 > outputs/1300/91-ibs-90.out 2> outputs/1300/91-ibs-90.err &



```

