---
title: "1301-uwisara"
output: html_document
date: "2025-04-07"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(dartR)
library(snpR)
library(grid)
library(ggpubr)
library(phytools)
library(tanggle)
library(phangorn)
library(viridis)
library(pcadapt)
library(ape)
library(ggrepel)
```

putting in data/uwisara

-at some point I can clean up the farm a bit

Two lanes of each, that should be combined. 

1613156
1613186
1613217
1613247

```{sh, eval=FALSE}
#checksums
 maccamp@farm:~/lates-wgs/data/uwisara/AGRF_NXGSQCAGRF24120169-1_22YMCCLT3$ for f in *.fastq.gz; do echo $f; md5sum $f  >> md5s.txt; done;

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613156_22YMCCLT3_TTGGACATCG-GAATCTGGTC_L007_R1.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613156_22YMCCLT3_TTGGACATCG-GAATCTGGTC_L008_R1.fastq.gz > 1613156-R1.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613156_22YMCCLT3_TTGGACATCG-GAATCTGGTC_L007_R2.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613156_22YMCCLT3_TTGGACATCG-GAATCTGGTC_L008_R2.fastq.gz > 1613156-R2.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613186_22YMCCLT3_AACGTTGAGA-CGGTCGGTAA_L007_R1.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613186_22YMCCLT3_AACGTTGAGA-CGGTCGGTAA_L008_R1.fastq.gz > 1613186-fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613186_22YMCCLT3_AACGTTGAGA-CGGTCGGTAA_L007_R2.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613186_22YMCCLT3_AACGTTGAGA-CGGTCGGTAA_L008_R2.fastq.gz > 1613186-R2.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613217_22YMCCLT3_GGCCAGTGAA-TCTGCTAATG_L007_R1.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613217_22YMCCLT3_GGCCAGTGAA-TCTGCTAATG_L008_R1.fastq.gz > 1613217-R1.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613217_22YMCCLT3_GGCCAGTGAA-TCTGCTAATG_L007_R2.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613217_22YMCCLT3_GGCCAGTGAA-TCTGCTAATG_L008_R2.fastq.gz > 1613217-R2.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613247_22YMCCLT3_ATGTCTCCGG-CTCTTATTCG_L007_R1.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613247_22YMCCLT3_ATGTCTCCGG-CTCTTATTCG_L008_R1.fastq.gz > 1613247-R1.fastq.gz

cat AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613247_22YMCCLT3_ATGTCTCCGG-CTCTTATTCG_L007_R2.fastq.gz AGRF_NXGSQCAGRF24120169-1_22YMCCLT3/1613247_22YMCCLT3_ATGTCTCCGG-CTCTTATTCG_L008_R2.fastq.gz > 1613247-R2.fastq.gz

```

bash ../../doAlign-zipped.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

```{r}
m118<-read_csv("meta/m118.csv")
m118 %>% select(NewPath) %>% write_tsv("bamlists/118.bamlist", col_names = FALSE)
ind<-m118 %>% filter(Lineage %in% c("IND","uwisara","lakdiva"))
ind  %>% select(NewPath) %>% write_tsv("bamlists/27.bamlist", col_names = FALSE)
down<-m118 %>% filter(Region=="Myanmar") %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/uwisara/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
 down %>% select(Command) %>% write_tsv(file="1301.1-downsample.sh", col_names = FALSE)
```

# Trial of Chromosome 1 to visualize PCA of all samples.


```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/118.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 106 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1301/118-ibs-90 > outputs/1301/118-ibs-90.out 2> outputs/1301/118-ibs-90.err &




srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/118.bamlist -rf meta/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 106 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1301/118-ibs-90 > outputs/1301/118-ibs-90.out 2> outputs/1301/118-ibs-90.err &

#	-> Number of sites retained after filtering: 160916 

```



```{r}
m <- as.matrix(read.table("outputs/1301/118-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8))

ggsave("outputs/1301/range-wide-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(m118)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Lineage), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("PCA of Lates WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1301/lates-range-wide-pc12.jpeg")

text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("PCA of Lates WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

```

# Indian Ocean ones

```{sh}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/27.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 24 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1301/27-ibs-90 > outputs/1301/27-ibs-90.out 2> outputs/1301/27-ibs-90.err &


#	-> Number of sites retained after filtering: 216071 


srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/27.bamlist -rf meta/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 24 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1301/27-ibs-90 > outputs/1301/27-ibs-90.out 2> outputs/1301/27-ibs-90.err &

```



```{r}
m <- as.matrix(read.table("outputs/1301/27-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8))

ggsave("outputs/1301/indian-ocean-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(ind)

text12<-covs %>% dplyr::select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Lineage), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("PCA of Lates WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

ggsave("outputs/1301/indian-ocean-lates-pc12.jpeg")

text13<-covs %>% dplyr::select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Lineage), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("PCA of Lates WGS Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

```


## uwisara mtDNA

Lets make assemblies of these fishes.

module load spades
srun -p bigmemm -t 24:00:00 --mem=240GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/bangladesh/split/SRR26358744_pass_1.fastq.gz  -2 data/bangladesh/split/SRR26358744_pass_2.fastq.gz -o data/denovo/SRR26358744 

No spades module, creating conda environment spades
mamba install spades
is spades.py

Create commands:

```{sh, eval=FALSE}
conda activate spades
#trialing indonesian sample
ind %>% filter(Lineage=="uwisara") %>% mutate(Command=paste0("srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/",Run,"-R1.fastq.gz -2 data/uwisara/",Run,"-R2.fastq.gz -o data/denovo/",Run," > data/denovo/",Run,".out 2> data/denovo/",Run,".err &")) %>% select(Command) %>%
  write_tsv("1301.2-spades.txt")
```



## Call some snps

Let's not include OceanPick or Mainstream fish. Compare to 1204

```{r}
m118 %>% filter(!Region %in% c("Mainstream","OceanPick")) %>% select(NewPath) %>% write_tsv("bamlists/96.bamlist", col_names = FALSE)
m118 %>% filter(!Region %in% c("Mainstream","OceanPick")) %>% select(Run) %>% write_tsv("bamlists/96.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 86 -bam bamlists/96.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/no-inv.txt -out outputs/1301/snps-wgs-01-glf  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1301/snps-wgs-01-glf.out 2> outputs/1301/snps-wgs-01-glf.err &
```

Possible commands
	-> Number of sites retained after filtering: 7625790 


```{sh}
plink --tped snps-wgs-01-glf.tped --tfam snps-wgs-01-glf.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/96.names -o outputs/1301/renamed-01.vcf.gz outputs/1301/plink-01.vcf.gz

bcftools +fill-tags outputs/1301/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1301/pruned-01-maf05-45.vcf

bcftools +fill-tags outputs/1301/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1301/pruned-01-maf01-45.vcf

bcftools +fill-tags outputs/1301/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1301/pruned-01-maf10-45.vcf

#Convert to phylip and then to nex
conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf01-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45.vcf;

conda deactivate;
```

```{r}
dat<-read.dna(file="outputs/1301/pruned-01-maf05-45.min4.phy")
write.nexus.data(dat, file="outputs/1301/pruned-01-maf05-45.nex")

dat<-read.dna(file="outputs/1301/pruned-01-maf01-45.min4.phy")
write.nexus.data(dat, file="outputs/1301/pruned-01-maf01-45.nex")

dat<-read.dna(file="outputs/1301/pruned-01-maf10-45.min4.phy")
write.nexus.data(dat, file="outputs/1301/pruned-01-maf10-45.nex")
```



```{r}
path_to_file <- "outputs/1301/pruned-01-maf05-45.vcf"

filename <- read.pcadapt(path_to_file, type = "vcf")
```



_1_ choose a K   

```{r}
x <- pcadapt(input = filename, K = 20) 
```

```{r}
var<-round(x$singular.values^2*100,2)
var
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```

```{r}
m<-m118 %>% filter(!Region %in% c("Mainstream","OceanPick"))
```

```{r}
plot(x, option = "scores", pop = factor(m$Region, levels=c("Northern Territory","Queensland",
                                                           "Papua New Guinea","Indonesia-K","Indonesia-SU",
                                                           "Indonesia-SJ",
                                                           
                                      "Philippines","Taiwan","Vietnam","Cambodia","Thailand","Myanmar","Bangladesh","Sri Lanka",
                                              "India East Coast","India West Coast"))) 
 
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Region), alpha=0.95) +
  #geom_text_repel(data=pcadata %>% filter(Region %in% c("Cambodia")), aes(x=V1, y=V2,label=Run)) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") 

ggplot(pcadata) +
  geom_point(aes(x=V1, y=V4, shape=Lineage, fill=Region), alpha=0.95) +
 # geom_text_repel(data=pcadata %>% filter(Region %in% c("Queensland")), aes(x=V1, y=V4,label=Run)) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") 
```

```{r}
pcb<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Region), alpha=0.95) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) 

#labels13<-pcadata %>% filter(Region %in% c("Indonesia-K","Indonesia-SU")) %>% 
 # select(Region,V1, V3) %>% group_by(Region) %>% summarize(x=mean(V1), y=mean(V3)) 

pcc<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, shape=Lineage, fill=Region), alpha=0.9) +
#  geom_text_repel(data=labels13, aes(x=x, y=y, label=Region))+
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +

  xlab("PC 1") +
  ylab("PC 3") +
  ggtitle("C") +
  theme(plot.title = element_text(size=14, face="bold")) 

pcd<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V4, shape=Lineage, fill=Region), alpha=0.9) +
#  geom_text_repel(data=labels13, aes(x=x, y=y, label=Region))+
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 4") +
  ggtitle("D") +
  theme(plot.title = element_text(size=14, face="bold")) 
```

```{r}
blank <- grid.rect(gp=gpar(col="white"))
```



```{r}
a<-ggplotGrob(ggarrange(panela, pcb, blank, ncol=3, widths=c(1,1.2,.5)))
bc<-ggplotGrob(ggarrange(pcc, pcd, ncol=2, widths=c(1,1.4)))
ggarrange(a,bc, ncol=1, heights=c(1,1))

ggsave("outputs/1301/called-snp-pcas.pdf", width=12, height=10)
```



## Trees

in outputs/1301/phylo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf01-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf01-45.min4.phy.varsites.phy --seqtype DNA --redo


iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf05-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf05-45.min4.phy.varsites.phy --seqtype DNA --redo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45.min4.phy.varsites.phy --seqtype DNA --redo


## IO only

filter renamed vcf

