---
title: "1302-de-novo-mtdna"
output: html_document
date: "2025-06-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ape)
library(ggtree)
```

```{r}
m118<-read_csv("meta/m118.csv")
```

see 1203 for a generation of paths


https://github.com/Kinggerm/GetOrganelle

Let's run this a few times and see what we get!!!

conda create -n getorganelle
conda activate getorganelle
conda install -c bioconda getorganelle
get_organelle_config.py -a animal_mt

Basic approach:
get_organelle_from_reads.py -1 forward.fq -2 reverse.fq -R 10 -k 21,45,65,85,105 -F animal_mt -o animal_mt_out   


Basic commands from spades:
srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/1613156-R1.fastq.gz -2 data/uwisara/1613156-R2.fastq.gz -o data/denovo/1613156 > data/denovo/1613156.out 2> data/denovo/1613156.err &
srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/1613186-R1.fastq.gz -2 data/uwisara/1613186-R2.fastq.gz -o data/denovo/1613186 > data/denovo/1613186.out 2> data/denovo/1613186.err &
srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/1613217-R1.fastq.gz -2 data/uwisara/1613217-R2.fastq.gz -o data/denovo/1613217 > data/denovo/1613217.out 2> data/denovo/1613217.err &
srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/1613247-R1.fastq.gz -2 data/uwisara/1613247-R2.fastq.gz -o data/denovo/1613247 > data/denovo/1613247.out 2> data/denovo/1613247.err &

```{sh, eval=FALSE}
conda activate getorganelle

srun -p high -t 48:00:00 --mem=62GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/uwisara/1613156-R1.fastq.gz -2 data/uwisara/1613156-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/1613156
```

Total cost 17105.34 s
2025-06-27 00:10:34,814 - INFO: Writing PATH1 of complete animal_mt to data/harvest/1613156/animal_mt.K105.complete.graph1.1.path_sequence.fasta

putting in outputs/1302/1613156.fasta and annotating.... Looks complete!

Set up additional samples
```{sh, eval=FALSE}

conda activate getorganelle

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/uwisara/1613186-R1.fastq.gz -2 data/uwisara/1613186-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/1613186 > data/harvest/1613186.out 2> data/harvest/1613186.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/uwisara/1613217-R1.fastq.gz -2 data/uwisara/1613217-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/1613217 > data/harvest/1613217.out 2> data/harvest/1613217.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/uwisara/1613247-R1.fastq.gz -2 data/uwisara/1613247-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/1613247 > data/harvest/1613247.out 2> data/harvest/1613247.err &

```

Pulled out CR

in data/cr

combined, but dropped 1613247 because lacks a big indel in the others:
>1613247 D-loop
ATACTGAACTTATATTACGTAAATACTAAATACATGTATGTATTTTCACCATACTTATAT
GTCAACCATAATTAATCCATATGGTGTAAATACATATATGTTTAATCAACATTAGACCCT
AAATGTACTATTAATCATATTAAGTATAAAAACATAATAACTATAAATGCTTTAGCGCTT
TAATTTTTAATCCAAACATTCGATAATCTCTAGTACAAGATTCCTCTAAACATTTTAAAA
TTCAATATACTAACCCTTAATTCAATATAATGAGAGACCACCAACCAGTGAATTTCGAAG
TGTTAGCGATCCTTGATAATCAAGGACAGATAACTGTGAGGGTAACACAAATTGAATTAT
TACTGGCATTTGGTTCCTACCTCAGGTCCATAACGAATAAAACATTCCCCTAACGTTCCT
TCATAAATGGCATAAGTTAATGGTGGAATACATGAACCCTCGTTACCCACCAAGCCGAGC
CTTCTCTCCAGCGGGTAAGGGGTTTTTTCTTATTTTTTTCTCATTTCATCTTGCATATCA
CAGTGTAAAAAACAATTGAAATTAAGGTGGCACATAATTTTTTTAAATATTGTAATAAAA
GTAATGCCAGGTGAATGGTGAAAGGACTTGGTTTAGAAAAGTTGACATAATAAAATCTCA
CGAGCATAAGCATTGACTGATCTCCCAACTCCCAGTTATATCTATTACCTACCCCGTTAA
TTTTACGCGCAAAAATCTTCGCAAAGATCGGAAGAGCACACGTCTGAACTCCAACATACC
CCCGGAGGGTGGAGAGAAAAAAAACCCCCCCCCCCCCCCCCCCCCAAAACTCCCGGGATC
ACTAACACTCCTGATAAACACCCCAGAAACAGGAAAGCCCCGCGATATTCGGCCTTAATT
TAAAAATACATTATTTACACCATTTACACTATTTAAAATATTTTAAGTT

Putting it back in, because whatever, doesn't seem to matter.



(base) ➜  cr git:(main) ✗ iqtree2 -m MFP -bb 10000 -s comb.afasta --redo

Trying with all the seqs...
cat vij-popset-cr.fasta uwisara-cr.fasta > popset-uwisara.fas
 mafft popset-uwisara.fas > popset-uwisara.afas
 iqtree2 -m MFP -bb 10000 -s popset-uwisara.afas --redo
 
 
## Lates lakdiva!
data/lakdiva

Mannar1
Trinco1
Jaffna1


Set up additional samples
```{sh, eval=FALSE}

conda activate getorganelle

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Jaffna1-R1.fastq.gz -2 data/lakdiva/Jaffna1-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Jaffna1 > data/harvest/Jaffna1.out 2> data/harvest/Jaffna1.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Mannar1-R1.fastq.gz -2 data/lakdiva/Mannar1-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Mannar1 > data/harvest/Mannar1.out 2> data/harvest/Mannar1.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Trinco1-R1.fastq.gz -2 data/lakdiva/Trinco1-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Trinco1 > data/harvest/Trinco1.out 2> data/harvest/Trinco1.err &

```


What about Wallacea---

I have de novo assembled this previously...
SRR3165613 is from Kalimantan

>SRR3165613-spades
ATATAATTATACATATATGTATTTACACCATGCATTTATGCTAACCAATAAGTGAATATA
TGGTGTAAATGCATATATATATATATGTATTAATCAACATCTGCTTGTAATCAAAGGACA
TATGTGCATTCAATGGTATTCGTAAATGTAATGTACGGTGACCTATAATTAATGTACTTT
AAACAATAATATTACATATTAATCATCAACAATAATTATTGGGCGTAGTGAGAGATCACC
AATTAGTAGGTACTCAGAGTGTTGACGGTTCTTGATAGTCAAGGACAAACACGGCGTGGG
GGTTACACAAATTGAACTATTACTGGCATTTGGTTCCTACCTCAGGTACATGGATGAGAA
AGATTCCACTGACGTTCCTTCATAAATGGCATAAGTTAATGGTGGATAACATGAATCCTC
GTTACCCACCAAGCCGAGCATTCTTTCCAGCGGGTAAGGGGTTTAATAATTTTTTTCTTA
TCATCTCATTTGACATCTCAGAGTGCACGCGCAGAGCATATAATATGGAAGTAGAACATA
AATCTTTTTATGGAAGAGAAATAGTCAGTATGAATGGTGAAAAGACTTGACCTGAAGAAT
TACATGATAGGATATCACGAGCATAACGTGCTAACCTTCTACTCCCAACATATCTATTAA
TCCACCCCGTTTTTATTACGCAACCCCCCCCCCCCCCCCCCACGAAAGTTCTTTTATCTC
ACCACACAAATTTTTCAACAATTATTCATACTTTGGGTGAGGTGTTAGTTTTTCGTCATA
ATTTTAGACACCCACACGCACACAAAGCCACACATTCATCCCGCGCACACATACACCACC
TCCATTCTCGGCATCACAAGGCCACCTCTTCTTCTCTCCAGCCCCCACCCTCCTCTCCCT
CTTCCTCTCTGTCTCTCTCTCTTTTCTTTCCTGCTCTCACTTCCCGCCTGCCCTTCTCCC
CCCTTTCCCATTATTTCATCTGTTCTTTTTTTTTTTTTCCTCTTGCCTCCTCATCTCACA
TCCTTTTCTCTCCTCACCTTTTTCCCGTTCCTTCACTTCACCCTGGCCCCCGTCCTCCCT
GTGCCCTCTGCTCCGTCCCAGCTGCATCTCCCTCCCTCGCTTTTTTCCCTCCCACTCCTT
AATCTTCCACCTCTCCTCCTCCTCCACCACCCCCTCCTCTGCTGTAAGTGGGCCAGGTGC
CTCCTGTGGCCTCCCCTACTCCATTTCTCATTTTTACCCTTCCTCTACTCCCTTTCTCAT
TTTTACCCTCTCCCTGTTCCCTTTCTCATTTTTATCCTTCCCCCTGACCCTAGCTCATAT
TTATACATCCTGTTAATGCAACAGCTCGCGTCAATGGACGACAGGTCCAGTTTTCCTGGT
AAAACACAAACATTTTTTGGCCTGATGCACCATTAGTTGCAAAGTTGAGCCGTCCAAGTT
AGAACAACTTCTCTCTTGAGGTTAGAGGGTGATGTCTCAAAGAGGGGAGTGGGTGATTTT
GAAATTAAAAGGCGTTTTCAGCACTGAAAGACTCATTCTTAGTATGTATCATTTCGTGAA
CTTTATAAATCCGCGATGAAAGTGCTGCTTTGTCATTGTTGCTGGACATTTTACGAGCTC
TCCTGCGTTAGTGTGGTTCCCTTTATGAACACGAGACGTGAAGCCAATCCAAATGTGAAC
AGAAGTGAGTAAGTTCACATCCCCTTTTCAGTGGTGAGGGGCAAATTTCCCCAACGGCCT
ACTCTCCACTCCTCCCAGTGTTGCATCTGCACAAGCACACACACACACACACACACACAC
ACACACCGAAACATGCGTATACATGGAAGTCTCATTGTTTCTGTCTTTTTGTTCTCCTCT
CGCTCTGTGAATGCACACGGTGAGCTGGCAGTGCGCTTTTCACCTCATCTCATCCCCTGT
ATTTTCTTCTTTTTTTTCCCCCCACACTCTTCCTCTTTTCACTCATTGTCATTCTCTTTG
CCTTTGTCTCTGGCTGGCTCTTTCCCTTGCCCTCTTCTTTTTTCTTTTTTTTTTTTTTTT
TTCTTCCCCTTTTCCTCACCCCCCCCCCCCCCCCCCCTTTTTTTTTTCCCCCCCCCCCCC
CCCCCCCACAATAACTTGATCTTAATCTAAAAATGCACTATTTACACTATTTAAAATATT
TTAAGAA

SRR3165614 is from Kalimantan
SRR3165612 is from Sulawesi

module load blast-plus
(base) maccamp@farm:~/lates-wgs/data/denovo/SRR3165613$ blastn -query contigs.fasta -db ~/lates-wgs/genome/NC_007439.1.fasta -outfmt 6
NODE_326_length_17913_cov_28.471721	NC_007439.1	98.649	15837	181	26	1363	17172	705	16535	0.0	28029
NODE_326_length_17913_cov_28.471721	NC_007439.1	94.026	703	29	9	17173	17874	1	691	0.0	1053

samtools faidx contigs.fasta
samtools faidx contigs.fasta NODE_326_length_17913_cov_28.471721 

SRR3165612 is fragmentary

SRR3165618 is NT
SRR3165629 is QLD
SRR3165624 is PNG

```{sh, eval=FALSE}

conda activate getorganelle

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165613.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165613.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165613 > data/harvest/SRR3165613.out 2> data/harvest/SRR3165613.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165614.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165614.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165614 > data/harvest/SRR3165614.out 2> data/harvest/SRR3165614.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165612.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165612.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165612 > data/harvest/SRR3165612.out 2> data/harvest/SRR3165612.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165618.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165618.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165618 > data/harvest/SRR3165618.out 2> data/harvest/SRR3165618.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165629.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165629.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165629 > data/harvest/SRR3165629.out 2> data/harvest/SRR3165629.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165624.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165624.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165624 > data/harvest/SRR3165624.out 2> data/harvest/SRR3165624.err &
```


These all ran to completion. Annotating with mitoannotator and putting in data/complete-mitogenomes-for-phame


let's combine and make a basic tree

```{sh, eval=FALSE}
cat vij-popset-cr.fasta lakdiva.fasta uwisara-cr.fasta wallacea.fasta australasia-cr.fasta > temp.fasta
mafft temp.fasta > temp.afasta
~/Dropbox/bin/seqConverter.pl -dtemp.afasta -on

```

neat!

paup execute
nj
save trees file=temp.tre brlens=yes

set criterion=distance
bootstrap nreps=10000 search=NJ  conLevel=50 treefile=temp.boot.tre
savetrees file=bootMajRule.tree from=1 to=1 savebootp=nodelabels brlens=yes;
gettrees file=temp.boot.tre StoreTreeWts=yes mode=3;
contree all/strict=no majrule=yes usetreewts=yes treefile=temp-major-rule.tre;

```{r}
t<-read.nexus("data/cr/temp.tre") %>% midpoint.root()
ggtree(t) %<+% m118 + geom_tiplab(size=2) +
  geom_nodelab() +
  geom_tiplab(aes(label=Locality,x=200), size=2, linetype = 0) +
  xlim(0,300) 

ggsave("outputs/1302/cr-nj-tree.pdf")
```

```{r}
nex<-read.dna("data/cr/temp.afasta",format = "fasta")
f <- function(x) nj(dist.dna(x))
tr <- f(nex) %>% midpoint.root()
nodes<-boot.phylo(tr, nex, f, quiet=FALSE, B=1000)

```
```{r}
boot<-ggtree(tr, ignore.negative.edge=TRUE) %<+% m118 
node<-boot$data %>% filter(isTip==FALSE)
node$Boot<-nodes/1000

boot +
  geom_nodelab(data=node %>% filter(Boot>.50),(aes(x=x, y=y, label=round(Boot,2)))) +
  geom_tiplab(size=2) +
  geom_tiplab(aes(label=Locality,x=.2), size=2, linetype = 0) +
  xlim(0,.35)

ggsave("outputs/1302/bootstrapped-nj-tre.pdf", height=11, width=8.5)
```
## Indian West Coast

hmmm... We can assemble all of these I think now less
SRR3165612
SRR3165613
SRR3165614
SRR3165618
SRR3165629
SRR3165624

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/SRR3165618.sralite.1_pass_1.fastq.gz -2 data/denovo/SRR3165618.sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR3165618 > data/harvest/SRR3165618.out 2> data/harvest/SRR3165618.err &

```{r}
m118 %>% filter(grepl("SRR",Run)) %>% filter(!Run %in% c("SRR3165612","SRR3165613","SRR3165614","SRR3165618","SRR3165629","SRR3165624") ) %>%
  mutate(Command=paste0("srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/denovo/",Run,".sralite.1_pass_1.fastq.gz -2 data/denovo/",Run,".sralite.1_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/",Run," > data/harvest/",Run,".out 2> data/harvest/",Run,".err &")) %>%
  select(Command) %>% write_tsv("1302-srr-assemblies.sh", col_names = FALSE)
```


Finish SL samples

```{sh, eval=FALSE}

conda activate getorganelle

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Jaffna2-R1.fastq.gz -2 data/lakdiva/Jaffna2-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Jaffna2 > data/harvest/Jaffna2.out 2> data/harvest/Jaffna2.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Mannar2-R1.fastq.gz -2 data/lakdiva/Mannar2-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Mannar2 > data/harvest/Mannar2.out 2> data/harvest/Mannar2.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Mannar3-R1.fastq.gz -2 data/lakdiva/Mannar3-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Mannar3 > data/harvest/Mannar3.out 2> data/harvest/Mannar3.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Mannar4-R1.fastq.gz -2 data/lakdiva/Mannar4-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Mannar4 > data/harvest/Mannar4.out 2> data/harvest/Mannar4.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Mannar5-R1.fastq.gz -2 data/lakdiva/Mannar5-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Mannar5 > data/harvest/Mannar5.out 2> data/harvest/Mannar5.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Tranco2-R1.fastq.gz -2 data/lakdiva/Tranco2-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Tranco2 > data/harvest/Tranco2.out 2> data/harvest/Tranco2.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Tranco3-R1.fastq.gz -2 data/lakdiva/Tranco3-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Tranco3 > data/harvest/Tranco3.out 2> data/harvest/Tranco3.err &

```

Taiwan

```{r}
m118 %>% filter(grepl("WJC",Run)) %>% 
  mutate(Command=paste0("srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/taiwan/",Run,"-R1.fastq.gz -2 data/taiwan/",Run,"-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/",Run," > data/harvest/",Run,".out 2> data/harvest/",Run,".err &")) %>%
  select(Command) %>% write_tsv("1302.2-taiwan-assemblies.sh", col_names = FALSE)
```

WA

```{r}
m118 %>% filter(grepl("Broome|Fitz",Run)) %>% 
  mutate(Command=paste0("srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/broome/",Run,"-R1.fastq.gz -2 data/broome/",Run,"-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/",Run," > data/harvest/",Run,".out 2> data/harvest/",Run,".err &")) %>%
  select(Command) %>% write_tsv("1302.3-broome-assemblies.sh", col_names = FALSE)
```

Bangladesh and bungled trinco 

```{sh, eval=FALSE}
srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Trinco2-R1.fastq.gz -2 data/lakdiva/Trinco2-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Trinco2 > data/harvest/Tranco2.out 2> data/harvest/Trinco2.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/lakdiva/Trinco3-R1.fastq.gz -2 data/lakdiva/Trinco3-R2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/Trinco3 > data/harvest/Trinco3.out 2> data/harvest/Trinco3.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/bangladesh/split/SRR26358744_pass_1.fastq.gz -2 data/bangladesh/split/SRR26358744_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR26358744 > data/harvest/SRR26358744.out 2> data/harvest/SRR26358744.err &

srun -p med -t 40:00:00 --mem=32GB --nodes=1 --cpus-per-task=4 get_organelle_from_reads.py -1 data/bangladesh/split/SRR26402930_pass_1.fastq.gz -2 data/bangladesh/split/SRR26402930_pass_2.fastq.gz -R 10 -k 21,45,65,85,105 -F animal_mt -o data/harvest/SRR26402930 > data/harvest/SRR26402930.out 2> data/harvest/SRR26402930.err &
```
