---
title: "1303-phame"
output: html_document
date: "2025-06-30"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

Let's use phame to compare our mitogenomes

Should we include an outgroup? I don't think so.

Here are some we can include.
NC_007439.1 Singapore
KR349919.1 Australia
OP373725.1 Malaysia
KY213962.1 Australia, Darwin
OP359071.1 Malaysia

Need a GFF file

refdir = genomes  # directory where reference (Complete) files are located
  workdir = mitos # directory where contigs/reads files are located and output is stored

reference = 1  # 0:pick a random reference from refdir; 1:use given reference; 2: use ANI based reference
  reffile = NC_007439.1.fa  # reference filename when option 1 is chosen

  project = mito1  # main alignment file name

  cdsSNPS = 1  # 0:no cds SNPS; 1:cds SNPs, divides SNPs into coding and non-coding sequences, gff file is required

  buildSNPdb = 0 # 0: only align to reference 1: build SNP database of all complete genomes from refdir

FirstTime = 1  # 1:yes; 2:update existing SNP alignment, only works when buildSNPdb is used first time to build DB

     data = 0  # *See below 0:only complete(F); 1:only contig(C); 2:only reads(R);
               # 3:combination F+C; 4:combination F+R; 5:combination C+R;
               # 6:combination F+C+R; 7:realignment  *See below
    reads = 2  # 1: single reads; 2: paired reads; 3: both types present;

     tree = 2  # 0:no tree; 1:use FastTree; 2:use RAxML; 3:use both;
bootstrap = 1  # 0:no; 1:yes;  # Run bootstrapping  *See below
        N = 10000  # Number of bootstraps to run *See below

PosSelect = 2  # 0:No; 1:use PAML; 2:use HyPhy; 3:use both # these analysis need gff file to parse genomes to genes

     code = 0  # 0:Bacteria; 1:Virus; 2: Eukarya # Bacteria and Virus sets ploidy to haploid

    clean = 1  # 0:no clean; 1:clean # remove intermediate and temp files after analysis is complete

  threads = 6  # Number of threads to use

   cutoff = 0.1  # Linear alignment (LA) coverage against reference - ignores SNPs from organism that have lower cutoff.

   * When using data option 1,2,5 need a complete reference to align/map to.
   * Use data option 7 when need to extract SNPs using a sublist of already aligned genomes.
   
   
   
srun -t 2-19:00:00 -p med --mem=32GB --nodes=1 --ntasks-per-node=1 --cpus-per-task=6 phame mitos.ctl

