---
title: "1303-wrasse"
output: html_document
date: "2025-06-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warnign=FALSE, message=FALSE)
```

```{r, message=FALSE, warning=FALSE}
library(gplots)
library(pheatmap)
library(ozmaps)
library(ggplotify)
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(ggrepel)
library(RColorBrewer)
```

Have data from Sam in data/wrasse

gl.vcf = unfiltered data from DArT
bg_metadata_mc.csv = metadata for unfiltered data from DArT

gl.qc.vcf = filtered data with basic quality control:
	- read depth 10X to 200X
	- reproducibility > 90% (see dart for description) 
	- locus callrate (missing data) > 90%
	- minor allele count > 1 (singletons removed)
	- individual callrate > 90%
	- individual heterozygosity > 0.25 (outliers removed, plausible cross-contamination)
bg_metadata_mc_qc.csv = metadata for filtered data with basic qc


hmmm... I don't get the position information, asked Sam. Have new files from Sam.

now there are! 

   1 #CHROM
1165 0
1377 SUPER_1
1187 SUPER_10
1200 SUPER_11
1157 SUPER_12
1063 SUPER_13
1162 SUPER_14
 943 SUPER_15
1038 SUPER_16
1084 SUPER_17
 970 SUPER_18
 955 SUPER_19
1341 SUPER_2
 904 SUPER_20
1063 SUPER_21
 803 SUPER_22
 772 SUPER_23
 548 SUPER_24
1254 SUPER_3
1150 SUPER_4
1300 SUPER_5
1316 SUPER_6
1102 SUPER_7
   1 SUPER_7.1_path
1210 SUPER_8
1204 SUPER_9
   1 Scaffold_109
   1 Scaffold_124
   1 Scaffold_141
   1 Scaffold_143
   1 Scaffold_146
   1 Scaffold_153
   1 Scaffold_356
   2 Scaffold_378
  12 Scaffold_49
   9 Scaffold_51
   1 Scaffold_56
   2 Scaffold_57
   2 Scaffold_58
   2 Scaffold_66
   3 Scaffold_72
   

Let's get SUPER* and put in 'chroms.txt'.  
bgzip and tabix vcf.

```{sh, eval=FALSE}
cat data/wrasse/chroms.txt  | while read line; do bcftools view -Ov -r $line data/wrasse/gl.qc.vcf.gz  > outputs/1303/vcfs/$line.vcf; done;

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;
```
Get meta
```{r}
wrasse<-read_csv("data/wrasse/bg_metadata_mc_qc.csv")
wrasse
```

```{r, eval=FALSE}
files<-list.files("outputs/1303/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/1303/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% dplyr::arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/1303/chrom-ld/",chrom,"-ld.pdf"))
}

lapply(files, plotLd)
```

Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/1303/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-wrasse
  chrom<-gsub("outputs/1303/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_c(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pc12

ggsave(paste0("outputs/1303/chrom-pcs/",chrom,"-pcs.pdf"))
}

lapply(files, plotPCA)
```

### None of this is convincing for anything!

Local PCA

Need bcf files

```{sh, eval=FALSE}

cat data/wrasse/chroms.txt  | while read line; do bcftools view -Ob -r $line data/wrasse/gl.qc.vcf.gz > outputs/1303/bcf/$line.bcf; done;

for f in outputs/1303/bcf/*.bcf; do bcftools index $f; done;
```

need a file like so:

```{r}
samples<-wrasse %>% select(id) %>% rename(ID=id)

population<-wrasse %>% select(site_name) %>% rename(population=site_name)

table<-cbind(samples, population)
write.table(table, "outputs/1303/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

Run lostruct in `/Users/mac/github/local_pca/templated`
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/1303/bcf -t snp -s 10 -m 4 -I /Users/mac/github/lates-wgs/outputs/1303/bcf/sample_info.tsv -j 1303`    
`cp lostruct_results/type_snp_size_10_weights_none_jobid_1303/mds_coords.csv ~/github/lates-wgs/outputs/1303 `


`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/1303/bcf -t snp -s 15 -m 4 -I /Users/mac/github/lates-wgs/outputs/1303/bcf/sample_info.tsv -j 1304`
`cp lostruct_results/type_snp_size_15_weights_none_jobid_1304/mds_coords.csv ~/github/lates-wgs/outputs/1303/mdscoords15.csv `

This works.

```{r}
mds<-read_csv("outputs/1303/mds_coords.csv") # 10 snp windows
mds<-read_csv("outputs/1303/mdscoords15.csv") # 15 snp windows

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```

Locations:

```{r}
MDS1 %>% arrange(-Value1)
MDS2 %>% arrange(Value2)

```

```{r}
s14<-read_csv("~/github/local_pca/templated/lostruct_results/type_snp_size_15_weights_none_jobid_1304/SUPER_14.regions.csv") %>% mutate(Window=1:n())
#0 or 1 referenced? I don't know.
s14 %>% filter(Window %in% c(40,41))
```



Let's look at LD in this region:

```{r}
lc<-read.delim("outputs/1303/vcfs/SUPER_14.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("SUPER_14 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlim(15000000,18000000)
  
```
MDS 2

```{r}
s02<-read_csv("~/github/local_pca/templated/lostruct_results/type_snp_size_15_weights_none_jobid_1304/SUPER_19.regions.csv") %>% mutate(Window=1:n())
#0 or 1 referenced? I don't know.
s02 %>% filter(Window %in% c(9,42,43))
```

```{r}
lc<-read.delim("outputs/1303/vcfs/SUPER_19.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("SUPER_19 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlim(15000000,19000000)
  
```



## Pull Interesting regions


_1_ Pull windows in SUPER14 and SUPER19

```{sh, eval=FALSE}
bcftools view -Ov -r SUPER_14:16525923-17514154 data/wrasse/gl.qc.vcf.gz  > outputs/1303/sub/SUPER_14-region.vcf
bcftools view -Ov -r SUPER_19:20571979-21392500 data/wrasse/gl.qc.vcf.gz  > outputs/1303/sub/SUPER_19-region.vcf
```

_2_ PCA of regions


```{r}
meta<-wrasse
vcf<-read.vcfR(file="outputs/1303/sub/SUPER_14-region.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_c(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pc12
```

K, that is weird.

Loadings
```{r, eval=FALSE}
# This code works with other VCF files I have, but not for Sam's. Changing things up in the next code block
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Position<-gsub("NC_.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("NC.*\\.","",Allele)) 
tops$Chrom<-gsub("_1_.*$",".1",tops$Allele)

tops

```

Generating basic meta for this:

```{sh, eval=FALSE}
cat outputs/1303/sub/SUPER_14-region.vcf | grep -v "#" | cut -f 1-3 > outputs/1303/sub/SUPER_14-meta.tsv
cat outputs/1303/sub/SUPER_19-region.vcf | grep -v "#" | cut -f 1-3 > outputs/1303/sub/SUPER_19-meta.tsv

```
```{r}
sites<-read_tsv("outputs/1303/sub/SUPER_14-meta.tsv", col_names = c("Chrom","Position","Locus"))
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Locus<-gsub("\\.0$|\\.1$","",loadings$Allele)
loadings <-left_join(loadings,sites) %>% relocate(Chrom, Position, Locus, Allele)

# this proportion amount can vary based on the amount of variants total you have, may be as low as 0.01
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = -CS1,prop = .15) %>% select(-CS2, -CS3)
tops
```

```{r}
gensub<-gen[loc=tops$Locus]
Xsub <- tab(gensub, NA.method="mean")

pcsub<- dudi.pca(Xsub,scannf=FALSE,scale=FALSE,nf=3)

dfs<-pcsub$li
dfs<-dfs %>% bind_cols(meta)
eig<-pcsub$eig/sum(pcsub$eig)*100


pcss<-ggplot(dfs) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_c(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pcss

```

It is very weird that these have the same position and such
17019428	100164671-8-T/A
17019428	100164674-8-T/A

Looks like this:


SUPER_14	17319228	100191316-56-C/T	C	T	.	.	PR	GT	1/1	0/0	0/1	0/1	0/1	0/1	0/1	./.	1/1	1/1	0/1	1/1	0/1	0/1	0/1	1/1	1/1	0/1	1/1	1/1	0/0	0/0	1/1	0/1	./.	0/0	0/0	1/1	./.	0/0	0/0	1/1	1/1	0/1	1/1	0/0	0/1	1/1	0/1	1/1	1/1	0/0	0/0	1/1	1/1	0/1	1/1	1/1	./.	1/1	0/1	0/0	0/0	1/1	1/1	1/1	0/1	0/0	0/1	1/1	1/1	1/1	0/0	1/1	1/1	0/0	1/1	1/1	1/1	1/1	1/1	1/1	0/1	0/1	1/1	0/0	0/0	0/0	0/1	0/0	0/1	0/0	0/0	1/1	0/0	1/1	0/1	1/1	0/0	0/0	0/1	0/1	1/1	1/1	./.	0/1	0/1	1/1	0/1	0/0	0/1	1/1	0/1	0/1	0/0	0/1	0/1	0/0	0/1	0/0	0/1	0/1	0/1	1/1	1/1	0/1	1/1	0/1	./.	0/1	1/1	0/1	0/1	1/1	0/1	1/1	0/1	0/1	0/1	0/1	1/1	0/0	0/1	0/0	1/1	1/1	0/0	0/1	1/1	0/1	1/1	0/1	0/0	0/1	0/0	0/1	1/1	1/1	0/0	1/1	0/1	0/0	1/1	./.	0/1	1/1	0/1	0/0	0/1	0/1	0/0	1/1	0/0	0/0	1/1	1/1	1/1	1/1	1/1	1/1	0/1	1/1	0/1	0/1	1/1	0/0	0/0	0/1	1/1	1/1	0/0	0/1	0/1	0/1	0/1	0/0	0/1	1/1	1/1	./.	0/1	./.	1/1	0/1	1/1	0/0	0/0	1/1	1/1	0/0	0/1	1/1	0/0	0/0	1/1	0/1	1/1	1/1	0/1	0/1	1/1	1/1	0/0	1/1	1/1	0/0	0/0	0/0	0/0	0/0	0/1	0/1	0/1	1/1	1/1	0/0	1/1	0/0	0/0	0/1	0/1	1/1	0/0	1/1	0/1	1/1	0/0	0/0	1/1	1/1	0/0	0/0	./.	./.	0/1	1/1	0/1	1/1	1/1	0/1	0/0	0/0	1/1	1/1	0/1	1/1	0/0	1/1	0/0	0/0	1/1	0/0	1/1	0/0	0/1	1/1	0/1	0/1	0/1	0/0	0/1	0/1	0/0	1/1	0/0	1/1	1/1	0/1	1/1	0/1	0/0	1/1	0/0	1/1	0/0	0/1	1/1	1/1	0/0	0/1	1/1	1/1	1/1	1/1	1/1	1/1	0/1	1/1	0/0	1/1	1/1	./.	0/0	1/1	0/1	0/1	1/1	0/1	0/1	1/1	0/0	1/1	0/0	1/1	0/0	1/1	0/0	0/1	1/1


Heatmap of some kind

grep -v 100164674-8-T/A SUPER_14-region.vcf > SUPER_14-less.vcf
```{r}
tdf<-vcfR2tidy(read.vcfR(file="outputs/1303/sub/SUPER_14-less.vcf"))
gs<-tdf$gt %>% mutate(Genotype=ifelse(gt_GT=="0/0",0,
                                  ifelse(gt_GT=="0/1",1,
                                  ifelse(gt_GT=="1/1",2,ifelse(gt_GT==NA,NA)))))
gs$Geno<-as.factor(gs$Genotype)
gs$Indiv<-as.factor(gs$Indiv)
gs<-gs %>% filter(POS %in% tops$Position)
```


Cluster, have the one overlapping POS

```{r}
gst<-gs %>% select(POS, Indiv, Genotype) %>% pivot_wider(id_cols = POS, names_from = Indiv, values_from = Genotype)
m<-as.matrix(gst %>% select(-POS))

#pdf("outputs/1303/super14heatmap.pdf", width=22, height=8)
heatmap.2(m,
          ylab="SNPs",
          dendrogram = "both",
          trace="none", key=FALSE,
          labRow = gst$POS,
          xlab="Individuals",
          labCol = "",
          )
#dev.off()
```

Super19
```{r}
meta<-wrasse
vcf<-read.vcfR(file="outputs/1303/sub/SUPER_19-region.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_c(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pc12
```

Uhmm, this is some weird stuff

```{r}
sites<-read_tsv("outputs/1303/sub/SUPER_19-meta.tsv", col_names = c("Chrom","Position","Locus"))
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Locus<-gsub("\\.0$|\\.1$","",loadings$Allele)
loadings <-left_join(loadings,sites) %>% relocate(Chrom, Position, Locus, Allele)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = -CS1,prop = .25) %>% select(-CS2, -CS3)
tops
```


Ok, not so weird

SUPER_19	21045112	100137855-62-C/G	100137855-62-C/G.1	-0.40114801
SUPER_19	21044777	100147234-30-T/G	100147234-30-T/G.1	-0.39680523
SUPER_19	21045005	100143791-48-G/A	100143791-48-G/A.1	-0.37455155

I'll check these ones...

```{r}
gensub<-gen[loc=tops$Locus]

Xsub <- tab(gensub, NA.method="mean")

pcsub<- dudi.pca(Xsub,scannf=FALSE,scale=FALSE,nf=3)

dfs<-pcsub$li
dfs<-dfs %>% bind_cols(meta)
eig<-pcsub$eig/sum(pcsub$eig)*100


pcss<-ggplot(dfs %>% filter(site_name %in% c("Abrolhos Islands north", "Abrolhos Islands south","Qoubba","Coral Bay"))) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=site_name),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pcss

```


Heatmap of some kind
```{r}
tdf<-vcfR2tidy(vcf)
gs<-tdf$gt %>% mutate(Genotype=ifelse(gt_GT=="0/0",0,
                                  ifelse(gt_GT=="0/1",1,
                                  ifelse(gt_GT=="1/1",2,ifelse(gt_GT==NA,NA)))))
gs$Geno<-as.factor(gs$Genotype)
gs$Indiv<-as.factor(gs$Indiv)
gs<-gs %>% filter(POS %in% tops$Position)
```

Cluster

```{r}
gst<-gs %>% select(POS, Indiv, Genotype) %>% filter(POS %in% c("21045112","21044777","21045005")) %>% pivot_wider(id_cols = POS, names_from = Indiv, values_from = Genotype)
m<-as.matrix(gst %>% select(-POS))

#pdf("outputs/1303/super19heatmap.pdf", width=22, height=8)
#@heatmap.2(m,
 #        ylab="SNPs",
  #        dendrogram = "both",
   #       trace="none", key=FALSE,
    #      labRow = gst$POS,
     #     xlab="Individuals",
      #    labCol = "",
       #   )


gs2<-gs %>% select(Indiv) %>% unique() 
gs2$Indiv<-gsub("^\\d+_","",gs2$Indiv)
gs2<-gs2 %>% left_join(wrasse, by=c("Indiv"="id"))
```

```{r}
as.ggplot(pheatmap(m, cluster_cols = TRUE, cluster_rows = FALSE, labels_col = gs2$site_name, fontsize_col=4, labels_row = gst$POS))

ggsave("outputs/1303/super19-redo.pdf", width=22, height = 8)
#dev.off()
```
Check for HWE?

```{r}

gs3<-gs
gs3$Indiv<-gsub("^\\d+_","",gs3$Indiv)
gs3<-gs3 %>% left_join(wrasse, by=c("Indiv"="id")) %>% filter(POS %in% c("21045112","21044777","21045005"))
#remove the one NA in Coral Bay
gs3<-na.omit(gs3)

gs4<-gs3 %>% group_by(site_name, POS) %>% mutate(SampleSize=n()) %>% mutate(Freq=sum(Genotype)/(2*SampleSize))
gs5<-gs4 %>% select(POS, site_name, Freq, SampleSize) %>% unique() %>% arrange(-Freq)
```
```{r}
labels<-gs5 %>% filter(site_name %in% c("Abrolhos Islands north", "Abrolhos Islands south","Qoubba","Coral Bay","Dongara","Jurien Bay"))
ggplot(gs5) + 
  geom_point(aes(x=POS, y=Freq, fill=site_name), pch=21) +
  geom_text_repel(data=labels, aes(x=POS, y=Freq, label=site_name)) +
  theme_bw() +
  theme(legend.position = "none")
```

Plot...

```{r}
oz <- ozmap_data("states")
oz<-oz[oz$NAME=="Western Australia",]
ggplot(oz) +
  geom_sf()
```
```{r}
locs<-wrasse %>% group_by(site_name) %>% summarize(Lat=mean(lat), Long=mean(lon))
d6<-gs5 %>% left_join(locs)
```

```{r}
ggplot(oz) +
  geom_sf() +
  geom_jitter(data=d6, aes(x=Long, y=Lat, fill=Freq, size=SampleSize), pch=21, alpha=0.8) +
  scale_fill_viridis_c() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  facet_wrap(.~POS)

ggsave("outputs/1303/facetted-super19.pdf", width=12, height=6)
```