---
title: "1304-netviewr"
output: html_document
date: "2025-07-07"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)

library(tidyverse)
library(netviewr)
library(viridis)
```

NetviewR
`
```{r, eval=FALSE}
devtools::install_github("esteinig/netviewr")
```

Demo:

```{r}
matrix(rnorm(900), nrow=30) %>% netviewr::netview(k=20) %>% netviewr::plot_netview()

#equivalent
dist_matrix <- matrix(rnorm(900), nrow=30)
g <- netviewr::netview(dist_matrix, k=20)
netviewr::plot_netview(g)
```

Example plot.  

```{r}
node_data <- tibble::tibble(x=letters[1:10], y=1:10)    # generate 10 x 2 node data tibble

g <- igraph::sample_gnm(n=10, m=15) %@%                 # generate random graph with 10 nodes
     node_data %@%                                      # decorate graph with node data tibble
     node_color(data='x', palette='BuGn') %@%           # decorate nodes with colors paletted by x
     node_size(data='y', min=5, max=8)                  # decorate nodes with values rescaled by y
     
g %>% plot_netview()         
```


# Tutorial and stuff   

https://github.com/esteinig/netviewr/blob/master/docs/POPGENOM.md#population-genomics

Before using Netview to obtain a mutual k-nearest-neighbor graph, a distance matrix should be computed that is suitable for your type of data. For example, for eukaryotic SNP panels, you may want to use the 1-IBS matrix that can be computed in PLINK. For others applications, e.g. in bacterial pathogen whole genome sequence data, you may want to compute a simple SNP distance matrix from reference alignments, e.g. with the excellent Rust package psdm. You can also use a phylogenetic tree as the basis for your distance matrix, e.g. by computing pairwise root-to-tip distance between samples, or even using a non-genetic distance measure of similarity.

Graph construction from a distance matrix file (symmetrical without column or row names) using default plotting without data decoration:

convert to plink in outputs/1304/plink

```{sh, eval=FALSE}
plink --vcf SUPER_4.vcf --make-bed --out SUPER_4 --allow-extra-chr
plink --bfile SUPER_4 --distance square ibs --out SUPER_4-plink --allow-extra-chr

plink --vcf SUPER_19.vcf --make-bed --out SUPER_19 --allow-extra-chr
plink --bfile SUPER_19 --distance square ibs --out SUPER_19-plink --allow-extra-chr

plink --vcf gl.qc.vcf.gz --make-bed --out gl --allow-extra-chr
plink --bfile gl --distance square ibs --out gl-plink --allow-extra-chr

# how about something with expected outputs. Aussie Lates WGS thinned50.vcf.gz from 1305, 23,334 variants
plink --vcf thinned50.vcf.gz --make-bed --out thinned50 --allow-extra-chr
plink --bfile thinned50 --distance square ibs --out thinned50-plink --allow-extra-chr


```


## Barra

```{r}
g<-as.matrix(read.csv("outputs/1304/thinned50-plink.mibs", sep="\t", header=F)) %>% netview(k=8) 
lm<-read_csv("meta/aus40.csv")
#lm$Locality<-factor(lm$Locality, levels=c("Broome","Fitzroy River","Daly River","Darwin Harbour","Roper River","Fly River","Hinchinbrook Channel"))
#Net to set legend some other way

cols<-viridis(length(unique(lm$Locality)), option="H")

g %@% lm %@% node_color(data='Locality', palette = cols) %>% plot_netview(legend = "Locality" )

```

I wonder if the technical differences are showing up here, as Darwin Harbour/Fly River are from the same older data.

# Groper

```{r}
#g<-as.matrix(read.csv("outputs/1304/plink/SUPER_4-plink.mibs", sep="\t", header=F)) %>% netview(k=6) 
g<-as.matrix(read.csv("outputs/1304/plink/gl-plink.mibs", sep="\t", header=F)) %>% netview(k=40) 

g %>% plot_netview()

```


Need node data

```{r}
wd<-read_csv("data/wrasse/bg_metadata_mc_qc.csv")
wd$Pop<-as.factor(wd$pop)
wd<-wd %>% mutate(NS=ifelse(lat > -28.36334, "N","S"))
cols<-viridis(length(unique(wd$site_name)), option="viridis")
g %@% wd  %@% node_color(data='NS', palette = cols) %>% plot_netview(legend="NS")
```


# Admixture
outputs/1304/plink

```{sh, eval=FALSE}
plink --vcf gl.qc.vcf.gz --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb

for K in 1 2 3 4 5; \
do ~/github/admixture/admixture --cv myplink2.ped $K | tee log${K}.out; done

grep -h CV log*.out > cross-validation.txt

```
CV error (K=1): 0.39171
CV error (K=2): 0.39625

Plot cross validation scores

No to long at scores