---
title: "1304-netviewr"
output: html_document
date: "2025-07-07"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)

library(tidyverse)
library(netviewr)
library(viridis)
```

NetviewR
`
```{r, eval=FALSE}
devtools::install_github("esteinig/netviewr")
```

Demo:

```{r}
matrix(rnorm(900), nrow=30) %>% netviewr::netview(k=20) %>% netviewr::plot_netview()

#equivalent
dist_matrix <- matrix(rnorm(900), nrow=30)
g <- netviewr::netview(dist_matrix, k=20)
netviewr::plot_netview(g)
```

Example plot.  

```{r}
node_data <- tibble::tibble(x=letters[1:10], y=1:10)    # generate 10 x 2 node data tibble

g <- igraph::sample_gnm(n=10, m=15) %@%                 # generate random graph with 10 nodes
     node_data %@%                                      # decorate graph with node data tibble
     node_color(data='x', palette='BuGn') %@%           # decorate nodes with colors paletted by x
     node_size(data='y', min=5, max=8)                  # decorate nodes with values rescaled by y
     
g %>% plot_netview()         
```


#Genomics
https://github.com/esteinig/netviewr/blob/master/docs/POPGENOM.md#population-genomics

Before using Netview to obtain a mutual k-nearest-neighbor graph, a distance matrix should be computed that is suitable for your type of data. For example, for eukaryotic SNP panels, you may want to use the 1-IBS matrix that can be computed in PLINK. For others applications, e.g. in bacterial pathogen whole genome sequence data, you may want to compute a simple SNP distance matrix from reference alignments, e.g. with the excellent Rust package psdm. You can also use a phylogenetic tree as the basis for your distance matrix, e.g. by computing pairwise root-to-tip distance between samples, or even using a non-genetic distance measure of similarity.

Graph construction from a distance matrix file (symmetrical without column or row names) using default plotting without data decoration:

convert to plink in outputs/1304/plink

````{sh, eval=FALSE}
plink --vcf SUPER_4.vcf --make-bed --out SUPER_4 --allow-extra-chr
plink --bfile SUPER_4 --distance square ibs --out SUPER_4-plink --allow-extra-chr

plink --vcf SUPER_19.vcf --make-bed --out SUPER_19 --allow-extra-chr
plink --bfile SUPER_19 --distance square ibs --out SUPER_19-plink --allow-extra-chr

plink --vcf gl.qc.vcf.gz --make-bed --out gl --allow-extra-chr
plink --bfile gl --distance square ibs --out gl-plink --allow-extra-chr

```



```{r}
g<-as.matrix(read.csv("outputs/1304/plink/SUPER_19-plink.mibs", sep="\t", header=F)) %>% netview(k=8) 
g %>% plot_netview()

```


Need node data

```{r}
wd<-read_csv("data/wrasse/bg_metadata_mc_qc.csv")
cols<-viridis(length(unique(wd$site_name)), option="H")
```





```{r}
g %@% wd  %@% node_color(data='site_name', palette = cols) %>% plot_netview()
```