---
title: "1304-daly-roper"
output: html_document
date: "2025-07-03"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(grid)
library(ggpubr)
```

rsync -avh -e ssh MatthewCampbell1@agrf-data.agrf.org.au:files/AGRF_NXGSQCAGRF25050141-1_22YGFCLT4 ./
D1 - D8 are spread across 8 lines

sending to data/daly

```{sh,eval=FALSE}
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D1*R1.fastq.gz > Daly01-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D1*R2.fastq.gz > Daly01-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D2*R1.fastq.gz > Daly02-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D2*R2.fastq.gz > Daly02-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D3*R1.fastq.gz > Daly03-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D3*R2.fastq.gz > Daly03-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D4*R1.fastq.gz > Daly04-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D4*R2.fastq.gz > Daly04-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D5*R1.fastq.gz > Daly05-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D5*R2.fastq.gz > Daly05-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D6*R1.fastq.gz > Daly06-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D6*R2.fastq.gz > Daly06-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D7*R1.fastq.gz > Daly07-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D7*R2.fastq.gz > Daly07-R2.fastq.gz

cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D8*R1.fastq.gz > Daly08-R1.fastq.gz
cat AGRF_NXGSQCAGRF25050141-1_22YGFCLT4/D8*R2.fastq.gz > Daly08-R2.fastq.gz
```

basic align

bash ../../doAlign-zipped.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

View stats:

Daly01,316492070,271334403,58.7496
Daly02,326046177,283926139,60.2108
Daly03,250931941,216745151,47.0122
Daly04,228200523,200254294,43.2608
Daly05,358110468,308080850,66.6556
Daly06,288183066,246600489,53.1609
Daly07,302850614,258326860,55.7428
Daly08,277003928,240831949,52.1401

Yowsers!!

Set up downsample.

```{r}
m126<-read_csv("meta/m126.csv")
m126
```

_3_ Downsample.  

```{r}

down<-m126 %>% filter(Locality=="Daly River") %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/daly/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
down %>% select(Command) %>% write_tsv(file="1304.1-downsample.sh", col_names = FALSE)
 
```

module load parallel
srun -p high --time 4:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=8 parallel -j 8 < 1304.1-downsample.sh

Let's just look at the first LG in AUS


Look at basic popgen in Australasia

```{r}
m48<-m126 %>% filter(Lineage %in% c("AUS","Mainstream")) %>% mutate(Individual=1:n())

m48 %>% select(NewPath) %>% write_tsv("bamlists/48.bamlist", col_names=FALSE)
m48 %>% select(Run) %>% write_tsv("bamlists/48-sample-names.txt", col_names=FALSE)

meta<-m48
meta$Locality<-factor(meta$Locality, levels=c("Broome","Fitzroy River","Daly River","Darwin Harbour","Fly River","Hinchinbrook Channel","Mainstream"))

m36<-m126 %>% filter(Lineage %in% c("AUS")) %>% mutate(Individual=1:n())
m36 %>% select(NewPath) %>% write_tsv("bamlists/36.bamlist", col_names=FALSE)
m36 %>% select(Run) %>% write_tsv("bamlists/36-sample-names.txt", col_names=FALSE)

meta<-m36
meta$Locality<-factor(meta$Locality, levels=c("Broome","Fitzroy River","Daly River","Darwin Harbour","Fly River","Hinchinbrook Channel"))
```

Chrom01 then all chroms

```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/48.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1304/48-ibs-90-Lcal01 > outputs/1304/48-ibs-90-Lcal01.out 2> outputs/1304/48-ibs-90-Lcal01.err &

srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/48.bamlist -rf meta/lates-lgs.txt \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1304/48-ibs-90 > outputs/1304/48-ibs-90.out 2> outputs/1304/48-ibs-90.err &

srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/36.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 32 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1304/36-ibs-90-Lcal01 > outputs/1304/36-ibs-90-Lcal01.out 2> outputs/1304/36-ibs-90-Lcal01.err &

```

	-> Number of sites retained after filtering: 75260 (chrom01) with 40 AUS samples, with 48 78166
	-> Number of sites retained after filtering: 1953593  with 40 AUS samples, with 48


```{r}
m <- as.matrix(read.table("outputs/1304/48-ibs-90-Lcal01.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

scree<-ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8)) +
  ggtitle("Scree Plot") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

scree
ggsave("outputs/1304/Australasian-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

text12L<-covs %>% select(Run, Locality, V1, V2) %>%
  group_by(Locality) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  geom_text_repel(data=text12L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ggtitle("PCs 1 and 2") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pc12
ggsave("outputs/1304/australasian-lates-pc12.jpeg")
```

The Maintream3, 7, 5 and 12
```{r}
text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

text13L<-covs %>% select(Run, Locality, V1, V3) %>%
  group_by(Locality) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pcs13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  geom_text_repel(data=text13L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="H") +
  ggtitle("PCs 1 and 3") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pcs13
```
```{r}
blank <- grid.rect(gp=gpar(col="white"))
```

```{r}
ggarrange(ggplotGrob(ggarrange(scree, blank, ncol=1)), ggplotGrob(ggarrange(pc12+theme(legend.position = "none"), pcs13, ncol=2, widths=c(1,1.3))), ncol=2, widths=c(1,4)) 
ggsave("outputs/1304/daly-pcs.pdf", width=16, height=6)
```


# Now with Roper
_1_ Consider dropping Mainstream kin from analyses?
_2_ Then make calls across chroms and try a local pca?
_3_ PCAdapt
_4_ Per chromosome PCAs and LD plots?
-ref and -rf
