---
title: "1306-figures-for-AFSB"
output: html_document
date: "2025-07-23"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(raster)
library(ozmaps)
library(ggrepel)
library(ggtree)
library(tanggle)
library(phangorn)
library(viridis)
library(ggpubr)
library(ape)
library(pcadapt)
library(ggpubr)
```
Plot Range on world map.

```{r}
range<-read_csv("outputs/range-map/1693178727.csv")
range$`Overall Probability`<-as.numeric(range$`Overall Probability`)
range
```

```{r}
ggplot() +
geom_raster(data=range, aes(x=`Center Long`, y=`Center Lat`, fill=`Overall Probability`),alpha=0.75) +
  scale_fill_gradient(low="yellow", high="red")
```

```{r}
range2<-na.omit(range)
domain <- c(min(range2$`Center Long`)-5, max(range2$`Center Long`)+5, 
            min(range2$`Center Lat`)-5, max(range2$`Center Lat`)+5)
```



```{r, eval=FALSE}
nat.earth<-stack("~/github/fish-lake/data/earth/NE2_HR_LC_SR_W_DR/NE2_HR_LC_SR_W_DR.tif")

nat.crop <- crop(nat.earth, y=extent(domain))

rast.table <- data.frame(xyFromCell(nat.crop, 1:ncell(nat.crop)),
                         getValues(nat.crop/255))


rast.table$rgb <- with(rast.table, rgb(NE2_HR_LC_SR_W_DR_1,
                                       NE2_HR_LC_SR_W_DR_2,
                                       NE2_HR_LC_SR_W_DR_3,
                                       1))

save(rast.table, file="data/earth/world-rast.rda")
```

```{r, eval=FALSE}
load("data/earth/world-rast.rda")
ggplot() +
  geom_raster(data = rast.table, mapping = aes(x = x, y = y), fill = rast.table$rgb, interpolate = TRUE)  +
  geom_raster(data=range, aes(x=`Center Long`, y=`Center Lat`, fill=`Overall Probability`),alpha=0.75) +
  scale_fill_gradient(low="yellow", high="red") +
  ylab("Latitude\n") +
  xlab("\nLongitude") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  theme(panel.background = element_rect(fill="black", color="black")) +
  theme(plot.background = element_rect(fill="black", color="black")) +
  theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  theme(axis.text=element_text(color="white", size=12)) +
  theme(axis.title=element_text(color="white", face="bold", size=14))
  
ggsave("outputs/1306/basic-map.jpeg")

```


## Generate georeferenced existing names

```{r}
names<-read_tsv("meta/existing-names.tsv")
names
```
```{r}
ggplot() +
  geom_point(data=names, aes(x=Longitude, y=Latitude)) +
  geom_text_repel(data=names, aes(x=Longitude, y=Latitude, label=paste0(Name, " ", Authority)))
```

```{r, eval=FALSE}
ggplot() +
  geom_raster(data = rast.table, mapping = aes(x = x, y = y), fill = rast.table$rgb, interpolate = TRUE)  +
  geom_point(data=names, aes(x=Longitude, y=Latitude)) +
  geom_text_repel(data=names, aes(x=Longitude, y=Latitude, label=paste0(Name, " ", Authority))) +
  scale_fill_gradient(low="yellow", high="red") +
  ylab("Latitude\n") +
  xlab("\nLongitude") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  theme(panel.background = element_rect(fill="black", color="black")) +
  theme(plot.background = element_rect(fill="black", color="black")) +
  theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  theme(axis.text=element_text(color="white", size=12)) +
  theme(axis.title=element_text(color="white", face="bold", size=14)) 
  
ggsave("outputs/1306/existing-names.pdf")
ggsave("outputs/1306/existing-names.jpeg")

```

## Colletion information for our samples

```{r}
mafsb<-read_csv("meta/m131.csv") %>% filter(!Lineage %in% c("Mainstream","OceanPick")) %>% filter(Run !="Broome01")
locs<-read_tsv("outputs/range-map/points-edited-2 copy.tsv") %>% select(Locality, long, lat)
points<-mafsb %>% left_join(locs) %>% group_by(Lineage,Region,Locality, lat, long) %>% summarise("Sample Size"=n())
points$Lineage<-factor(points$Lineage, levels=c("IND","uwisara","lakdiva","SEA","WAL","AUS"))
mafsb$Lineage<-factor(mafsb$Lineage, levels=c("IND","uwisara","lakdiva","SEA","WAL","AUS"))
```
```{r}
ggplot() +
  geom_point(data=points, aes(x=long, y=lat, size=`Sample Size`,fill=Lineage), pch=21, alpha=0.75) +
  scale_size_continuous(breaks=c(1,5,10)) +
  scale_fill_viridis_d(option="H")
```

```{r, eval=FALSE}
ggplot() +
  geom_raster(data = rast.table, mapping = aes(x = x, y = y), fill = rast.table$rgb, interpolate = TRUE)  +
  geom_point(data=points, aes(x=long, y=lat, size=`Sample Size`,fill=Lineage), pch=21, alpha=0.75) +
  scale_size_continuous(breaks=c(1,5,10)) +
  scale_fill_viridis_d(option="H")+
  ylab("Latitude\n") +
  xlab("\nLongitude") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  theme(panel.background = element_rect(fill="black", color="black")) +
  theme(plot.background = element_rect(fill="black", color="black")) +
  theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  theme(axis.text=element_text(color="white", size=12)) +
  theme(axis.title=element_text(color="white", face="bold", size=14))  +
  theme(legend.position = "none")
  
ggsave("outputs/1306/sample-map.pdf")
ggsave("outputs/1306/sample-map.jpeg")

```

## Call Snps for network etc.
```{r}
mafsb %>% select(NewPath) %>% write_tsv("bamlists/108.bamlist", col_names = FALSE)
mafsb %>% select(Run) %>% write_tsv("bamlists/108.names", col_names = FALSE)
mafsb %>% filter(Lineage %in% c("IND","uwisara","lakdiva")) %>% select(Run) %>% write_tsv("bamlists/108-ind.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -t 48:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 97 -bam bamlists/108.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/no-inv.txt -out outputs/1306/snps-wgs-01-glf  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1306/snps-wgs-01-glf.out 2> outputs/1306/snps-wgs-01-glf.err &

```

Possible commands
	-> Number of sites retained after filtering: 7032672 
(base) maccamp@farm:~/lates-wgs/outputs/1306$ mv plink-01.vcf.gz plink-01.vcf.gz.bak

Downsampling IND individuals
	-> Number of sites retained after filtering: 7074172 


```{sh}
plink --tped snps-wgs-01-glf.tped --tfam snps-wgs-01-glf.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/108.names -o outputs/1306/renamed-01.vcf.gz outputs/1306/plink-01.vcf.gz

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf05-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf01-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf10-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1306/pruned-01-maf10-45-max.vcf



bcftools view -S bamlists/108-ind.names -o outputs/1306/ind.vcf outputs/1306/renamed-01.vcf.gz; bgzip outputs/1306/ind.vcf; tabix outputs/1306.ind.vcf.gz

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf05-45-ind.vcf

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf10-45-ind.vcf

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1306/pruned-01-maf10-45-max-ind.vcf

#Convert to phylip and then to nex
conda activate py2;

~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf01-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-max.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45-ind.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-ind.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-max-ind.vcf;

conda deactivate;
```

```{r}
dat<-read.dna(file="outputs/1306/pruned-01-maf05-45.min4.phy")
write.nexus.data(dat, file="outputs/1306/pruned-01-maf05-45.nex")

dat<-read.dna(file="outputs/1306/pruned-01-maf01-45.min4.phy")
write.nexus.data(dat, file="outputs/1306/pruned-01-maf01-45.nex")

dat<-read.dna(file="outputs/1306/pruned-01-maf10-45.min4.phy")
write.nexus.data(dat, file="outputs/1306/pruned-01-maf10-45.nex")

dat<-read.dna(file="outputs/1306/pruned-01-maf10-45-max.min4.phy")
write.nexus.data(dat, file="outputs/1306/pruned-01-maf10-45-max.nex")


dat<-read.dna(file="outputs/1306/pruned-01-maf05-45-ind.min4.phy")
write.nexus.data(dat, file="outputs/1306/pruned-01-maf05-45-ind.min4.nex")
```
Plotting a network

```{r}
#net<-read.nexus.networx("outputs/1306/pruned-01-maf10-45-network.nex")
#net<-read.nexus.networx("outputs/1306/pruned-01-maf01-45-downin.network.nex")
net<-read.nexus.networx("outputs/1306/pruned-01-maf10-45-max.network")
m<-mafsb
m<-m %>% mutate(Lineage2=ifelse(Locality=="Pazhayar-TamilNadu","calcarifer",as.character(Lineage)))
m$Lineage<-factor(m$Lineage, levels=c("IND","uwisara","lakdiva","SEA","WAL","AUS"))
m$Lineage2<-factor(m$Lineage2, levels=c("calcarifer","uwisara","lakdiva","IND","SEA","WAL","AUS"))

```

```{r}
g<-ggsplitnet(net, color="black")  
g$data<-left_join(g$data, m, by=c("label"="Run")) 

n<-g + 
  geom_tippoint(aes(fill=Lineage2), cex=5, alpha=1.00, pch=21) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
#  theme(legend.position = "none") +
  theme(panel.grid=element_blank())
  #theme(panel.background = element_rect(fill="black", color="black")) +
  #theme(plot.background = element_rect(fill="black", color="black")) +
  #theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  #theme(axis.text=element_text(color="white", size=12)) +
  #theme(axis.title=element_text(color="white", face="bold", size=14))  +
 # theme(legend.position = "none")
  
  
  #color white and black
n
ggsave("outputs/1306/wgs-network.jpeg", width=9, height=8)
ggsave("outputs/1306/wgs-network.pdf", width=9, height=8)

```

Call snps in an inversion zone.

NC_066837.1:5247652-

```{sh, eval=FALSE}
srun -t 12:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 97 -bam bamlists/108.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-r NC_066837.1:5247652- -out outputs/1306/lca05  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1306/lca05.out 2> outputs/1306/lca05.err &
```

Convert to vcf and prune

```{sh}
plink --tped lca05.tped --tfam lca05.tfam  --out lca05-binary --recode --allow-extra-chr --noweb
plink --ped lca05-binary.ped --map lca05-binary.map --recode vcf --allow-extra-chr -out plink-lca05
bgzip plink-lca05.vcf 
tabix plink-lca05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/108.names -o outputs/1306/lca05-renamed.vcf.gz outputs/1306/plink-lca05.vcf.gz

bcftools +fill-tags outputs/1306/lca05-renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/lca05-pruned.vcf

bcftools +fill-tags outputs/1306/lca05-renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/lca05-pruned-01.vcf
```

417 snps
```{sh}
conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i lca05-pruned.vcf
~/github/mccloud-rrt/vcf2phylip.py -i lca05-pruned-01.vcf

conda deactivate;
```



```{r}
path_to_file <- "outputs/1306/pruned-01-maf05-45.vcf"
path_to_file <- "outputs/1306/pruned-01-maf10-45-max.vcf"
filename <- read.pcadapt(path_to_file, type = "vcf")
x <- pcadapt(input = filename, K = 20) 
var<-round(x$singular.values^2*100,2)
var
```
Note to self to run a structure-type analysis of this.

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```

```{r}
m<-mafsb
plot(x, option = "scores", pop = m$Lineage)
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
wgs<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, fill=Lineage), pch=21, alpha=0.95, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("\nPC1 ", round(var[1],2),"%")) +
  ylab(paste0("PC2 ", round(var[2],2),"%\n")) +
  ggtitle("Genome Wide SNPs\n") +
  theme(plot.title = element_text(size=14, hjust=0.5, face="bold"))

wgs

ggsave("outputs/1306/genome-wide-snps.jpeg", width=7, height=6)
```



```{r}
path_to_file <- "outputs/1306/lca05-pruned.vcf"

filename <- read.pcadapt(path_to_file, type = "vcf")
x <- pcadapt(input = filename, K = 20) 
var<-round(x$singular.values^2*100,2)
var
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```

```{r}
m<-mafsb
plot(x, option = "scores", pop = m$Lineage)
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
lca05<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, fill=Lineage), pch=21, alpha=0.95, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("\nPC1 ", round(var[1],2),"%")) +
  ylab(paste0("PC2 ", round(var[2],2),"%\n")) +
  ggtitle("Lca05\n") +
  theme(plot.title = element_text(size=14, hjust=0.5, face="bold"))

lca05

ggsave("outputs/1306/lca05-snps.jpeg", width=7, height=6)

```


## Trees

in outputs/1306/phylo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf01-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf01-45.min4.phy.varsites.phy --seqtype DNA --redo


iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf05-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf05-45.min4.phy.varsites.phy --seqtype DNA --redo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45.min4.phy.varsites.phy --seqtype DNA --redo

in outputs/1306/max
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45-max.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45-max.min4.phy.varsites.phy --seqtype DNA --redo

in outputs/1306/ind
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-01-maf10-45-max-ind.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s  pruned-01-maf10-45-max-ind.min4.phy.varsites.phy --seqtype DNA --redo


```{r}
tree<-read.tree("outputs/1306/max/pruned-01-maf10-45-max.min4.phy.varsites.phy.contree")
t1<-ggtree(tree) + geom_nodelab(aes(label=node))
t1
```

```{r}
#ind<-m %>% filter(Lineage=="IND")
#t2<-root.phylo(tree,outgroup = ind$Run)
#t2<-midpoint.root(t2)
t2<-root.phylo(tree, node=73)
t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 75)

t<-t3

t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```


## Downsample 
SRR3183258	SAMN04482526	SRS1284026	India West Coast	Mumbai	196695055	178632224	34.8059
SRR3183259	SAMN04482527	SRS1284027	India West Coast	Mumbai	143785172	133091074	25.5825

```{r}
down<-mafsb %>% filter(Run %in% c("SRR3183258","SRR3183259")) %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/split/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
down %>% select(Command) %>% write_tsv(file="1306.1-downsample.sh", col_names = FALSE)
 
```

Ran it, has some effects.

Make a PCA now of ind

```{r}
path_to_file <- "outputs/1306/pruned-01-maf05-45-ind.vcf"
path_to_file <- "outputs/1306/pruned-01-maf10-45-max-ind.vcf"

filename <- read.pcadapt(path_to_file, type = "vcf")
x <- pcadapt(input = filename, K = 20) 
var<-round(x$singular.values^2*100,2)
var
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```

```{r}
m<-mafsb %>% filter(Lineage %in% c("IND","uwisara","lakdiva"))
plot(x, option = "scores", pop = m$Lineage)
```


Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
ind12<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, fill=Lineage), pch=21, alpha=0.95, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("\nPC1 ", round(var[1],2),"%")) +
  ylab(paste0("PC2 ", round(var[2],2),"%\n")) +
  ggtitle("Indian Ocean PCA\n") +
  theme(plot.title = element_text(size=14, hjust=0.5, face="bold"))

ind12

ind13<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, fill=Lineage), pch=21, alpha=0.95, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("\nPC1 ", round(var[1],2),"%")) +
  ylab(paste0("PC3 ", round(var[3],2),"%\n")) +
  ggtitle("Indian Ocean PCA\n") +
  theme(plot.title = element_text(size=14, hjust=0.5, face="bold"))

ind13

ggsave("outputs/1306/ind-snps.jpeg", width=7, height=6)

```


## Stats for Matt

```{r}
mafsb %>% filter(Locality %in% c("Broome", "Fitzroy River","river estuaries between Yangon and Sittang")) %>% select(Sample, Locality, Aligned, Filtered, Coverage) %>% 
  mutate("% Dup"=(Aligned-Filtered)/Filtered * 100) %>% write_csv("outputs/1306/stats-for-Matt.csv")

```


# Admixture

(base) ➜  admixture git:(main) ✗ ln -s ../pruned-01-maf10-45-max.vcf ./

plink --vcf pruned-01-maf10-45-max.vcf --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb

no1 in subdir 1306.admixture

```{sh, eval=FALSE}
for K in 1 2 3 4 5; \
do ~/github/admixture/admixture --cv myplink2.ped $K | tee log${K}.out; done
```

```{r}
tbl=read.table("outputs/1204/admixture/myplink2.2.Q")
admix<-mafsb %>% bind_cols(tbl)
```



```{r, echo=FALSE}
PlotAdmixture<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```

```{r, echo=FALSE}
PlotAdmixtureNoLabels<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
    theme(axis.text = element_text(size=10, face="bold")) +
  #theme(axis.text.x = element_text(angle=45, hjust=1)) +
  theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}

```


##Admixture

```{r}
PlotAdmixture("outputs/1306/admixture/myplink2.2.Q",mafsb) 
PlotAdmixture("outputs/1306/admixture/myplink2.3.Q",mafsb) 
PlotAdmixture("outputs/1306/admixture/myplink2.4.Q",mafsb) 
PlotAdmixture("outputs/1306/admixture/myplink2.5.Q",mafsb) 

```
pb<-PlotAdmixtureNoLabels("outputs/1204/admixture/myplink2.3.Q",m103) 
pc<-PlotAdmixtureNoLabels("outputs/1204/admixture/myplink2.4.Q",m103) 
pd<-PlotAdmixtureNoLabels("outputs/1204/admixture/myplink2.5.Q",m103) 
pe<-PlotAdmixture("outputs/1204/admixture/myplink2.6.Q",m103) 



plink --vcf pruned-01-maf10-45-max-ind.vcf --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb


Get these fish 
```{r}
tind<-mafsb %>% filter(Lineage %in% c("IND","lakdiva","uwisara"))

PlotAdmixture("outputs/1306/ind-admixture/myplink2.2.Q",tind) 
PlotAdmixture("outputs/1306/ind-admixture/myplink2.3.Q",tind) 

```