---
title: "1308-pilbara"
output: html_document
date: "2025-11-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(raster)
library(ozmaps)
library(ggrepel)
library(ggtree)
library(tanggle)
library(phangorn)
library(viridis)
library(ggpubr)
library(ape)
library(RColorBrewer)
library(pcadapt)
library(ggpubr)
```

Have the new Pilbara seqs, there are a few lanes to combine

Don't forget to md5sum -c checksums.md5
srun -p high -t 12:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 md5sum -c checksums.md5 > check.out 2> check.err
```{sh, eval=FALSE}
module load parallel
srun -p high -t 12:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < ../../1308.1-combine.sh  
bash ../../do-align.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz
cat *stats > piblara.csv
```

Can downsample, need to downsample fitzroy QLD fish too 

Downsample.  



```{r}
m152<-read_csv("meta/m152.csv")
m129<-m152 %>% filter(Coverage > 3) %>% filter(!Locality %in% c("Mainstream","OceanPick"))

m129 %>% select(NewPath) %>% write_tsv("bamlists/129.bamlist", col_names = FALSE)
m129 %>% select(Run) %>% write_tsv("bamlists/129.names", col_names = FALSE)

m69<-m129 %>% filter(Source=="New")
m69 %>% select(NewPath) %>% write_tsv("bamlists/69.bamlist", col_names = FALSE)
m69 %>% select(Run) %>% write_tsv("bamlists/69.names", col_names = FALSE)

aus<-m129 %>% filter(Lineage=="AUS")

aus %>% select(NewPath) %>% write_tsv("bamlists/61.bamlist", col_names = FALSE)
aus %>% select(Run) %>% write_tsv("bamlists/61.names", col_names = FALSE)
write.csv(aus, file="meta/aus-wgs-61.csv")

meta<-m129

# Create a rooted tree for Australia.  Wallacea + Taiwan?

t<- m129 %>% filter(Locality =="Tamsui River")
w<- m129 %>% filter(Locality %in% c("Samarinda","Sulawesi"))

m76<-bind_rows(aus,w,t) %>%  filter(Run !="WAF011639") %>% filter(Run != "Broome01") %>% write_csv(file="meta/m76.csv")
m76 %>% select(NewPath) %>% write_tsv("bamlists/76.bamlist", col_names = FALSE)
m76 %>% select(Run) %>% write_tsv("bamlists/76.names", col_names = FALSE)


m65<-bind_rows(aus,w) %>%  filter(Run !="WAF011639") %>% filter(Run != "Broome01") %>% write_csv(file="meta/m65.csv")
m65 %>% select(NewPath) %>% write_tsv("bamlists/65.bamlist", col_names = FALSE)
m65 %>% select(Run) %>% write_tsv("bamlists/65.names", col_names = FALSE)

```


```{r}
down<-m152 %>% filter(Locality %in% c("Fitzroy River QLD","DeGrey River","Tully Inlet")) %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", Path, " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
down %>% select(Command) %>% write_tsv(file="1308.2-downsample.sh", col_names = FALSE)
```

```{sh, eval=FALSE}
module load parallel
srun -p high -t 4:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < 1308.2-downsample.sh
```



One chrom GLs for sanity

```{sh, eval=FALSE}
#All Data
srun -p high -t 5:00:00 --mem=32G --nodes=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/129.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 116 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1308/ibs-90-Lcal01 > outputs/1308/ibs-90-Lcal01.out 2> outputs/1308/ibs-90-Lcal01.err &

#Only Newly Determined data
srun -p high -t 5:00:00 --mem=32G --nodes=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/69.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 62 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1308/69-ibs-90-Lcal01 > outputs/1308/69-ibs-90-Lcal01.out 2> outputs/1308/69-ibs-90-Lcal01.err &

#Only Australasian Fish
srun -p high -t 5:00:00 --mem=32G --nodes=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/61.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 55 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1308/aus-ibs-90-Lcal01 > outputs/1308/aus-ibs-90-Lcal01.out 2> outputs/1308/aus-ibs-90-Lcal01.err &

```



```{r}
m <- as.matrix(read.table("outputs/1308/ibs-90-Lcal01.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

scree<-ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8)) +
  ggtitle("Scree Plot") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

scree
ggsave("outputs/1308/chrom01-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:4] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

text12L<-covs %>% select(Run, Locality, V1, V2) %>%
  group_by(Locality) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
 # geom_text_repel(data=text12L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="C") +
  ggtitle("PCs 1 and 2") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc12
ggsave("outputs/1308/chrom01-lates-pc12.jpeg", width=6, height=6)
ggsave("outputs/1308/chrom01-lates-pc12.pdf", width=6, height=6)

```

```{r}
text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

text13L<-covs %>% select(Run, Locality, V1, V3) %>%
  group_by(Locality) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
 # geom_text_repel(data=text13L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="C") +
  ggtitle("PCs 1 and 3") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc13
```


# Call SNPs


```{sh, eval=FALSE}
srun -t 48:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 116 -bam bamlists/129.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/no-inv.txt -out outputs/1308/snps-wgs-01-glf  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1308/snps-wgs-01-glf.out 2> outputs/1308/snps-wgs-01-glf.err &
```


6936789 variants loaded from

```{sh}
plink --tped snps-wgs-01-glf.tped --tfam snps-wgs-01-glf.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/129.names -o outputs/1308/renamed-01.vcf.gz outputs/1308/plink-01.vcf.gz

bcftools +fill-tags outputs/1308/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1308/pruned-01-maf05-45.vcf

bcftools +fill-tags outputs/1308/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1308/pruned-01-maf01-45.vcf

bcftools +fill-tags outputs/1308/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1308/pruned-01-maf10-45.vcf

bcftools +fill-tags outputs/1308/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1308/pruned-01-maf10-45-max.vcf



#Convert to phylip and then to nex
conda activate py2;

~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf01-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-max.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45.vcf;

conda deactivate;
```


# Call SNPS in Australia

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 55 -bam bamlists/61.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf meta/lates-lgs.txt -out outputs/1308/snps-wgs-05-glf  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1308/snps-wgs-05-glf.out 2> outputs/1308/snps-wgs-05-glf.err &

```
1153426 variants and 61 people pass filters and QC.
976163 in filtered vcf

```{sh, eval=FALSE}
plink --tped snps-wgs-05-glf.tped --tfam snps-wgs-05-glf.tfam  --out plink-binary-05 --recode --allow-extra-chr --noweb
plink --ped plink-binary-05.ped --map plink-binary-05.map --recode vcf --allow-extra-chr -out plink-05
bgzip plink-05.vcf 
tabix plink-05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/61.names -o outputs/1308/renamed-05.vcf.gz outputs/1308/plink-05.vcf.gz
bcftools +fill-tags outputs/1308/renamed-05.vcf.gz | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' > outputs/1308/filtered-05.vcf

bcftools +fill-tags outputs/1308/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1308/pruned-05.vcf

bcftools +fill-tags outputs/1308/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1308/pruned-05-rand.vcf


bcftools +fill-tags outputs/1308/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1308/pruned-10.vcf

bgzip outputs/1308/pruned-05.vcf
tabix outputs/1308/pruned-05.vcf.gz

## Set up for chrom-by-chrom PCAS
cat filtered-05.vcf |  grep "#" > header.txt 
cat filtered-05.vcf | grep -v "#" | awk 'BEGIN{i=0}{i++;if (i%100==0) print}' > t100.vcf
cat filtered-05.vcf | grep -v "#" | awk 'BEGIN{i=0}{i++;if (i%50==0) print}' > t50.vcf
cat filtered-05.vcf | grep -v "#" | awk 'BEGIN{i=0}{i++;if (i%20==0) print}' > t20.vcf

cat header.txt t100.vcf > thinned100.vcf; bgzip thinned100.vcf; tabix thinned100.vcf.gz;
cat header.txt t50.vcf > thinned50.vcf; bgzip thinned50.vcf; tabix thinned50.vcf.gz;
cat header.txt t20.vcf > thinned20.vcf; bgzip thinned20.vcf; tabix thinned20.vcf.gz;

```




## Copy from here

```{r}
path_to_file <- "outputs/1308/pruned-05.vcf.gz"

filename <- read.pcadapt(path_to_file, type = "vcf")
```



_1_ choose a K   

```{r}
x <- pcadapt(input = filename, K = 20) 
```

```{r}
var<-round(x$singular.values^2*100,2)
var
```

```{r}
#plot(x, option = "screeplot")
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela
```

```{r}
m<-aus
```

```{r}
plot(x, option = "scores", pop = factor(m$Region, levels=c("Western Australia","Northern Territory","Papua New Guinea","Queensland")))
 
```




Get scree plot and PCA    
```{r}
pcadata<-x$scores[,1:5] %>% as_tibble()
pcadata<-pcadata %>% bind_cols(m)

```

```{r}
ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Region), alpha=0.95) +
  #geom_text_repel(data=pcadata %>% filter(Region %in% c("Cambodia")), aes(x=V1, y=V2,label=Run)) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="C") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") 

ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, shape=Lineage, fill=Region), alpha=0.95) +
 # geom_text_repel(data=pcadata %>% filter(Region %in% c("Queensland")), aes(x=V1, y=V4,label=Run)) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="C") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 3") 
```


```{r}
labels12<-pcadata %>% select(Locality,V1, V2) %>% group_by(Locality) %>% summarize(x=mean(V1), y=mean(V2)) 


pcb<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Locality), alpha=0.95) +
  geom_label_repel(data=labels12, aes(x=x, y=y, label=Locality, fill=Locality), alpha=0.8, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 2") +
  ggtitle("B") +
  theme(plot.title = element_text(size=14, face="bold")) 

labels13<-pcadata %>% select(Locality,V1, V3) %>% group_by(Locality) %>% summarize(x=mean(V1), y=mean(V3)) 

pcc<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V3, shape=Lineage, fill=Locality), alpha=0.9) +
  geom_label_repel(data=labels13, aes(x=x, y=y, label=Locality, fill=Locality), alpha=0.8, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +

  xlab("PC 1") +
  ylab("PC 3") +
  ggtitle("C") +
  theme(plot.title = element_text(size=14, face="bold")) 

labels14<-pcadata %>% select(Locality,V1, V4) %>% group_by(Locality) %>% summarize(x=mean(V1), y=mean(V4)) 

pcd<-ggplot(pcadata) +
  geom_point(aes(x=V1, y=V4, shape=Lineage, fill=Locality), alpha=0.9) +
  geom_label_repel(data=labels14, aes(x=x, y=y, label=Locality, fill=Locality), alpha=0.8, size=3) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("PC 1") +
  ylab("PC 4") +
  ggtitle("C") +
  theme(plot.title = element_text(size=14, face="bold")) 
```

```{r}
blank <- grid.rect(gp=gpar(col="white"))
```



```{r}
a<-ggplotGrob(ggarrange(panela, pcb, blank, ncol=3, widths=c(1,1.2,.5)))
bc<-ggplotGrob(ggarrange(pcc, pcd, ncol=2, widths=c(1,1.4)))
ggarrange(a,bc, ncol=1, heights=c(1,1))

ggsave("outputs/1308/called-snp-pcas.pdf", width=12, height=10)
```


```{r}
Region<-pcadata %>% select(Region,V1, V2) %>% group_by(Region) %>% summarize(x=mean(V1), y=mean(V2)) 
Locality<-pcadata %>% select(Locality,V1, V2) %>% group_by(Locality) %>% summarize(x=mean(V1), y=mean(V2)) 
pvar <- x$singular.values^2

ggplot(pcadata) +
  geom_point(aes(x=V1, y=V2, shape=Lineage, fill=Locality), alpha=0.8, size=4) +
  geom_text_repel(data=Region, aes(x=x, y=y, label=Region), face="bold", size=8) +
  geom_text_repel(data=Locality, aes(x=x, y=y, label=Locality)) +
  scale_shape_manual(values=c(21,23,23,24,22)) + 
  scale_fill_viridis_d(option="C") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(legend.position = "none") +
  theme(panel.grid = element_blank()) +
  xlab(paste0("PC1 ",round(pvar[1]*100,2), "%" )) +
  ylab(paste0("PC2 ",round(pvar[2]*100,2), "%" )) +
  ggtitle("Genome-Wide PCA of Australasian Fish") +
  theme(plot.title = element_text(size=14, face="bold", hjust=0.5)) 

ggsave("outputs/1305/genome-wide-pcs12.pdf", width=8, height=6)
```


```{r}
x <- pcadapt(filename, K = 4)
summary(x)
```

```{r}
plot(x , option = "manhattan")
```

```{r}
plot(x, option = "qqplot")
```

```{r}
hist(x$pvalues, xlab = "p-values", main = NULL, breaks = 50, col = "orange")
```

```{r}
plot(x, option = "stat.distribution")
```


Manual suggests LD thining for snps from WGS data, but not RAD, checking for clustering around certain regions    
```{R}
par(mfrow = c(2, 2))
for (i in 1:4)
  plot(x$loadings[, i], pch = 19, cex = .3, ylab = paste0("Loadings PC", i))
```    
```{r}
plot(x)
```
```{r}
padj <- p.adjust(x$pvalues,method="BH")
alpha <- 0.05
outliers <- which(padj < alpha)
length(outliers)

snp_pc <- get.pc(x, outliers)
snp_pc %>% group_by(PC) %>% summarize(Count=n())

```

```{r}
padj <- p.adjust(x$pvalues,method = "BH")
alpha <- 0.05
outliers <- which(padj < alpha)
length(outliers)

snp_pc <- get.pc(x, outliers)
snp_pc %>% group_by(PC) %>% summarize(Count=n())

```


Get snp numbers: gunzip -c pruned-05.vcf.gz| grep -v "^#" | cut -f1-3 | awk '{print $0"\t"NR}' > pruned-05-snp-numbers.tsv


PC 3 is De Grey river
```{r}
snp_pc %>% filter(PC==3)
```


```{r}
snpnumbers<-read_tsv("outputs/1308/pruned-05-snp-numbers.tsv", col_names = c("Chrom","Pos","Marker","SNP"))
snps<-left_join(as_tibble(snp_pc), snpnumbers) 

snps %>% group_by(PC, Chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
snps %>% filter(PC==3)
```


## LD and Local PCA


```{sh, eval=FALSE}
cat meta/lates-lgs.txt  | while read line; do bcftools view -Ov -r $line outputs/1308/thinned20.vcf.gz  > outputs/1308/vcfs/$line.vcf; done;

bcftools view  outputs/1308/plink-05.vcf.gz  -r NC_066835.1:6806740-7206740 > outputs/1308/vcfs/NC_066835.1-region.vcf
bcftools view  outputs/1308/plink-05.vcf.gz  -r NC_066837.1:8781783-9181783 > outputs/1308/vcfs/NC_066837.1-region.vcf
bcftools view  outputs/1308/plink-05.vcf.gz  -r NC_066842.1:15758873-16158873 > outputs/1308/vcfs/NC_066842.1-region.vcf
bcftools view  outputs/1308/plink-05.vcf.gz  -r NC_066844.1:10777571-11177571 > outputs/1308/vcfs/NC_066844.1-region.vcf
bcftools view  outputs/1308/plink-05.vcf.gz  -r NC_066853.1:28367120-28767120 > outputs/1308/vcfs/NC_066844.1-region.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;
```

```{r, eval=FALSE}
files<-list.files("outputs/1308/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/1308/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% dplyr::arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/1308/chrom-ld/",chrom,"-ld.pdf"))
}

lapply(files, plotLd)
```

Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/1308/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-aus
  chrom<-gsub("outputs/1308/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Locality)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Locality),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pc12

ggsave(paste0("outputs/1308/chrom-pcs/",chrom,"-pcs.pdf"))
}

lapply(files, plotPCA)
```

### None of this is convincing for anything!

Local PCA

Need bcf files

```{sh, eval=FALSE}

cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/1308/thinned20.vcf.gz > outputs/1308/bcf/$line.bcf; done;
#cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/1305/renamed-05.vcf.gz > outputs/1305/bcf/$line.bcf; done;

for f in outputs/1308/bcf/*.bcf; do bcftools index $f; done;
```

need a file like so:

```{r}
samples<-aus %>% select(Run) %>% rename(ID=Run)

population<-aus %>% select(Locality) %>% rename(population=Locality)

table<-cbind(samples, population)
write.table(table, "outputs/1308/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

```{sh, eval=FALSE}
./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/1308/bcf -t snp -s 20 -m 4 -I /Users/mac/github/lates-wgs/outputs/1308/bcf/sample_info.tsv -j 1308   
cp lostruct_results/type_snp_size_20_weights_none_jobid_1308/mds_coords.csv ~/github/lates-wgs/outputs/1308 
```

```{r}
mds<-read_csv("outputs/1308/mds_coords.csv") # 20 snp windows

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


# Phylo

```{sh, eval=FALSE}
conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-05.vcf.gz;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-10.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-05-rand.vcf;

conda deactivate;

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-05.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s pruned-05.min4.phy.varsites.phy --seqtype DNA --redo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-10.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s pruned-10.min4.phy.varsites.phy --seqtype DNA --redo


iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s pruned-05-rand.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s pruned-05-rand.min4.phy.varsites.phy --seqtype DNA --redo
```


```{r}
dat<-read.dna(file="outputs/1308/phylo/pruned-05.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/1308/phylo/pruned-05-maf05.nex")

dat<-read.dna(file="outputs/1308/phylo/pruned-10.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/1308/phylo/pruned-10-maf10.nex")

dat<-read.dna(file="outputs/1308/phylo/pruned-05-rand.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/1308/phylo/pruned-05-rand.nex")
```



```{r}
net<-read.nexus.networx("outputs/1308/phylo/pruned-05-maf05-network")
m<-aus
```

```{r}
g<-ggsplitnet(net, color="black")  
g$data<-left_join(g$data, m, by=c("label"="Run")) 

n<-g + 
  geom_tippoint(aes(fill=Locality), cex=5, alpha=1.00, pch=21) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
#  theme(legend.position = "none") +
  theme(panel.grid=element_blank())
  #theme(panel.background = element_rect(fill="black", color="black")) +
  #theme(plot.background = element_rect(fill="black", color="black")) +
  #theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  #theme(axis.text=element_text(color="white", size=12)) +
  #theme(axis.title=element_text(color="white", face="bold", size=14))  +
 # theme(legend.position = "none")
  
  
  #color white and black
n
ggsave("outputs/1308/aus-wgs-network.jpeg", width=10, height=8)
ggsave("outputs/1308/aus-wgs-network.pdf", width=10, height=8)

```


```{r}
net<-read.nexus.networx("outputs/1308/phylo/pruned-05-rand-network")
m<-aus
```

```{r}
g<-ggsplitnet(net, color="black")  
g$data<-left_join(g$data, m, by=c("label"="Run")) 

n<-g + 
  geom_tippoint(aes(fill=Locality), cex=5, alpha=1.00, pch=21) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
#  theme(legend.position = "none") +
  theme(panel.grid=element_blank())
  #theme(panel.background = element_rect(fill="black", color="black")) +
  #theme(plot.background = element_rect(fill="black", color="black")) +
  #theme(legend.background =element_rect(fill="black", color="black"), legend.text = element_text(color="white"), legend.title = element_text(color="white")) +
  #theme(axis.text=element_text(color="white", size=12)) +
  #theme(axis.title=element_text(color="white", face="bold", size=14))  +
 # theme(legend.position = "none")
  
  
  #color white and black
n

```

WAF011639

```{r}
tree<-read.tree("outputs/1308/phylo/pruned-05-rand.min4.phy.varsites.phy.contree")
t1<-ggtree(tree) + geom_nodelab(aes(label=node)) +geom_tiplab()
t1
```

```{r}
#ind<-m %>% filter(Lineage=="IND")
#t2<-root.phylo(tree,outgroup = ind$Run)
#t2<-midpoint.root(t2)
t2<-root.phylo(tree, node=98)
#t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 25)
t3<-midpoint(tree)
t<-t3
t<-drop.tip(t,"WAF011639")
t<-drop.tip(t,"Broome01")
t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]


f <- t$data
f <- f[!f$isTip,]
f$label <- as.numeric(f$label)
f <- f[f$label < 75,]
f <- f[f$label >= 50,]
```

```{r}
u<-t %<+% m + #bind_rows(m58, rm) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=f,  fill="grey90", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.002,fill=Locality, shape=Lineage), cex=4) + 
  geom_tiplab(aes(label=paste0(Locality, " ",label), x=x+0.01), align = FALSE, size=3) +
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +
  scale_shape_manual(values=c(21,23,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.2) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale(x = 0)

u

ggsave("outputs/1308/aus-phylo.pdf", width=6, height=9)
```


## What about life history variants?

```{r}
gl<-read_tsv("outputs/1308/snps-wgs-05-glf.geno.gz", col_names = c("Chrom","Position",aus$Run)) %>% select(-X64) %>% select(-"WAF011639")
gl2<-gl %>% filter(Chrom=="NC_066856.1") %>% filter(Position %in% c("2922305","2922339","2922605","2996993","3001384"))

m60<-aus %>% filter(Run!="WAF011639")

gl2[gl2==-1]<-NA
m2<-gl2 %>% dplyr::select(-Chrom, -Position) %>% as.matrix()
colnames(m2)<-m60$Locality
rownames(m2)<-gl2$Position
```

```{r}
library(ggplotify)
library(pheatmap)
as.ggplot(pheatmap(m2, cluster_cols = FALSE, cluster_rows = FALSE, color=colorRampPalette(viridis(3, option="cividis", direction=-1, alpha = 0.75))(3)
                   )) +
    ggtitle("Heatmap of associated barra variants on Lca24\n") + 
  theme(plot.title = element_text(hjust=0.5))

ggsave("outputs/1308/associated-variants-australia-wide.jpeg", width=12, height=5)
ggsave("outputs/1308/associated-variants-australia-wide.pdf", width=12, height=5)

```



```{r}
gl<-read_tsv("outputs/1308/snps-wgs-01-glf.geno.gz", col_names = c("Chrom","Position",m129$Run)) %>% select(-X132) %>% select(-"WAF011639")
gl2<-gl %>% filter(Chrom=="NC_066856.1") %>% filter(Position %in% c("2922305","2922339","2922605","2996993","3001384"))
```

```{r}
m128<-m128 %>% filter(Run!="WAF011639")
```

```{r}
gl2[gl2==-1]<-NA
m2<-gl2 %>% dplyr::select(-Chrom, -Position) %>% as.matrix()
colnames(m2)<-m128$Locality
rownames(m2)<-gl2$Position
```

```{r}
library(ggplotify)
library(pheatmap)
as.ggplot(pheatmap(m2, cluster_cols = FALSE, cluster_rows = FALSE, color=colorRampPalette(viridis(3, option="cividis", direction=-1, alpha = 0.75))(3)
                   )) +
    ggtitle("Heatmap of associated barra variants on Lca24\n") + 
  theme(plot.title = element_text(hjust=0.5))

ggsave("outputs/1308/associated-variants-range-wide.jpeg", width=12, height=5)

```