---
title: "1308-pilbara"
output: html_document
date: "2025-11-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(raster)
library(ozmaps)
library(ggrepel)
library(ggtree)
library(tanggle)
library(phangorn)
library(viridis)
library(ggpubr)
library(ape)
library(pcadapt)
library(ggpubr)
```

Have the new Pilbara seqs, there are a few lanes to combine

Don't forget to md5sum -c checksums.md5
srun -p high -t 12:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 md5sum -c checksums.md5 > check.out 2> check.err
```{sh, eval=FALSE}
module load parallel
srun -p high -t 12:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < ../../1308.1-combine.sh  
bash ../../do-align.sh to-align.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz
cat *stats > piblara.csv
```

Can downsample, need to downsample fitzroy QLD fish too 

Downsample.  



```{r}
m152<-read_csv("meta/m152.csv")
m129<-m152 %>% filter(Coverage > 3) %>% filter(!Locality %in% c("Mainstream","OceanPick"))

m129 %>% select(NewPath) %>% write_tsv("bamlists/129.bamlist", col_names = FALSE)
m129 %>% select(Run) %>% write_tsv("bamlists/129.names", col_names = FALSE)

m69<-m129 %>% filter(Source=="New")
m69 %>% select(NewPath) %>% write_tsv("bamlists/69.bamlist", col_names = FALSE)
m69 %>% select(Run) %>% write_tsv("bamlists/69.names", col_names = FALSE)

meta<-m129
```


```{r}
down<-m152 %>% filter(Locality %in% c("Fitzroy River QLD","DeGrey River","Tully Inlet")) %>% mutate(Frac=7/Coverage) %>%
  mutate(Command = paste0("samtools view -bs ", Frac, " ", Path, " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 
down %>% select(Command)
down %>% select(Command) %>% write_tsv(file="1308.2-downsample.sh", col_names = FALSE)
```

```{sh, eval=FALSE}
module load parallel
srun -p high -t 4:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < 1308.2-downsample.sh
```



One chrom GLs for sanity

```{sh, eval=FALSE}
#All Data
srun -p high -t 5:00:00 --mem=32G --nodes=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/129.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 116 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1308/ibs-90-Lcal01 > outputs/1308/ibs-90-Lcal01.out 2> outputs/1308/ibs-90-Lcal01.err &

#Only Newly Determined data
srun -p high -t 5:00:00 --mem=32G --nodes=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/69.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 62 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1308/68-ibs-90-Lcal01 > outputs/1308/69-ibs-90-Lcal01.out 2> outputs/1308/69-ibs-90-Lcal01.err &

```



```{r}
m <- as.matrix(read.table("outputs/1308/ibs-90-Lcal01.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

scree<-ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8)) +
  ggtitle("Scree Plot") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

scree
ggsave("outputs/1308/chrom01-scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:4] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

text12L<-covs %>% select(Run, Locality, V1, V2) %>%
  group_by(Locality) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
#  geom_text_repel(data=text12L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="C") +
  ggtitle("PCs 1 and 2") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc12
ggsave("outputs/1308/chrom01-lates-pc12.jpeg", width=6, height=6)
ggsave("outputs/1308/chrom01-lates-pc12.pdf", width=6, height=6)

```

```{r}
text13<-covs %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
#  geom_text_repel(data=text12L, aes(x=x, y=y, label=Locality), size=3, max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="C") +
  ggtitle("PCs 1 and 3") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc13
```


# Call SNPs


```{sh, eval=FALSE}
srun -t 48:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 116 -bam bamlists/129.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf  meta/no-inv.txt -out outputs/1306/snps-wgs-01-glf  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1306/snps-wgs-01-glf.out 2> outputs/1306/snps-wgs-01-glf.err &



```



```{sh}
plink --tped snps-wgs-01-glf.tped --tfam snps-wgs-01-glf.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/108.names -o outputs/1306/renamed-01.vcf.gz outputs/1306/plink-01.vcf.gz

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf05-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.01' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf01-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf10-45.vcf

bcftools +fill-tags outputs/1306/renamed-01.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1306/pruned-01-maf10-45-max.vcf



bcftools view -S bamlists/108-ind.names -o outputs/1306/ind.vcf outputs/1306/renamed-01.vcf.gz; bgzip outputs/1306/ind.vcf; tabix outputs/1306.ind.vcf.gz

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf05-45-ind.vcf

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/1306/pruned-01-maf10-45-ind.vcf

bcftools +fill-tags outputs/1306/ind.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF  > outputs/1306/pruned-01-maf10-45-max-ind.vcf

#Convert to phylip and then to nex
conda activate py2;

~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf01-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-max.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf05-45-ind.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-ind.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01-maf10-45-max-ind.vcf;

conda deactivate;
```

