---
title: "205-mtDNA"
output: html_document
date: "2023-09-01"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, echo=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(pegas)
library(ape)
library(viridis)
library(ggtree)
library(phytools)
```

```{r}
m58<-read_csv("meta/58.csv") %>% mutate(Lineage=ifelse(Region %in% c("India Eastern Coast", "India Western Coast"), "IND",
                                                 ifelse(Region %in% c("Northern Territory","Queensland" ,"Papua New Guinea"),
                                                                      "AUS+NG","SEA")))
m58$Region<-factor(m58$Region, levels=c("Northern Territory","Queensland","Papua New Guinea",
                                             "Indonesia","Philippines","Cambodia","Thailand", "Vietnam",
                                              "India Eastern Coast","India Western Coast"))

m47<-m58 %>% filter(!Region %in% c("India Eastern Coast", "India Western Coast"))

m60split<-read_csv("meta/60-Indo-split.csv")

m60split$Region<-factor(m60split$Region, levels=c("Northern Territory","Queensland","Papua New Guinea",
                                             "Indonesia-K", "Indonesia-SJ", "Indonesia-SU",
                                             "Philippines","Cambodia","Thailand", "Vietnam","Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

```
Let's get sequences for mtDNA and see what they are.

NC_007439.1  cds 6329..7879
                     /gene="COX1"    
                     
                     
Select alignmets and get mtdna.  


_1_ create bam for each
_2_ convert to consensus fasta
_3_ getbarcode region

```{sh,eval=FALSE}
samtools view -b data/split/SRR3165618.sort.flt.bam NC_007439.1 > data/mtdna/SRR3165618-mtdna.bam
samtools index data/mtdna/SRR3165618-mtdna.bam

srun -p high --nodes=1 --mem=32G -t 00:30:00 /home/maccamp/proovread/bin/bam2cns --bam data/mtdna/SRR3165618-mtdna.bam --ref genome/NC_007439.1.fasta --prefix data/mtdna/SRR3165618-mtdna

 ~/seqtk/seqtk seq -a -q20 -n N data/mtdna/SRR3165618-mtdna.fq > data/mtdna/SRR3165618-mtdna-q20.fasta
samtools faidx data/mtdna/SRR3165618-mtdna-q20.fasta NC_007439.1:6329-7879 > data/mtdna/SRR3165618-coi.fasta
```

Seems to work.  

```{sh,eval=FALSE}
srun -p high --nodes=1 --mem=32G -t 01:30:00 cut -f 1 -d ',' meta/58.csv | while read run; do samtools view -b data/split/$run.sort.flt.bam NC_007439.1 > data/mtdna/$run-mtdna.bam; samtools index data/mtdna/$run-mtdna.bam; /home/maccamp/proovread/bin/bam2cns --bam data/mtdna/$run-mtdna.bam --ref genome/NC_007439.1.fasta --prefix data/mtdna/$run-mtdna;  ~/seqtk/seqtk seq -a -q20 -n N data/mtdna/$run-mtdna.fq > data/mtdna/$run-mtdna-q20.fasta; samtools faidx data/mtdna/$run-mtdna-q20.fasta NC_007439.1:6329-7879 > data/mtdna/$run-coi.fasta; done;

#rename and consolidate.
awk '/^>/{print ">" substr(FILENAME,1,length(FILENAME)-10); next} 1' *-coi.fasta

#mitogenomes
awk '/^>/{print ">" substr(FILENAME,1,length(FILENAME)-10); next} 1' *mtdna-q20.fasta > mitogenomes.fasta
```


Now to merge with barcode data set!!

(base) Macs-MacBook-Pro-2:mtdna mac$ cat ward-vi-renamed.afas new-coi.fasta > combined.fas

(base) Macs-MacBook-Pro-2:mtdna mac$ cat ward-vi-renamed.afas new-coi.fasta > combined.fas

mafft combined.fas > combined.afas`

trimm 
(base) Macs-MacBook-Pro-2:mtdna mac$ java -jar ~/BMGE-1.12/BMGE.jar -i combined.afas -t DNA -o combined.phy 
   DNA sequence alignment combined.afas
   before : 148 sequences / 1576 characters
   after :  148 sequences / 655 characters  
  
 iqtree -s combined.phy -bb 1000 -m GTR+G --redo 

Dropping outgroups
(base) Macs-MacBook-Pro-2:mtdna mac$ cat ward-vi-renamed.fas new-coi.fasta > comb2.fas
(base) Macs-MacBook-Pro-2:mtdna mac$ mafft comb2.fas > comb2.afas
java -jar ~/BMGE-1.12/BMGE.jar -i comb2.afas -t DNA -o combined.phy
143 seqs 655 characters 

 iqtree -s combined.phy -bb 1000 -m GTR+G --redo 
 
 also just using new seqs
 
 Neat!!!
 
 Now to make a network of AUS+NG and SEA inds.   
 
```{r, eval=FALSE}
Naso<-read.dna(file="data/mtdna/new-coi.afasta", format="fasta")
Naso<-updateLabel(Naso, labels(Naso), m58$Region)
labels(Naso)
```
 
```{r, eval=FALSE}
Naso<-Naso[labels(Naso) %in% m47$Region,]
```

```{r, eval=FALSE}
NasoHaps <- haplotype(Naso)
NasoHaps
```

```{r, eval=FALSE}
NasoNet <- haploNet(NasoHaps)
plot(NasoNet, size=attr(NasoNet,"freq"), fast=FALSE)
```


```{r,eval=FALSE}
ind.hap<-with(
  utils::stack(setNames(attr(NasoHaps, "index"), rownames(NasoHaps))),
  table(hap=ind, individuals=rownames(Naso)[values])
 )
ind.hap
```


```{r}
cols<-viridis(n=length(unique(m58$Region)), option="H")
```
```{r, eval=FALSE}
plot(NasoNet, size=attr(NasoNet, "freq"), scale.ratio = .3, cex = 1, pie=ind.hap, labels = FALSE, bg=cols)

#legend("topright", sort(rownames(Naso)) %>% unique(), text.col = cols)
```

```{r, eval=FALSE}
args(pegas:::plot.haploNet)
```



Adding in our reference seqs

(base) Macs-MacBook-Pro-2:coi mac$ mafft new-coi-ref-seqs.fasta > new-coi-ref-seqs.afasta
(base) Macs-MacBook-Pro-2:coi mac$ java -jar ~/BMGE-1.12/BMGE.jar -i new-coi-ref-seqs.afasta -t DNA -op trim.phy -g 0.01
 iqtree -s trim.phy -bb 10000 -m GTR+G --redo 

 iqtree -s new-coi-ref-seqs.afasta -bb 1000 -m HKY+G --redo 
iqtree -s new-coi.afasta -bb 10000 -m HKY+G --redo 

refseq meta

```{r}
Run<-c("EU189379","KJ573921","KJ573895")
Lineage<-c("AUS+NG","SEA","IND")

rm<-cbind(Run,Lineage) %>% as_tibble() 
rm
```

## Tree
```{r}
tree<-read.tree("data/mtdna/new-coi.afasta.contree")
#tree<-read.tree("data/coi/trim.phy.contree")
#tree<-read.tree("data/coi/new-coi-ref-seqs.afasta.contree")
t2<-midpoint.root(tree)
t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 75)

t<-t3

t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```

```{r}
m2<-m58
m2$Region<-m58split$Region
u<-t %<+% m2 + #bind_rows(m58, rm) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=3) + 
  geom_tiplab(aes(label=Region, x=x+0.01), align = FALSE, size=3) +
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +
  scale_shape_manual(values=c(21,23,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.1) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale()

u
ggsave("outputs/205/mtdna-phylo.pdf", width=6.5, height=8)

```

```{r}
t+geom_tiplab(size=2) +
  geom_nodelab() +
  xlim(0,0.1) 

ggsave("outputs/205/tiplabs-phylo.pdf", width=8.5, height=11)

```

```{r}
species<-read_csv("outputs/205/new-coi.afasta.Partition_1.csv", col_names=c("label","Candidate Species"))
species
```

```{r}
t %<+% species +
  geom_tippoint(aes(fill=`Candidate Species`),shape=21)
```


## Mitogenomes

including >NC_007439.1 and refseq for sanity, looks ok. Dropping out.  Need to redo with Bangladesh samples
/Users/mac/github/lates-wgs/data/mitogenomes


```{sh, eval=FALSE}
mafft mitogenomes.fasta > mitogenomes.afas  
bmge -i mitogenomes.afas -t DNA -of tmsa.fasta
#14897
bmge -i mitogenomes.afas -t DNA -of tmsa.fasta -h 0.4 -g 0.01
#14419
bmge -i mitogenomes.afas -t DNA -of tmsa.fasta -h 0.4 -g 0.05
#14419
~/Dropbox/bin/seqConverter.pl -dtmsa.fasta -on
iqtree2 -s tmsa.fasta -st DNA -m MFP -bb 10000 --redo
iqtree2 -s mitogenomes.afas -st DNA -m MFP -bb 10000 --redo

```

```{r}
m3 <- m60split %>% mutate(label=paste0(Run,"-mtdna")) %>% relocate(label)
tree<-read.tree("data/mitogenomes/tmsa.fasta.contree")
tree<-read.tree("data/mitogenomes/mitogenomes.afas.contree")

tree<-midpoint.root(tree)
ggtree(tree) %<+% m3 +
  geom_tiplab(aes(label=Region)) +
  xlim(0,.2) +
  geom_treescale()

ggsave("data/mitogenomes/mito-tree.pdf", width=8.5, height=11)


```

```{r}
ggtree(tree, branch.length = 'none') %<+% m3 +
  geom_tiplab(aes(label=Region)) +
  geom_nodelab() +
  xlim(0,20)
ggsave("data/mitogenomes/mito-cladogram.pdf", width=8.5, height=11)

```

Dropping japonicus
```{sh, eval=FALSE}

mafft no-japonicus.fas > no-japonicus.afas  
bmge -i no-japonicus.afas -t DNA -of no-japonicus-tmsa.fas
#14893
bmge -i no-japonicus.afas -t DNA -of no-japonicus-tmsa.fas -h 0.4
#14422
iqtree2 -s no-japonicus-tmsa.fas -st DNA -m MFP -bb 10000 --redo
iqtree2 -s no-japonicus.afas -st DNA -m MFP -bb 10000 --redo

```

```{r}
tree<-read.tree("data/mitogenomes/no-japonicus-tmsa.fas.contree")
tree<-read.tree("data/mitogenomes/no-japonicus.afas.contree")

tree<-midpoint.root(tree)
ggtree(tree) %<+% m3 +
  geom_tiplab(aes(label=Region)) +
  xlim(0,.2) +
  geom_treescale()

ggsave("data/mitogenomes/no-japonicus-mito-tree.pdf", width=8.5, height=11)


```

```{r}
ggtree(tree, branch.length = 'none') %<+% m3 +
  geom_tiplab(aes(label=Region)) +
  geom_nodelab() +
  xlim(0,20)
ggsave("data/mitogenomes/no-japonicus-mito-cladogram.pdf", width=8.5, height=11)

```