---
title: "400-additional-data"
output: html_document
date: "2023-11-27"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggtree)
library(phytools)
library(ggrepel)
```

Adding in three more samples.
SRS19173540, SRS19136325 from bangladesh in data/bangladesh

 srun -p high -t 12:00:00 --mem=24GB fastq-dump --outdir split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip SRR26402930  
 
  srun -p high -t 12:00:00 --mem=24GB fastq-dump --outdir split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip SRR26358744   
 
Lates japonicus, in data/japonicus
wget https://sra-pub-run-odp.s3.amazonaws.com/sra/DRR391968/DRR391968

(base) maccamp@farm:~/lates-wgs/data/japonicus$ srun -p high -t 36:00:00 --mem=24GB fastq-dump --outdir split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip DRR391968     

An australian fish in data/australia 

SRR1973498
 Purified DNA extracted from an Australian barramundi was sequenced at the Australian Genome Research Facility (Illumina HiSeq 2000 platform).
 srun -p high -t 12:00:00 --mem=24GB fastq-dump --outdir split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip SRR1973498     

## Need to align to reference and compute metrics. 

Also at some point retrieve mtDNA seqs.

can use doAlign-zipped.sh, later  linking fastqs/bams in "data/split" so I can copy path generation.

(base) maccamp@farm:~/lates-wgs/data/bangladesh/split$ cat *stats
SRR26358744,49704585,46328960,10.0176
SRR26402930,50013614,46662535,10.0569

Bangladesh samples 10X coverage, that's great!!

Create new test data set for pop-gen-pca


```{r}
m60<-read_csv("meta/60-Indo-split.csv")
m60 %>% select(Path) %>% write_tsv("bamlists/60.bamlist", col_names = FALSE)
```

Take a look at one chrom
```{sh, eval=FALSE}
srun -p high -t 12:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  \
-bam bamlists/60.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 54 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/400/60-ibs-90 > outputs/400/60-ibs-90.out 2> outputs/400/60-ibs-90.err &


srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/400/60-ibs-90.beagle.gz -K 2 -o outputs/400/60-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/400/60-ibs-90.beagle.gz -K 3 -o outputs/400/60-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/400/60-ibs-90.beagle.gz -K 4 -o outputs/400/60-K4 &

```


	-> Total number of sites analyzed: 24787981    
	-> Number of sites retained after filtering: 277598     


```{r}
meta<-m60
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

m <- as.matrix(read.table("outputs/400/60-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo")
```

```{r}
text13<-covs  %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo")
```


Seems alright!! NGS Admix 

```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  theme(legend.position = "") +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="magma")

}
```

```{r}
PlotAdmix("outputs/400/60-K2.qopt", meta)
```

```{r}
PlotAdmix("outputs/400/60-K3.qopt", meta)
```


```{r}
PlotAdmix("outputs/400/60-K4.qopt", meta)
```

##Generate MTDNA
SRR26402930 

SRR26358744
```{sh, eval=FALSE}
samtools view -b data/split/SRR26402930.sort.flt.bam NC_007439.1 > data/mtdna/SRR26402930-mtdna.bam
samtools index data/mtdna/SRR26402930-mtdna.bam

srun -p high --nodes=1 --mem=32G -t 00:30:00 /home/maccamp/proovread/bin/bam2cns --bam data/mtdna/SRR26402930-mtdna.bam --ref genome/NC_007439.1.fasta --prefix data/mtdna/SRR26402930-mtdna

 ~/seqtk/seqtk seq -a -q20 -n N data/mtdna/SRR26402930-mtdna.fq > data/mtdna/SRR26402930-mtdna-q20.fasta
samtools faidx data/mtdna/SRR26402930-mtdna-q20.fasta NC_007439.1:6329-7879 > data/mtdna/SRR26402930-coi.fasta


samtools view -b data/split/SRR26358744.sort.flt.bam NC_007439.1 > data/mtdna/SRR26358744-mtdna.bam
samtools index data/mtdna/SRR26358744-mtdna.bam

srun -p high --nodes=1 --mem=32G -t 00:30:00 /home/maccamp/proovread/bin/bam2cns --bam data/mtdna/SRR26358744-mtdna.bam --ref genome/NC_007439.1.fasta --prefix data/mtdna/SRR26358744-mtdna

 ~/seqtk/seqtk seq -a -q20 -n N data/mtdna/SRR26358744-mtdna.fq > data/mtdna/SRR26358744-mtdna-q20.fasta
samtools faidx data/mtdna/SRR26358744-mtdna-q20.fasta NC_007439.1:6329-7879 > data/mtdna/SRR26358744-coi.fasta

```
Generate a PCA for sanity, then call SNPs.

mafft to align

iqtree -s new-coi.afasta -bb 10000 -m HKY+G --redo 


```{r}
tree<-read.tree("outputs/400/new-coi.afasta.contree")
t2<-midpoint.root(tree)
t3<-as.polytomy(t2, feature='node.label', fun=function(x) as.numeric(x) < 75)

t<-t3

t<-ggtree(t)

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```


```{r}
m61<-read_csv("meta/61-Indo-split.csv")
m61$Region<-factor(m61$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

m61$Lineage<-factor(m61$Lineage, levels=c("AUS+NG","IND","SEA","japonicus"))
```


```{r}
m2<-m61
u<-t %<+% m2 + #bind_rows(m58, rm) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.005,fill=Region, shape=Lineage), cex=3) +
  geom_tiplab(aes(label=Region, x=x+0.01), align = FALSE, size=3) +
  scale_shape_manual(values=c(21,23,24,25)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.2) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale()

u

ggsave("outputs/400/mtdna-phylo.pdf", width=6.5, height=8)

```
+
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +


```

## L. japonicus test

