---
title: "403-japonicus"
output: html_document
date: "2024-01-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(phytools)
library(tidyverse)
library(ggtree)
library(treeio)
library(ggrepel)
library(tanggle)
library(phangorn)
library(viridis)
library(ggpubr)
library(ape)
library(snpR)
```


Downloaded and aligned japonicus. Using to generate a rooted tree.     
`(base) maccamp@farm:~/lates-wgs/data/japonicus/split$ cat DRR391968.stats`     
DRR391968,972202037,621949413,110.071     
20/110.071     
0.1817009     

(20/110.071)*621949413      
113008769     

srun -p high -t 24:00:00 --mem=32G --nodes=1 samtools view -bs 0.1817009 /home/maccamp/lates-wgs/data/split/DRR391968.sort.flt.bam > /home/maccamp/lates-wgs/data/downsample/DRR391968.reduced.bam


Downsample to 20x coverage.

Check sanity with a basic PCA using .cov approach with all the data and one chrom

/home/maccamp/lates-wgs/data/japonicus/split/DRR391968.sort.flt.bam

```{r}
m61<-read_csv("meta/61-Indo-split.csv")
m61 %>% select(Path) %>% write_tsv("bamlists/61.bamlist", col_names = FALSE)
m61 %>% select(Run) %>% write_tsv("bamlists/61-sample-names.txt", col_names = FALSE)
```


Generate PCA and GL file
-rf genome/lates-lgs.txt 


```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  \
-bam bamlists/61.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 55 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/403/61-ibs-90 > outputs/403/61-ibs-90.out 2> outputs/403/61-ibs-90.err &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/403/61-ibs-90.beagle.gz -K 2 -o outputs/403/61-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/403/61-ibs-90.beagle.gz -K 3 -o outputs/403/61-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/403/61-ibs-90.beagle.gz -K 4 -o outputs/403/61-K4 &

```


	-> Number of sites retained after filtering: 271672 



```{r}
meta<-m61
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast",
                                          "Japan"))

m <- as.matrix(read.table("outputs/403/61-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo")
```

```{r}
text13<-covs  %>% select(Run, Region, V1, V3) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo")
```


## Calling SNPs

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 55 -bam bamlists/61.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/403/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/403/snps-wgs.out 2> outputs/403/snps-wgs.err &

```
	-> Number of sites retained after filtering: 3283323 

Creating a vcf

```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#renaming vcf in 403 locally
bcftools reheader --samples bamlists/61-sample-names.txt -o outputs/403/renamed.vcf.gz outputs/403/plink.vcf.gz

bcftools +prune -l 0.20 -w 10000 outputs/403/renamed.vcf.gz > outputs/403/pruned.vcf
bcftools +prune -l 0.30 -w 10000 outputs/403/renamed.vcf.gz > outputs/403/pruned-03.vcf
```

in 403 locally   

in 403
```{sh, eval=FALSE}
~/github/mccloud-rrt/vcf2phylip.py -i pruned.vcf 
~/github/mccloud-rrt/vcf2phylip.py -i pruned-03.vcf 

```

```{sh, eval=FALSE}
iqtree -s pruned.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo -o DRR391968
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 10000 --redo -o DRR391968
raxmlHPC-PTHREADS -T 3 -m ASC_GTRGAMMA --asc-corr lewis -p 2 -s  pruned.min4.phy.varsites.phy  -n asc -o DRR391968 
```
-p is random number seed  


ERROR: Invalid use of +ASC because of 14065 invariant sites in the alignment

Alignment has 61 sequences with 8529 columns, 8529 distinct patterns
6456 parsimony-informative, 2073 singleton sites, 0 constant sites

Note, without specifying outgroup, clear clades not found. NJ tree shows clear clades though....

iqtree -s pruned-03.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree -s pruned-03.min4.phy.varsites.phy -st DNA -m GTR+ASC -bb 10000 --redo  -o DRR391968
Alignment has 61 sequences with 36765 columns, 36765 distinct patterns
27762 parsimony-informative, 9003 singleton sites, 0 constant sites
 
             
             
## MTDNA

(didnt' work well with japonicus with all the data, trying with 20x downsampled data too, work less well)
options
--coverage=200

samtools view -b data/split/DRR391968.sort.flt.bam NC_007439.1 > data/mtdna/DRR391968-mtdna.bam
samtools index data/mtdna/DRR391968-mtdna.bam

srun -p high --nodes=1 --mem=32G -t 00:30:00 /home/maccamp/proovread/bin/bam2cns --bam data/mtdna/DRR391968-mtdna.bam --ref genome/NC_007439.1.fasta --prefix data/mtdna/DRR391968-mtdna

 ~/seqtk/seqtk seq -a -q20 -n N data/mtdna/DRR391968-mtdna.fq > data/mtdna/DRR391968-mtdna-q20.fasta
samtools faidx data/mtdna/DRR391968-mtdna-q20.fasta NC_007439.1:6329-7879 > data/mtdna/DRR391968-coi.fasta


~/seqtk/seqtk seq -a -q20 -n N data/mtdna/DRR391968-mtdna.fq 


Alternatively

# call variants
bcftools mpileup -d 400 -Ou -f genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna data/mtdna/DRR391968-mtdna.bam | bcftools call -mv -Oz -o outputs/403/calls.vcf.gz
bcftools index calls.vcf.gz

# normalize indels
bcftools norm -f reference.fa calls.vcf.gz -Ob -o calls.norm.bcf

# filter adjacent indels within 5bp
bcftools filter --IndelGap 5 calls.norm.bcf -Ob -o calls.norm.flt-indels.bcf

# apply variants to create consensus sequence
cat reference.fa | bcftools consensus calls.vcf.gz > consensus.fa

# output IUPAC ambiguity codes based on REF+ALT columns (regardless of genotype)
cat reference.fa | bcftools consensus --iupac-codes calls.vcf.gz > consensus.fa

# output IUPAC ambiguity codes based on sample genotypes
cat reference.fa | bcftools consensus --haplotype I calls.vcf.gz > consensus.fa


## SVDQuartets

Running on Farm

conda activate py2
(py2) maccamp@farm:~/bin$ LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/maccamp/miniconda3/envs/py2/lib/

now it works!

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=32G --nodes=1 paup4a168_ubuntu64 lates-svd.nex  

```

Plotting a figure.

```{r}
q<-read.tree("outputs/403/svd-boot-reroot.tree") 
#q<-root(q,"japonicus")
qt<-ggtree(q, branch.length = "none") 
qt + geom_tiplab() + geom_nodelab(aes(label=branch.length)) +
  xlim(0,10)
```

```{r}
tips<-read_tsv("meta/species-tree-tips.txt")

tips$Region<-factor(tips$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

```

```{r}

qt %<+% tips + 
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=4) + 
  geom_tiplab(aes(x=x+.3, label=Region)) +
  geom_nodelab(aes(label=paste0(round(branch.length,0)), y=y+.4, x=x-.3)) +
  scale_shape_manual(values=c(21,23,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,10) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") 

ggsave("outputs/403/species-tree.pdf")
```


```{r}
qt %<+% tips + 
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=4) + 
  geom_tiplab(aes(x=x+.6, label=Region)) +
  geom_nodelab(aes(label=paste0(round(branch.length,0)), y=y+.4, x=x-.3)) +
  scale_shape_manual(values=c(21,23,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,10) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  xlim(0,15)
ggsave("outputs/403/species-tree-narrow.pdf", width=4, height=6)
```