---
title: "601-local-pca"
output: html_document
date: "2024-01-31"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggrepel)
library(pcadapt)
library(adegenet)
library(vcfR)
library(snpR)
library(ggpubr)
library(RColorBrewer)
```


meta

```{r}
meta<-read_csv("meta/wang.csv") %>% mutate(Region=Pop)
write_tsv(meta %>% select(Run), col_names=FALSE, file="bamlists/130-sample-names.txt")
meta$Pop<-factor(meta$Pop, levels=c("AUW","AUE","PNG","INA","MAL","THA"))
meta<-meta %>% mutate(Lineage=ifelse(Pop %in% c("AUW","AUE","PNG"), "AUS+NG", "SEA"))
ozpng<-meta %>% filter(Pop %in% c("AUW","AUE","PNG"))
sea<-meta %>% filter(Lineage=="SEA")
```
Let's pull out the chroms and do a local PCA.

Set up local PCA outputs/600/vcf

```{sh, eval=FALSE}
cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/600/sea-maf05.vcf.gz > outputs/600/vcf/$line.bcf; done;
for f in outputs/600/vcf/*.bcf; do bcftools index $f; done;

cat meta/lates-lgs.txt  | while read line; do bcftools view -Ov -r $line outputs/600/sea-maf05.vcf.gz > outputs/600/sea/$line.vcf; done;

```

```{r}
samples<-sea %>% select(Run) %>% rename(ID=Run)

population<-sea %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/600/vcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

We have 79813 loci or so, 79813/15=5320 windows, 79813/25=3192 windows

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/600/vcf -t snp -s 15 -m 4 -I /Users/mac/github/lates-wgs/outputs/600/vcf/sample_info.tsv -j 1601`  
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/600/vcf -t snp -s 25 -m 4 -I /Users/mac/github/lates-wgs/outputs/600/vcf/sample_info.tsv -j 1602`  
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/600/vcf -t snp -s 50 -m 4 -I /Users/mac/github/lates-wgs/outputs/600/vcf/sample_info.tsv -j 1603`  

With 15 SNP windows
    user   system  elapsed      
1568.540   42.512 1775.611      
With 25 SNP windows    
589.670  17.449 663.288 

```{r}
mds<-read_csv("outputs/600/mds_coords.csv") # 15 snps
mds<-read_csv("outputs/600/mds_coords-25.csv") # 25 snps
#mds<-read_csv("outputs/600/mds_coords-50.csv") # 50 snps

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

```{r}
outliers %>% filter(chrom=="NC_066837.1")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```
      
# NC_066837.1
Let's look at a PCA of NC_066837.1


```{r}
vcf<-read.vcfR(file="outputs/600/sea/NC_066837.1.vcf.gz")
genind<-vcfR2genind(vcf)
genind@pop<-factor(sea$Pop, levels=c("INA","MAL","THA"))

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

```

```{r}
df<-pca1$li
df<-df %>% bind_cols(sea)
eig<-pca1$eig/sum(pca1$eig)*100
df$Population<-factor(df$Pop, levels=c("INA","MAL","THA"))
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Population, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 
pc12
```


# NC_066835.1
Let's look at a pca of NC_066835.1


```{r}
vcf<-read.vcfR(file="outputs/600/sea/NC_066835.1.vcf.gz")
genind<-vcfR2genind(vcf)
genind@pop<-factor(sea$Pop, levels=c("INA","MAL","THA"))

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

```

```{r}
df<-pca1$li
df<-df %>% bind_cols(sea)
eig<-pca1$eig/sum(pca1$eig)*100
df$Population<-factor(df$Pop, levels=c("INA","MAL","THA"))
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Population, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 
pc12
```


# NC_066852.1
Let's look at a pca of NC_066852.1


```{r}
vcf<-read.vcfR(file="outputs/600/sea/NC_066852.1.vcf.gz")
genind<-vcfR2genind(vcf)
genind@pop<-factor(sea$Pop, levels=c("INA","MAL","THA"))

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

```

```{r}
df<-pca1$li
df<-df %>% bind_cols(sea)
eig<-pca1$eig/sum(pca1$eig)*100
df$Population<-factor(df$Pop, levels=c("INA","MAL","THA"))
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Population, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 
pc12
```

