---
title: "606-local-pca-wang"
output: html_document
date: "2024-02-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(ggrepel)
library(pcadapt)
library(adegenet)
library(vcfR)
library(snpR)
library(ggpubr)
library(RColorBrewer)
```


## Meta

```{r}
meta<-read_csv("meta/wang.csv") %>% mutate(Region=Pop)
write_tsv(meta %>% select(Run), col_names=FALSE, file="bamlists/130-sample-names.txt")
meta$Pop<-factor(meta$Pop, levels=c("AUW","AUE","PNG","INA","MAL","THA"))
meta<-meta %>% mutate(Lineage=ifelse(Pop %in% c("AUW","AUE","PNG"), "AUS+NG", "SEA"))
ozpng<-meta %>% filter(Pop %in% c("AUW","AUE","PNG"))
sea<-meta %>% filter(Lineage=="SEA")
```
## Set up

Previously, in 102   
```{r, eval=FALSE}
srun -t 36:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 117 \
-bam bamlists/130.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/102/snps-wang  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/102/snps-wang.out 2> outputs/102/snps-wang.err &
```

Then in 106        
````{sh, eval=FALSE}
plink --tped snps-wang.tped --tfam snps-wang.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz
```    

```{sh, eval=FALSE}
bcftools reheader --samples bamlists/130-sample-names.txt outputs/106/plink.vcf.gz |  bcftools +fill-tags | bcftools view -q 0.05:minor -O z > outputs/606/wang-05.vcf.gz
bcftools index outputs/606/wang-05.vcf.gz    

bcftools +prune -l 0.20 -w 10000 outputs/606/wang-05.vcf.gz -O z > outputs/606/wang-05-pruned.vcf.gz
```

```{sh, eval=FALSE}
cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/606/wang-05.vcf.gz > outputs/606/bcf/$line.bcf; done;
for f in outputs/606/bcf/*.bcf; do bcftools index $f; done;

cat meta/lates-lgs.txt  | while read line; do bcftools view -Oz -r $line outputs/606/wang-05.vcf.gz > outputs/606/vcf/$line.vcf.gz; done;
for f in outputs/606/vcf/*.vcf.gz; do bcftools index $f; done;
```

```{r}
samples<-meta %>% select(Run) %>% rename(ID=Run)

population<-meta %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/606/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```


### Checking SNP signals

```{r}

snps<-import.snpR.data(genotypes = "outputs/606/wang-05-pruned.vcf.gz", 
      sample.meta = meta %>% select(Run, Region, Lineage))
```



```{r}
p <- plot_clusters(snps, facets = c("Region"))
p$plot$pca
```


Running lostruct:      

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/606/bcf -t snp -s 25 -m 4 -I /Users/mac/github/lates-wgs/outputs/606/bcf/sample_info.tsv -j 1606`    

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/606/bcf -t snp -s 50 -m 4 -I /Users/mac/github/lates-wgs/outputs/606/bcf/sample_info.tsv -j 1607`      
Done finding PCs, computing distances.     
   user  system elapsed       
936.946  27.646 997.603       
4000 windows and it didn't work so hot.
240.935   5.264 258.473     

```{r}
#mds<-read_csv("outputs/606/mds_coords-snp25.csv") # 25 SNP windows
mds<-read_csv("outputs/606/mds_coords-snp50.csv") # 25 SNP windows

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```
