---
title: "607-combined-data"
output: html_document
date: "2024-02-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(vcfR)
library(adegenet)
library(ggrepel)
library(ggpubr)
library(phytools)
library(tanggle)
library(phangorn)
library(viridis)
library(grid)
library(snpR)
library(RColorBrewer)
```


Can I generate SNP calls from the combined data sets?      

Generate PCA and GL file

```{r}
bam1<-read_csv("meta/60-Indo-split.csv") %>% select(Run, Region, Lineage, Path) %>% mutate(DataType="WGS") 
bam2<-read_csv("meta/wang.csv") %>%  mutate(Region=Pop) %>% 
  mutate(Lineage=ifelse(Pop %in% c("AUW","AUE","PNG"), "AUS+NG", "SEA")) %>%
  select(Run, Region, Lineage, Path) %>% mutate(DataType="RAD")

bams<-bind_rows(bam1,bam2)

bams %>% group_by(DataType, Lineage) %>% summarize(Count=n())
bams %>% select(Path) %>% write_tsv("bamlists/190.bamlist", col_names = FALSE)
bams %>% select(Run) %>% write_tsv("bamlists/190-sample-names.txt", col_names = FALSE)

meta<-bams

write_csv(meta, "meta/190.csv")
meta$Region<-gsub(" ","", meta$Region)
```

```{sh, eval=FALSE}
srun -p high -t 1-12:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 12  \
-bam bamlists/190.bamlist -rf genome/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 171 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/607/190-ibs-90 > outputs/607/190-ibs-90.out 2> outputs/607/190-ibs-90.err &
```

258278 gls   


```{r}
m <- as.matrix(read.table("outputs/607/190-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

text12<-covs %>% select(Run, Region, DataType, V1, V2) %>%
  group_by(Region, DataType) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region), pch=21, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of L. calcarifer WGS and RADseq Data") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  facet_wrap(.~DataType, ncol = 1)

ggsave("outputs/607/lates-c-genome-wide-pca-12.jpeg")
```

## Calling SNPs


```{sh, eval=FALSE}
srun -t 96:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 171 -bam bamlists/190.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/607/snps  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/607/snps-wgs.out 2> outputs/607/snps-wgs.err &


#Calling snps with doGeno 4
srun -t 96:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 171 -bam bamlists/190.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/607/dogeno4  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.9  > outputs/607/snps-wgs.out 2> outputs/607/snps-wgs.err &

```

104440 sites     
Creating a vcf

```{sh, eval=FALSE}
plink --tped snps.tped --tfam snps.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#renaming vcf
bcftools reheader --samples bamlists/190-sample-names.txt outputs/607/plink.vcf.gz |  bcftools +fill-tags | bcftools view -q 0.05:minor -O z > outputs/607/snps-05.vcf.gz  

bcftools +prune -l 0.20 -w 10000 outputs/607/snps-05.vcf.gz -O z > outputs/607/pruned.vcf.gz

```

### Checking signals


```{r}
snps<-import.snpR.data(genotypes = "outputs/607/pruned.vcf.gz", 
      sample.meta = meta %>% select(Run, Region, Lineage, DataType))
```


```{r}
p <- plot_clusters(snps, facets = c("Region.DataType"), viridis.option = "H")
p$plot$pca
```


Make a version with ggplot:

```{r}
pcadf<-p$plot$pca$data %>% as_tibble()
pcadf$Region<-factor(pcadf$Region, levels=c("AUE","Queensland","AUW","NorthernTerritory","PNG","PapuaNewGuinea","INA","Indonesia-K","Indonesia-SJ","Indonesia-SU",
                                               "MAL","Philippines","Vietnam","Cambodia","THA","Thailand","Bangladesh","IndiaEasternCoast","IndiaWesternCoast"))

```

```{r}
labels<-p$plot$pca$data %>% select(Run, Region, Lineage, DataType, PC1, PC2) %>%
  group_by(Lineage, DataType) %>% summarize(x=mean(PC1), y=mean(PC2))
```
```{r}
ggplot(pcadf) +
  geom_point(aes(x=PC1, y=PC2, shape=Lineage), alpha=0.75, cex=2, fill="grey50") +
  geom_label_repel(data=labels, aes(x=x, y=y, label=Lineage), face="bold", size=6)+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab(p$plot$pca$labels$x) +
  ylab(p$plot$pca$labels$y) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(21,23,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  facet_wrap(.~DataType) +
  theme(legend.position="none")

ggsave("outputs/607/basic-pca.pdf")

```


Genome-wide individual heterozygosity

```{r}
het<-calc_hs(snps)
ho <- get.snpR.stats(het, stats = "hs")
```

```{r}
hets<-ggplot(ho$sample) +
  geom_boxplot(aes(x=Lineage, y=hs), alpha=0.9) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ylab("Individual Heterozygosity\n") +
  xlab("\nLineage") +
  theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) +
  theme(legend.position = "none") +
  facet_wrap(.~DataType)
hets

ggsave("outputs/607/genome-wide-individual-heterozygosity.jpeg")
```

```{r, eval=FALSE}
q<-plot_structure(snps, facet = "Region", method = "snmf", k=c(1,2,3,4,5,6), qsort=FALSE)
save(q, file="outputs/607/q.rda")
```


```{r}
load("outputs/607/q.rda")
ent<-ggplot(q$K_plot, aes(x=k, y=Cross.Entropy)) +
  geom_line() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_discrete(limits=seq(2,6), labels=seq(2,6)) +
  theme(axis.title = element_text(size=12, face="bold")) +
  theme(axis.text=element_text(size=10)) +
  ylab("Cross Entropy\n") +
  xlab("\nK")

ent
```


```{r}
q$plot
```

Local PCA:     

```{sh, eval=FALSE}
cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/607/snps-05.vcf.gz > outputs/607/bcf/$line.bcf; done;
for f in outputs/607/bcf/*.bcf; do bcftools index $f; done;
``` 


```{r}
samples<-meta %>% select(Run) %>% rename(ID=Run)

population<-meta %>% select(Region) %>% rename(population=Region)

table<-cbind(samples, population)
write.table(table, "outputs/607/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/607/bcf -t SNP -s 25 -m 4 -I /Users/mac/github/lates-wgs/outputs/607/bcf/sample_info.tsv -j 1607`

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/607/bcf -t SNP -s 50 -m 4 -I /Users/mac/github/lates-wgs/outputs/607/bcf/sample_info.tsv -j 1607`

```{r}
#mds<-read_csv("outputs/607/mds_coords.csv") # 1e5 bp windows
mds<-read_csv("outputs/607/mds_coords-snp25.csv") # 25 SNP windows
mds<-read_csv("outputs/607/mds_coords-snp50.csv") # 25 SNP windows

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)
#mycolors<-viridis(n=nb.cols, option="plasma")

md1<-ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=1, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


md2<-ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
length(out_ind3)
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
md3<-ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


```{r}
ggarrange(md1,md2,md3, ncol = 1)

ggsave("outputs/607/Supplemental Figure S1.pdf")
```
Chrom lengths
Lca03 23499962
Lca05 28963731
Lca20 23753645

```{sh, eval=FALSE}
bcftools view -r NC_066852.1:6500000-8000000 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066852.1-dummy-sub.vcf.gz     

bcftools view -r NC_066852.1:3500000-23737464 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066852.1-sub.vcf.gz    
bcftools view -r NC_066837.1:5247652-28551740 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066837.1-sub.vcf.gz    
bcftools view -r NC_066835.1:1-17500000 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066835.1-sub.vcf.gz    
bcftools view -r NC_066833.1 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066833.1.vcf.gz    


```


```{r}
vcf<-read.vcfR(file="outputs/607/NC_066852.1-sub.vcf.gz")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$Region)

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
df$Lineage<-factor(df$Lineage, levels=c("AUS+NG","SEA","IND"))
df$Region<-factor(df$Region, levels=c("AUE","Queensland","AUW","NorthernTerritory","PNG","PapuaNewGuinea","INA","Indonesia-K","Indonesia-SJ","Indonesia-SU",
                                               "MAL","Philippines","Vietnam","Cambodia","THA","Thailand","Bangladesh","IndiaEasternCoast","IndiaWesternCoast"))

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
lc20pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Region, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(21,24,23)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  facet_wrap(.~DataType) +
  theme(legend.position = "top")
lc20pc12

ggsave("outputs/607/NC_066852.1.jpeg", width=8)
```

Pulling loadings on axis1    
```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Position<-gsub("NC_.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .05) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("NC.*\\.","",Allele)) 
tops$Chrom<-gsub("_1_.*$",".1",tops$Allele)
```

```{r}
ggplot(loadings, aes(x=Position, y=CS1)) +
  geom_point(pch=21, alpha=0.50, fill="black") +
  geom_point(data=tops, aes(x=Position, y=CS1), fill="skyblue", pch=21, alpha=0.8) +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
```

```{sh, eval=FALSE}
 gunzip -c NC_066852.1-sub.vcf.gz  | grep -v "#" | cut -f 1-5 > NC_066852.1-major-minor.tsv
```

```{r}
sites<-read_tsv("outputs/607/NC_066852.1-major-minor.tsv", col_names = c("Chrom","Position","Site","Major","Minor")) %>% select(-Site)
sites
```


```{r}
tops2<-tops %>% left_join(sites) %>% mutate(Diagnosis=ifelse(MajorMinor=="0",Major,Minor)) %>%
  mutate(Alternate=ifelse(MajorMinor=="1",Major,Minor)) %>%
           mutate(Genotype=paste0(Diagnosis,Diagnosis))
tops2 %>% select(Position, Diagnosis, Alternate) %>% arrange(Position) %>% write_tsv(file="outputs/607/lc20-diagnostic.tsv",col_names = FALSE)
```


## Lca05

```{r}
vcf<-read.vcfR(file="outputs/607/NC_066837.1-sub.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$Region)

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
df$Lineage<-factor(df$Lineage, levels=c("AUS+NG","SEA","IND"))
df$Region<-factor(df$Region, levels=c("AUE","Queensland","AUW","NorthernTerritory","PNG","PapuaNewGuinea","INA","Indonesia-K","Indonesia-SJ","Indonesia-SU",
                                               "MAL","Philippines","Vietnam","Cambodia","THA","Thailand","Bangladesh","IndiaEasternCoast","IndiaWesternCoast"))

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
lc05pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Region, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(21,24,23)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  facet_wrap(.~DataType) +
  theme(legend.position = "top")
lc05pc12

ggsave("outputs/607/NC_066837.1.jpeg", width=8)
```

Pulling loadings on axis1    
```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Position<-gsub("NC_.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .05) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("NC.*\\.","",Allele)) 
tops$Chrom<-gsub("_1_.*$",".1",tops$Allele)
```

```{r}
ggplot(loadings, aes(x=Position, y=CS1)) +
  geom_point(pch=21, alpha=0.50, fill="black") +
  geom_point(data=tops, aes(x=Position, y=CS1), fill="skyblue", pch=21, alpha=0.8) +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
```

```{sh, eval=FALSE}
 gunzip -c NC_066837.1-sub.vcf  | grep -v "#" | cut -f 1-5 > NC_066837.1-major-minor.tsv
```


```{r}
sites<-read_tsv("outputs/607/NC_066837.1-major-minor.tsv", col_names = c("Chrom","Position","Site","Major","Minor")) %>% select(-Site)
```


```{r}
tops2<-tops %>% left_join(sites) %>% mutate(Diagnosis=ifelse(MajorMinor=="0",Major,Minor)) %>%
  mutate(Alternate=ifelse(MajorMinor=="1",Major,Minor)) %>%
           mutate(Genotype=paste0(Diagnosis,Diagnosis))
tops2 %>% select(Position, Diagnosis, Alternate) %>% arrange(Position) %>% write_tsv(file="outputs/607/lc05-diagnostic.tsv",col_names = FALSE)
```

## Lca03


```{r}
vcf<-read.vcfR(file="outputs/607/NC_066835.1-sub.vcf.gz")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$Region)

```

```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
df$Lineage<-factor(df$Lineage, levels=c("AUS+NG","SEA","IND"))
df$Region<-factor(df$Region, levels=c("AUE","Queensland","AUW","NorthernTerritory","PNG","PapuaNewGuinea","INA","Indonesia-K","Indonesia-SJ","Indonesia-SU",
                                               "MAL","Philippines","Vietnam","Cambodia","THA","Thailand","Bangladesh","IndiaEasternCoast","IndiaWesternCoast"))

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
lc03pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Region, shape=Lineage), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(21,24,23)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  facet_wrap(.~DataType) +
  theme(legend.position = "top")
lc03pc12

ggsave("outputs/607/NC_066835.1.jpeg", width=8)
```

Pulling loadings on axis1    
```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Position<-gsub("NC_.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .05) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("NC.*\\.","",Allele)) 
tops$Chrom<-gsub("_1_.*$",".1",tops$Allele)
```

```{r}
ggplot(loadings, aes(x=Position, y=CS1)) +
  geom_point(pch=21, alpha=0.50, fill="black") +
  geom_point(data=tops, aes(x=Position, y=CS1), fill="skyblue", pch=21, alpha=0.8) +
  theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
```

```{sh, eval=FALSE}
 gunzip -c NC_066835.1-sub.vcf.gz  | grep -v "#" | cut -f 1-5 > NC_066835.1-major-minor.tsv
```


```{r}
sites<-read_tsv("outputs/607/NC_066835.1-major-minor.tsv", col_names = c("Chrom","Position","Site","Major","Minor")) %>% select(-Site)
```


```{r}
tops2<-tops %>% left_join(sites) %>% mutate(Diagnosis=ifelse(MajorMinor=="0",Major,Minor)) %>%
  mutate(Alternate=ifelse(MajorMinor=="1",Major,Minor)) %>%
           mutate(Genotype=paste0(Diagnosis,Diagnosis))
tops2 %>% select(Position, Diagnosis, Alternate) %>% arrange(Position) %>% write_tsv(file="outputs/607/lc03-diagnostic.tsv",col_names = FALSE)
```


## Combine plots

```{r}
ggarrange(lc03pc12 + ggtitle(label="A. Lca03") + theme(plot.title = element_text(size=20, face="bold")), 
          lc05pc12 + theme(legend.position = "none") + ggtitle(label="B. Lca05") + theme(plot.title = element_text(size=20, face="bold")),
          lc20pc12 + theme(legend.position = "none") + ggtitle(label="B. Lca20") + theme(plot.title = element_text(size=20, face="bold")),
          ncol = 1, heights=c(1.5,1,1))

ggsave("outputs/607/inversion-pcas.pdf", width=11, height=13)
```


## LD

outputs/607/NC_066852.1-sub.vcf.gz is only inversion zone:

bcftools view -r NC_066833.1 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066833.1-whole-chrom.vcf.gz    
bcftools view -r NC_066835.1 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066835.1-whole-chrom.vcf.gz    
bcftools view -r NC_066837.1 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066837.1-whole-chrom.vcf.gz    
bcftools view -r NC_066852.1 outputs/607/snps-05.vcf.gz -O z > outputs/607/NC_066852.1-whole-chrom.vcf.gz    

plink --vcf NC_066833.1-whole-chrom.vcf.gz --r2 inter-chr --ld-window-r2 0.3 --out NC_066833.1-ldf --allow-extra-chr --double-id
plink --vcf NC_066835.1-whole-chrom.vcf.gz --r2 inter-chr --ld-window-r2 0.3 --out NC_066835.1-ldf --allow-extra-chr --double-id
plink --vcf NC_066837.1-whole-chrom.vcf.gz --r2 inter-chr --ld-window-r2 0.3 --out NC_066837.1-ldf --allow-extra-chr --double-id
plink --vcf NC_066852.1-whole-chrom.vcf.gz --r2 inter-chr --ld-window-r2 0.3 --out NC_066852.1-ldf --allow-extra-chr --double-id


```{r}
ld<-read.delim("outputs/607/NC_066835.1-ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.8)

l03<-  ggplot(ld) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  theme_bw() +
  xlab("\nSNP A\n")+
  ylab("\nSNP B\n")+
  ggtitle("A. Lca03") +
  theme(plot.title=element_text(size=14, face="bold"))

l03
ggsave("outputs/607/lca-03.jpeg")
```

```{r}
ld<-read.delim("outputs/607/NC_066837.1-ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.8)

l05<-  ggplot(ld) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  theme_bw() +
  xlab("\nSNP A\n")+
  ylab("\nSNP B\n")+
  ggtitle("B. Lca05") +
  theme(plot.title=element_text(size=14, face="bold"))

l05
ggsave("outputs/607/lca-05.jpeg")
```

```{r}
ld<-read.delim("outputs/607/NC_066852.1-ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.8)

l20<-  ggplot(ld) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  theme_bw() +
  xlab("\nSNP A\n")+
  ylab("\nSNP B\n")+
  ggtitle("C. Lca20") +
  theme(plot.title=element_text(size=14, face="bold"))

l20
ggsave("outputs/607/lca-20.jpeg")
```

```{r}
ld<-read.delim("outputs/607/NC_066833.1-ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.8)

l01<-  ggplot(ld) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  theme_bw() +
  xlab("\nSNP A\n")+
  ylab("\nSNP B\n")+
  ggtitle("D. Lca01") +
  theme(plot.title=element_text(size=14, face="bold"))

l01
ggsave("outputs/607/lca-01.jpeg")
```

```{r}
ggarrange(l03,l05,l20,l01)
ggsave("outputs/607/Figure-3.jpeg", width=10,height=8)
ggsave("outputs/607/Figure-3.pdf", width=10,height=8)

```