---
title: "608-WGS"
output: html_document
date: "2024-02-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```



```{r}
library(MASS)
library(raster)
library(tidyverse)
library(ggrepel)
library(pcadapt)
library(adegenet)
library(vcfR)
library(snpR)
library(ggpubr)
library(RColorBrewer)
```


What if we use all samples from the WGS data set?


```{sh, eval=FALSE}
bcftools view outputs/401/renamed.vcf.gz | bcftools +fill-tags | bcftools view -q 0.05:minor -O z > outputs/608/wgs-05.vcf.gz

bcftools index outputs/608/wgs-05.vcf.gz

cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/608/wgs-05.vcf.gz > outputs/608/bcf/$line.bcf; done;
for f in outputs/608/bcf/*.bcf; do bcftools index $f; done;

cat meta/lates-lgs.txt  | while read line; do bcftools view -Oz -r $line outputs/608/wgs-05.vcf.gz > outputs/608/vcf/$line.vcf.gz; done;
for f in outputs/608/vcf/*.vcf.gz; do bcftools index $f; done;

#pruned version, with bcftools 1.19-29-gf33fd1d1

bcftools +prune -m 0.20 -w 10000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/wgs-pruned.vcf.gz
cat meta/lates-lgs.txt  | while read line; do bcftools view -Oz -r $line outputs/608/wgs-pruned.vcf.gz > outputs/608/vcf-pruned/$line.vcf.gz; done;
for f in outputs/608/vcf-pruned/*.vcf.gz; do bcftools index $f; done;



#Pulling some regions:
bcftools view -r NC_066852.1:3500000-23737464 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-sub.vcf.gz     
bcftools view -r NC_066852.1:6500000-8000000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-dummy-sub.vcf.gz     


bcftools view -r NC_066852.1:3500000-23737464 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-sub.vcf.gz    
bcftools view -r NC_066837.1:5247652-28551740 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066837.1-sub.vcf.gz    
bcftools view -r NC_066835.1:1-17500000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066835.1-sub.vcf.gz    
bcftools view -r NC_066833.1 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066833.1.vcf.gz    


```
 

Note: 
(base) macs-mbp-2:lates-wgs mac$ gunzip -c outputs/608/wgs-05.vcf.gz | wc -l
 2651716
(base) macs-mbp-2:lates-wgs mac$ gunzip -c outputs/401/renamed.vcf.gz | wc -l
 3327172
 
 
 
```{r}
meta<-read_csv("meta/60-Indo-split.csv") 
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

```


```{r}
samples<-meta %>% select(Run) %>% rename(ID=Run)

population<-meta %>% select(Locality) %>% rename(population=Locality)

table<-cbind(samples, population)
write.table(table, "outputs/608/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```



Using a pruned version 
```{r}
sample.meta<-meta %>% select(Run, Region, Lineage)
sample.meta$Region<-gsub(" ","",meta$Region)
sample.meta$Region<-factor(sample.meta$Region, levels=c("Queensland","NorthernTerritory","PapuaNewGuinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "IndiaEasternCoast","IndiaWesternCoast"))


snps<-import.snpR.data(genotypes = "outputs/608/wgs-pruned.vcf.gz", 
      sample.meta = sample.meta)
```


```{r}
p <- plot_clusters(snps, facets = "Region", viridis.option = "H")
p$plot$pca
```


```{r, eval=FALSE}
q<-plot_structure(snps, facet = "Region", method = "snmf", k=c(1,2,3,4,5,6), qsort=FALSE, 
                  facet.order = c("Queensland","NorthernTerritory","PapuaNewGuinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "IndiaEasternCoast","IndiaWesternCoast"))
save(q, file="outputs/608/q.rda")
```


```{r}
load("outputs/608/q.rda")
ent<-ggplot(q$K_plot, aes(x=k, y=Cross.Entropy)) +
  geom_line() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_discrete(limits=seq(2,6), labels=seq(2,6)) +
  theme(axis.title = element_text(size=12, face="bold")) +
  theme(axis.text=element_text(size=10)) +
  ylab("Cross Entropy\n") +
  xlab("\nK")

ent
```


```{r}
q$plot
```

## Local PCA

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 100 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 500 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 1000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 2000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t bp -s 200000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

1000 snps produces 2,641 windows

```{r}
mds<-read_csv("outputs/608/mds_coords-bp2e5.csv") 

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


## Read in chroms and split

```{r}

wgsmeta<-meta %>% select(Run, Region, Lineage)

lc3rhoms<-read_tsv("meta/lc03-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc3ahoms<-read_tsv("meta/lc03-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc3homs<-bind_rows(lc3ahoms,lc3rhoms)
lc3homs %>% select(Run) %>% write_tsv(file="meta/lc03-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc03-homs.tsv outputs/608/NC_066835.1-sub.vcf.gz  -O z > outputs/608/NC_066835.1-homs.vcf.gz    
```

```{r}
lc3<-import.snpR.data(genotypes = "outputs/608/NC_066835.1-homs.vcf.gz", 
      sample.meta = lc3homs)
```


```{r}
p <- plot_clusters(lc3, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc3 <- calc_pairwise_fst(lc3, "Genotype", method = "WC")
stats <- get.snpR.stats(mylc3, "Genotype", stats ="fst")
```

Local maxima.    
https://stackoverflow.com/questions/43190146/how-to-get-the-top-n-largest-density-spot-coordinates-in-ggmap
```{r}
m3<-stats$pairwise %>% as_tibble() %>% filter(fst > 0.85)

w = matrix(1,3,3)
x = kde2d(m3$position, m3$fst, n=1000)
x2<-x %>% as_tibble()
r = raster(x)
f <- function(X) max(X, na.rm=TRUE)
localmax <- focal(r, w, fun = f, pad=TRUE, padValue=NA)
r2 <- r==localmax
maxXY <- xyFromCell(r2, Which(r2==1, cells=TRUE))
```

```{r}
image(x)
points(maxXY)
```
```{r}
maxXY %>% as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% write_csv("outputs/608/lc03-local-maxima.csv")

maxXY %>%  as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% mutate(Chrom="NC_066835.1", Start=round(x)-1, Stop=round(x)+1, Feature=round(y, 2), Field="0", Strand="+") %>%
  dplyr::select(-x, -y) %>% write_tsv("outputs/608/lc03-local-maxima.bed", col_names = FALSE)

```

```{r}
p3<-ggplot(stats$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  geom_point(data=maxXY %>% as_tibble() %>% filter(y>0.98), aes(x=x, y=y), pch=21, alpha=0.75, fill="grey50")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
  theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("A. Lca03") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 
p3
```




### lc5
```{r}
lc5rhoms<-read_tsv("meta/lc05-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc5ahoms<-read_tsv("meta/lc05-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc5homs<-bind_rows(lc5ahoms,lc5rhoms)
lc5homs %>% dplyr::select(Run) %>% write_tsv(file="meta/lc05-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc05-homs.tsv outputs/608/NC_066837.1-sub.vcf.gz  -O z > outputs/608/NC_066837.1-homs.vcf.gz    
```

```{r}
lc5<-import.snpR.data(genotypes = "outputs/608/NC_066837.1-homs.vcf.gz", 
      sample.meta = lc5homs)
```


```{r}
p <- plot_clusters(lc5, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc5 <- calc_pairwise_fst(lc5, "Genotype", method = "WC")
stats5 <- get.snpR.stats(mylc5, "Genotype", stats ="fst")
```


```{r}
m5<-stats5$pairwise %>% as_tibble() %>% filter(fst > 0.85)

w = matrix(1,3,3)
x = kde2d(m5$position, m5$fst, n=1000)
x2<-x %>% as_tibble()
r = raster(x)
f <- function(X) max(X, na.rm=TRUE)
localmax <- focal(r, w, fun = f, pad=TRUE, padValue=NA)
r2 <- r==localmax
maxXY <- xyFromCell(r2, Which(r2==1, cells=TRUE))
```

```{r}
image(x)
points(maxXY) 
```
```{r}
maxXY %>% as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% write_csv("outputs/608/lc05-local-maxima.csv")

maxXY %>%  as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% mutate(Chrom="NC_066837.1", Start=round(x)-1, Stop=round(x)+1, Feature=round(y, 2), Field="0", Strand="+") %>%
  dplyr::select(-x, -y) %>% write_tsv("outputs/608/lc05-local-maxima.bed", col_names = FALSE)
```


```{r}
p5<-ggplot(stats5$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  geom_point(data=maxXY %>% as_tibble() %>% filter(y>0.98), aes(x=x, y=y), pch=21, alpha=0.75, fill="grey50")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
  theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("B. Lca05") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 
p5
```



### lc20
Using a pruned version 
```{r}
lc20rhoms<-read_tsv("meta/lc20-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc20ahoms<-read_tsv("meta/lc20-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc20homs<-bind_rows(lc20ahoms,lc20rhoms)
lc20homs %>% select(Run) %>% write_tsv(file="meta/lc20-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc20-homs.tsv outputs/608/NC_066852.1-sub.vcf.gz  -O z > outputs/608/NC_066852.1-homs.vcf.gz    
bcftools view -r NC_066852.1 -S meta/lc20-homs.tsv outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-whole-chrom-homs.vcf.gz    

```

```{r}
lc20<-import.snpR.data(genotypes = "outputs/608/NC_066852.1-homs.vcf.gz", 
      sample.meta = lc20homs)

lc20c<-import.snpR.data(genotypes = "outputs/608/NC_066852.1-whole-chrom-homs.vcf.gz", 
      sample.meta = lc20homs)
```


```{r}
p <- plot_clusters(lc20, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc20<- calc_pairwise_fst(lc20, "Genotype", method = "WC")
stats20<- get.snpR.stats(mylc20, "Genotype", stats ="fst")

mylc20c<- calc_pairwise_fst(lc20c, "Genotype", method = "WC")
stats20c<- get.snpR.stats(mylc20c, "Genotype", stats ="fst")
```


```{r}
m20<-stats20$pairwise %>% as_tibble() %>% filter(fst > 0.85)

w = matrix(1,3,3)
x = kde2d(m20$position, m20$fst, n=1000)
x2<-x %>% as_tibble()
r = raster(x)
f <- function(X) max(X, na.rm=TRUE)
localmax <- focal(r, w, fun = f, pad=TRUE, padValue=NA)
r2 <- r==localmax
maxXY <- xyFromCell(r2, Which(r2==1, cells=TRUE))
```

```{r}
image(x)
points(maxXY)
```
```{r}
maxXY %>% as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% write_csv("outputs/608/lc20-local-maxima.csv")

maxXY %>%  as_tibble() %>% filter(y>0.98) %>% arrange(x) %>% mutate(Chrom="NC_066852.1", Start=round(x)-1, Stop=round(x)+1, Feature=round(y, 2), Field="0", Strand="+") %>%
  dplyr::select(-x, -y) %>% write_tsv("outputs/608/lc20-local-maxima.bed", col_names = FALSE)

```



```{r}
p20<-ggplot(stats20$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  geom_point(data=maxXY %>% as_tibble() %>% filter(y>0.98), aes(x=x, y=y), pch=21, alpha=0.75, fill="grey50")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
   theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("C. Lca20") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 

p20
```

```{r}
p20c<-ggplot(stats20c$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
   theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("C. Lca20") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 

p20c
```

```{r}
ggarrange(p3,p5,p20, ncol=1)
ggsave("outputs/608/fst-density.pdf", height=12, width=6.5)
```

(base) ➜  fst-density head lc03-local-maxima.csv 
x,y
4434512.570570571,0.9978983184969785
11228186.132132135,0.9971477179601851
11594939.71771772,0.9816853469022395


Looking around some peaks
(enBank:XP_018556853.1;Name=XP_018556853.1;gbkey=CDS;gene=slc66a2;product=solute carrier family 66 member 2 isoform X2;protein_id=XP_018556853.1

(base) ➜  genome-annotation cat GCF_001640805.2_TLL_Latcal_v3_genomic.gff | grep NC_066835 | awk '{ if ($4 > 4430000  && $4 < 4436000) print }'
NC_066835.1	Gnomon	exon	4435702	4435835	.	-	.	ID=exon-XM_018701336.2-2;Parent=rna-XM_018701336.2;Dbxref=GeneID:108900348,GenBank:XM_018701336.2;gbkey=mRNA;gene=slc66a2;product=solute carrier family 66 member 2%2C transcript variant X1;transcript_id=XM_018701336.2
NC_066835.1	Gnomon	exon	4433642	4433692	.	-	.	ID=exon-XM_018701336.2-3;Parent=rna-XM_018701336.2;Dbxref=GeneID:108900348,GenBank:XM_018701336.2;gbkey=mRNA;gene=slc66a2;product=solute carrier family 66 member 2%2C transcript variant X1;transcript_id=XM_018701336.2
NC_066835.1	Gnomon	CDS	4435702	4435835	.	-	1	ID=cds-XP_018556852.1;Parent=rna-XM_018701336.2;Dbxref=GeneID:108900348,GenBank:XP_018556852.1;Name=XP_018556852.1;gbkey=CDS;gene=slc66a2;product=solute carrier family 66 member 2 isoform X1;protein_id=XP_018556852.1
NC_066835.1	Gnomon	CDS	4433642	4433692	.	-	2	ID=cds-XP_018556852.1;Parent=rna-XM_018701336.2;Dbxref=GeneID:108900348,GenBank:XP_018556852.1;Name=XP_018556852.1;gbkey=CDS;gene=slc66a2;product=solute carrier family 66 member 2 isoform X1;protein_id=XP_018556852.1
NC_066835.1	Gnomon	exon	4435702	4435835	.	-	.	ID=exon-XM_018701337.2-2;Parent=rna-XM_018701337.2;Dbxref=GeneID:108900348,GenBank:XM_018701337.2;gbkey=mRNA;gene=slc66a2;product=solute carrier family 66 member 2%2C transcript variant X2;transcript_id=XM_018701337.2
NC_066835.1	Gnomon	CDS	4435702	4435835	.	-	1	ID=cds-XP_018556853.1;Parent=rna-XM_018701337.2;Dbxref=GeneID:108900348,GenBank:XP_018556853.1;Name=XP_018556853.1;gbkey=CDS;gene=slc66a2;product=solute carrier family 66 member 2 isoform X2;protein_id=XP_018556853.1

slc66a2, hmm, this seems familiar!
Oh, in smelt slc6a3	Solute carrier family 6 member3

Members of the solute carrier family 6 including scl6a3 have been previously associated with adaptation to salinity variation in threespine sticklebacks (Gasterosteus aculeatus) as well as a scl6a3 specifically being identified as a candidate gene from freshwater and marine divergence (Hohenlohe et al. 2010; Guo et al. 2015).


cat GCF_001640805.2_TLL_Latcal_v3_genomic.gff | grep NC_066835 | awk '{ if ($4 > 11210000 && $4 < 11220000) print }'
(base) ➜  genome-annotation cat GCF_001640805.2_TLL_Latcal_v3_genomic.gff | grep NC_066835 | awk '{ if ($4 > 11210000 && $4 < 11220000) print }'

NC_066835.1	Gnomon	exon	11210374	11210794	.	+	.	ID=exon-XR_001963358.1-4;Parent=rna-XR_001963358.1;Dbxref=GeneID:108897849,GenBank:XR_001963358.1;gbkey=ncRNA;gene=LOC108897849;product=uncharacterized LOC108897849;transcript_id=XR_001963358.1

cat GCF_001640805.2_TLL_Latcal_v3_genomic.gff | grep NC_066835 | awk '{ if ($4 > 11580000 && $4 < 11600000) print }'
Near: gene=nup153;product=nucleoporin
in gene=ptdss1b;product=phosphatidylserine
Predicted to enable L-serine-phosphatidylethanolamine phosphatidyltransferase activity. Predicted to be involved in phosphatidylserine biosynthetic process. Predicted to act upstream of or within lipid metabolic process. 
near gene=LOC108898154;product=cytochrome b-c1 complex subunit

making a bed file to find +- closest features

cat lc03-local-maxima.bed lc05-local-maxima.bed lc20-local-maxima.bed > ~/Dropbox/Australian-fishes/barra-inversions/fst-density/local-maxima.bed

(base) ➜  fst-density bedtools closest -a local-maxima.bed -b ../genome-annotation/lates.bed 
NC_066833.1	4433513	4435513	1	0	+	NC_066833.1	4424205	4464338	XM_051066902.1	0	+
NC_066833.1	4433513	4435513	1	0	+	NC_066833.1	4429411	4439246	XM_018704330.2	0	+
NC_066833.1	11227186	11229186	1	0	+	NC_066833.1	11180641	11239114	XM_018704818.2	0	-
NC_066833.1	11593940	11595940	0.98	0	+	NC_066833.1	11364237	11568867	XM_051070155.1	0	+
NC_066835.1	9918036	9920036	1	0	+	NC_066835.1	9879534	9926105	XM_018697574.2	0	-
NC_066835.1	14372162	14374162	1	0	+	NC_066835.1	14372772	14382518	XM_018685013.1	0	-
NC_066835.1	18896248	18898248	1	0	+	NC_066835.1	18892437	18906822	XM_051068507.1	0	-
NC_066835.1	23466973	23468973	1	0	+	NC_066835.1	23456425	23460832	XM_018686663.2	0	+
NC_066852.1	9153386	9155386	1	0	+	NC_066852.1	9082853	9174863	XM_018692157.2	0	+
NC_066852.1	15368384	15370384	1	0	+	NC_066852.1	15341888	15387344	XM_018691898.2	0	-
NC_066852.1	19093334	19095334	1	0	+	NC_066852.1	19075879	19114210	XM_018692505.2	0	-

-k 2 for nearest 2
(base) ➜  fst-density bedtools closest -a local-maxima.bed -b ../genome-annotation/lates.bed -d -k 2 > closest.bed 

cut -f 10 closest.bed | while read line; do grep $line ../genome-annotation/GCF_001640805.2_TLL_Latcal_v3_genomic.gff | grep "Parent=gene" >> parent-genes-for-bed-closest.txt; done;

cygb2 - cytoglobin oxygen transport
nomo development

NC_066852.1	Gnomon	mRNA	9174986	9178108	.	-	.	ID=rna-XM_018692155.2;Parent=gene-LOC108893796;Dbxref=GeneID:108893796,GenBank:XM_018692155.2;Name=XM_018692155.2;gbkey=mRNA;gene=LOC108893796;model_evidence=Supporting evidence includes similarity to: 4 Proteins;product=spermatogenesis-associated protein 22%2C transcript variant X3;transcript_id=XM_018692155.2
