---
title: "608-WGS"
output: html_document
date: "2024-02-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```



```{r}
library(tidyverse)
library(ggrepel)
library(pcadapt)
library(adegenet)
library(vcfR)
library(snpR)
library(ggpubr)
library(RColorBrewer)
```


What if we use all samples from the WGS data set?


```{sh, eval=FALSE}
bcftools view outputs/401/renamed.vcf.gz | bcftools +fill-tags | bcftools view -q 0.05:minor -O z > outputs/608/wgs-05.vcf.gz

bcftools index outputs/608/wgs-05.vcf.gz

cat meta/lates-lgs.txt  | while read line; do bcftools view -Ob -r $line outputs/608/wgs-05.vcf.gz > outputs/608/bcf/$line.bcf; done;
for f in outputs/608/bcf/*.bcf; do bcftools index $f; done;

cat meta/lates-lgs.txt  | while read line; do bcftools view -Oz -r $line outputs/608/wgs-05.vcf.gz > outputs/608/vcf/$line.vcf.gz; done;
for f in outputs/608/vcf/*.vcf.gz; do bcftools index $f; done;

#pruned version, with bcftools 1.19-29-gf33fd1d1

bcftools +prune -m 0.20 -w 10000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/wgs-pruned.vcf.gz
cat meta/lates-lgs.txt  | while read line; do bcftools view -Oz -r $line outputs/608/wgs-pruned.vcf.gz > outputs/608/vcf-pruned/$line.vcf.gz; done;
for f in outputs/608/vcf-pruned/*.vcf.gz; do bcftools index $f; done;



#Pulling some regions:
bcftools view -r NC_066852.1:3500000-23737464 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-sub.vcf.gz     
bcftools view -r NC_066852.1:6500000-8000000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-dummy-sub.vcf.gz     


bcftools view -r NC_066852.1:3500000-23737464 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066852.1-sub.vcf.gz    
bcftools view -r NC_066837.1:5247652-28551740 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066837.1-sub.vcf.gz    
bcftools view -r NC_066835.1:1-17500000 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066835.1-sub.vcf.gz    
bcftools view -r NC_066833.1 outputs/608/wgs-05.vcf.gz -O z > outputs/608/NC_066833.1.vcf.gz    


```
 

Note: 
(base) macs-mbp-2:lates-wgs mac$ gunzip -c outputs/608/wgs-05.vcf.gz | wc -l
 2651716
(base) macs-mbp-2:lates-wgs mac$ gunzip -c outputs/401/renamed.vcf.gz | wc -l
 3327172
 
 
 
```{r}
meta<-read_csv("meta/60-Indo-split.csv") 
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast"))

```


```{r}
samples<-meta %>% select(Run) %>% rename(ID=Run)

population<-meta %>% select(Locality) %>% rename(population=Locality)

table<-cbind(samples, population)
write.table(table, "outputs/608/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```



Using a pruned version 
```{r}
sample.meta<-meta %>% select(Run, Region, Lineage)
sample.meta$Region<-gsub(" ","",meta$Region)
sample.meta$Region<-factor(sample.meta$Region, levels=c("Queensland","NorthernTerritory","PapuaNewGuinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "IndiaEasternCoast","IndiaWesternCoast"))


snps<-import.snpR.data(genotypes = "outputs/608/wgs-pruned.vcf.gz", 
      sample.meta = sample.meta)
```


```{r}
p <- plot_clusters(snps, facets = "Region", viridis.option = "H")
p$plot$pca
```


```{r, eval=FALSE}
q<-plot_structure(snps, facet = "Region", method = "snmf", k=c(1,2,3,4,5,6), qsort=FALSE, 
                  facet.order = c("Queensland","NorthernTerritory","PapuaNewGuinea",
                                              "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "IndiaEasternCoast","IndiaWesternCoast"))
save(q, file="outputs/608/q.rda")
```


```{r}
load("outputs/608/q.rda")
ent<-ggplot(q$K_plot, aes(x=k, y=Cross.Entropy)) +
  geom_line() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_discrete(limits=seq(2,6), labels=seq(2,6)) +
  theme(axis.title = element_text(size=12, face="bold")) +
  theme(axis.text=element_text(size=10)) +
  ylab("Cross Entropy\n") +
  xlab("\nK")

ent
```


```{r}
q$plot
```

## Local PCA

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 100 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 500 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`
`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 1000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t SNP -s 2000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

`./run_lostruct.R -i /Users/mac/github/lates-wgs/outputs/608/bcf -t bp -s 200000 -m 4 -I /Users/mac/github/lates-wgs/outputs/608/bcf/sample_info.tsv -j 608`

1000 snps produces 2,641 windows

```{r}
mds<-read_csv("outputs/608/mds_coords-bp2e5.csv") 

#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


## Read in chroms and split

Using a pruned version 
```{r}

wgsmeta<-meta %>% select(Run, Region, Lineage)

lc3rhoms<-read_tsv("meta/lc03-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc3ahoms<-read_tsv("meta/lc03-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc3homs<-bind_rows(lc3ahoms,lc3rhoms)
lc3homs %>% select(Run) %>% write_tsv(file="meta/lc03-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc03-homs.tsv outputs/608/NC_066835.1-sub.vcf.gz  -O z > outputs/608/NC_066835.1-homs.vcf.gz    
```

```{r}
lc3<-import.snpR.data(genotypes = "outputs/608/NC_066835.1-homs.vcf.gz", 
      sample.meta = lc3homs)
```


```{r}
p <- plot_clusters(lc3, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc3 <- calc_pairwise_fst(lc3, "Genotype", method = "WC")
stats <- get.snpR.stats(mylc3, "Genotype", stats ="fst")
```


```{r}
p3<-ggplot(stats$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
  theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("A. Lca03") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 
p3
```




### lc5
```{r}
lc5rhoms<-read_tsv("meta/lc05-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc5ahoms<-read_tsv("meta/lc05-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc5homs<-bind_rows(lc5ahoms,lc5rhoms)
lc5homs %>% select(Run) %>% write_tsv(file="meta/lc05-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc05-homs.tsv outputs/608/NC_066837.1-sub.vcf.gz  -O z > outputs/608/NC_066837.1-homs.vcf.gz    
```

```{r}
lc5<-import.snpR.data(genotypes = "outputs/608/NC_066837.1-homs.vcf.gz", 
      sample.meta = lc5homs)
```


```{r}
p <- plot_clusters(lc5, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc5 <- calc_pairwise_fst(lc5, "Genotype", method = "WC")
stats5 <- get.snpR.stats(mylc5, "Genotype", stats ="fst")
```


```{r}

p5<-ggplot(stats5$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
  theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("B. Lca05") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 
p5
```



### lc20
Using a pruned version 
```{r}
lc20rhoms<-read_tsv("meta/lc20-rhoms.tsv", col_names = c("Run")) %>% mutate(Genotype="RHom")
lc20ahoms<-read_tsv("meta/lc20-ahoms.tsv", col_names = c("Run")) %>% mutate(Genotype="AHom")
lc20homs<-bind_rows(lc20ahoms,lc20rhoms)
lc20homs %>% select(Run) %>% write_tsv(file="meta/lc20-homs.tsv", col_names = FALSE)
```

```{sh, eval=FALSE}
bcftools view -S meta/lc20-homs.tsv outputs/608/NC_066852.1-sub.vcf.gz  -O z > outputs/608/NC_066852.1-homs.vcf.gz    
```

```{r}
lc20<-import.snpR.data(genotypes = "outputs/608/NC_066852.1-homs.vcf.gz", 
      sample.meta = lc20homs)
```


```{r}
p <- plot_clusters(lc20, facets = "Genotype", viridis.option = "H")
p$plot$pca
```
```{r}
mylc20<- calc_pairwise_fst(lc20, "Genotype", method = "WC")
stats20<- get.snpR.stats(mylc20, "Genotype", stats ="fst")
```


```{r}
p20<-ggplot(stats20$pairwise %>% as_tibble() %>% filter(fst > 0.85)) +
  geom_density_2d_filled(aes(x=position, y=fst), bins=10) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("Position") +
  ylab("Fst") +
   theme(axis.title=element_text(face="bold", size=14)) +
  theme(axis.text=element_text(size=12)) +
  ggtitle("C. Lca20") +
  theme(plot.title=element_text(size=14, face="bold", hjust=0)) 

p20
```


```{r}
ggarrange(p3,p5,p20, ncol=1)
ggsave("outputs/608/fst-density.pdf", height=12, width=6.5)
```

