---
title: "800-diamond"
output: html_document
date: "2024-03-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
```


## Prospecting for novel viruses

Downloaded NCBI refseq data for all viruses and parvoviridae, in outputs/800 
Added MZ702992.1 tilapia parvoviruse to parvoviridae.faa


Build databases.... 
diamond makedb --in probes.faa -d probes

```{sh, eval=FALSE}
diamond makedb --in parvoviridae.faa -d parvoviridae 
srun -p high -t 5:00:00 --nodes=1 --mem=32GB  diamond makedb --in ncbi-refseq.faa -d ncbi-refseq 
```

Refseq parvoviridae list see details in outputs/800/nuccore_result.txt  
MZ702992.1
NC_000883.2
NC_000936.1
NC_001401.2
NC_001510.1
NC_001539.1
NC_001540.1
NC_001662.1
NC_001701.1
NC_001718.1
NC_001729.1
NC_001829.1
NC_001899.1
NC_002077.1
NC_002190.2
NC_003346.1
NC_004284.1
NC_004285.1
NC_004286.1
NC_004287.1
NC_004288.1
NC_004289.1
NC_004290.1
NC_004295.1
NC_004442.1
NC_004828.1
NC_005040.1
NC_005041.2
NC_005341.1
NC_005889.1
NC_006147.2
NC_006148.1
NC_006152.1
NC_006259.1
NC_006260.1
NC_006261.1
NC_006263.1
NC_006555.1
NC_007018.1
NC_007218.1
NC_007455.1
NC_011317.1
NC_011545.2
NC_012042.1
NC_012564.1
NC_012636.1
NC_012685.1
NC_012729.2
NC_014357.1
NC_014358.1
NC_014468.1
NC_014665.1
NC_015115.1
NC_015718.1
NC_016031.1
NC_016032.1
NC_016647.1
NC_016744.1
NC_016752.1
NC_017823.1
NC_018399.1
NC_018450.1
NC_019492.1
NC_020499.1
NC_022089.1
NC_022104.1
NC_022564.1
NC_022748.1
NC_022800.1
NC_023020.1
NC_023673.1
NC_023842.1
NC_023860.1
NC_024452.1
NC_024453.2
NC_024454.1
NC_024888.1
NC_025825.1
NC_025891.1
NC_025965.1
NC_026251.1
NC_026815.1
NC_026943.1
NC_027429.1
NC_028136.1
NC_028650.1
NC_028973.1
NC_029133.1
NC_029300.1
NC_029797.1
NC_030296.1
NC_030402.1
NC_030837.1
NC_030873.1
NC_031450.1
NC_031670.1
NC_031695.1
NC_031751.1
NC_031959.2
NC_032097.1
NC_034445.1
NC_034532.1
NC_035180.1
NC_035185.1
NC_035186.1
NC_037053.1
NC_038532.1
NC_038533.1
NC_038534.1
NC_038535.1
NC_038536.1
NC_038537.1
NC_038538.1
NC_038539.1
NC_038540.1
NC_038541.1
NC_038542.1
NC_038543.1
NC_038544.1
NC_038545.1
NC_038546.1
NC_038547.1
NC_038883.1
NC_038895.1
NC_038898.1
NC_039043.1
NC_039044.1
NC_039045.1
NC_039046.1
NC_039047.1
NC_039048.1
NC_039049.1
NC_039050.1
NC_040533.1
NC_040562.1
NC_040603.1
NC_040623.1
NC_040626.1
NC_040652.1
NC_040671.1
NC_040672.1
NC_040694.1
NC_040695.1
NC_040843.1
NC_043446.1
NC_055458.1
NC_055465.1
NC_055485.1
NC_055486.1
NC_055487.1
NC_055518.1
NC_055527.1
NC_055583.1


### Data

https://www.ncbi.nlm.nih.gov/bioproject/713978.  
https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2021.666897/full.  
Expect a couple of viruses in these fish (but maybe a new one!!)
NC_027778 is the scale drop viruse, e.g. YP_009163762.1
acc
SRR13950739
SRR13950740
SRR13950741
SRR13950742
SRR13950743
SRR13950744
SRR13950745
SRR13950746
SRR13950747
SRR13950748
SRR13950749
SRR13950750
SRR13950751
SRR13950752
SRR13950753
SRR13950754
SRR13950755
SRR13950756
SRR13950757
SRR13950758
SRR13950759
SRR13950760
SRR13950761
SRR13950762
SRR13950763
SRR13950764
SRR13950765
SRR13950766
SRR13950767
SRR13950768
SRR13950769
SRR13950770

32 samples



Downloading like so:
https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR13950739/SRR13950739

(base) maccamp@farm:~/lates-wgs/outputs/800$ cat list.txt  | while read line; do wget https://sra-pub-run-odp.s3.amazonaws.com/sra/$line/$line; done;

Then to dump like so:
```{sh, eval=FALSE}
mkdir split
for f in raw/SRR*; do echo $f; fastq-dump --outdir split --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f; done;    
```

Ugh, why did I do that. so slow.   

organizing meta
```{r}
info1<-read_csv("outputs/800/scale-drop-rna-info.csv")
info2<-read_csv("outputs/800/scale-drop-sra.csv")

rnm<-left_join(info1, info2, by=c("Sample"="Sample Accession")) %>% select(Run, bases, Experiment, SampleName) 

rnm$Status<-gsub("\\w\\d+","",rnm$SampleName)
rnm$Tissue<-gsub("\\d+","",rnm$SampleName)
rnm$Tissue<-gsub("^S|H","",rnm$Tissue)

rnm
write_csv(rnm,"meta/rna-32-meta.csv")
```
Sick Fish Muscle, Sick Fish Liver

## Search

Can assemble with megahit or similar, but will lose rare data. 

Can try combining with flash



```{sh, eval=FALSE}
srun -p high -t 8:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q split/SRR13950739_pass_1.fastq.gz -o SRR13950739_pass_1.tsv > SRR13950739_pass_1.out 2> SRR13950739_pass_1.err &
#takes < 1 hr

srun -p high -t 8:00:00 --nodes=1 --mem=32G diamond blastx -d parvoviridae -q split/SRR13950739_pass_1.fastq.gz -o SRR13950739_pass_1-parvo.tsv > SRR13950739_pass_1-parvo.out 2> SRR13950739_pass_1-parvo.err &
```
'


SO many hits, maybe a poxvirus, but not hits to salmon gill poxvirus?

base) maccamp@farm:~/lates-wgs/outputs/800$  grep SRR13950739.9999976 SRR13950739_pass_1.tsv 
SRR13950739.9999976	YP_009447989.1	100	41	0	0	26	148	1	41	1.26e-21	81.3
SRR13950739.9999976	NP_955119.1	97.6	41	1	0	26	148	1	41	2.22e-21	80.9
SRR13950739.9999976	YP_009173411.1	92.7	41	3	0	26	148	1	41	2.09e-20	78.2
SRR13950739.9999976	YP_010783327.1	92.7	41	3	0	26	148	1	41	1.72e-19	75.9
SRR13950739.9999976	YP_009238796.1	90.2	41	4	0	26	148	1	41	2.26e-19	75.5
SRR13950739.9999976	YP_004347213.1	90.2	41	4	0	26	148	1	41	2.26e-19	75.5
SRR13950739.9999976	YP_009345355.1	92.7	41	3	0	26	148	2	42	3.29e-19	75.1
SRR13950739.9999976	YP_007676278.1	92.7	41	3	0	26	148	1	41	3.89e-19	76.6
SRR13950739.9999976	YP_001426684.1	92.7	41	3	0	26	148	1	41	9.93e-19	73.9
SRR13950739.9999976	YP_010780588.1	90.2	41	4	0	26	148	1	41	1.07e-18	73.9
SRR13950739.9999976	YP_010781883.1	90.2	41	4	0	26	148	1	41	1.07e-18	73.9
SRR13950739.9999976	YP_010084826.1	87.8	41	5	0	26	148	1	41	2.00e-18	73.2
SRR13950739.9999976	YP_009552359.1	87.8	41	5	0	26	148	1	41	2.00e-18	73.2
SRR13950739.9999976	YP_008003723.1	90.2	41	4	0	26	148	1	41	2.00e-18	73.2
SRR13950739.9999976	NP_064949.1	87.8	41	5	0	26	148	1	41	2.22e-18	73.2
SRR13950739.9999976	YP_010805497.1	87.8	41	5	0	26	148	1	41	2.60e-18	73.2
SRR13950739.9999976	YP_008004581.1	90.2	41	4	0	26	148	1	41	2.60e-18	73.2
SRR13950739.9999976	YP_009186717.1	90.2	41	4	0	26	148	1	41	2.67e-18	73.2
SRR13950739.9999976	YP_009001809.1	87.8	41	5	0	26	148	1	41	5.89e-18	72.0
SRR13950739.9999976	YP_002854658.1	85.4	41	6	0	26	148	1	41	5.89e-18	72.0
SRR13950739.9999976	YP_010086303.1	90.2	41	4	0	26	148	1	41	6.20e-18	72.0
SRR13950739.9999976	YP_006607868.1	87.8	41	5	0	26	148	19	59	9.39e-18	72.0
SRR13950739.9999976	YP_008004279.1	87.8	41	5	0	26	148	1	41	1.06e-17	71.6
SRR13950739.9999976	YP_009255031.1	85.4	41	6	0	26	148	1	41	1.07e-17	71.2
SRR13950739.9999976	YP_009552587.1	82.9	41	7	0	26	148	1	41	1.13e-17	71.2

qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore'
(base) maccamp@farm:~/lates-wgs/outputs/800$ grep MZ702992.1 SRR13950739_pass_1-parvo.tsv 
SRR13950739.5379756	lcl|MZ702992.1_prot_UTI93250.1_5	53.8	26	12	0	6	83	167	192	1.08e-04	32.3
SRR13950739.5380159	lcl|MZ702992.1_prot_UTI93250.1_5	53.8	26	12	0	6	83	167	192	1.08e-04	32.3
SRR13950739.27957673	lcl|MZ702992.1_prot_UTI93247.1_2	31.6	38	26	0	33	146	253	290	9.92e-04	29.6

SRR13950739.5379756
Potential interesting seqs match general fish proteins. Hmmm, maybe align to genome first, then search non-aligning seqs?

>SRR13950739.347963 347963 length=150
CAAGAGAGGCTTGTGGCATCATTGTGGAGCTCATTCGCTCAAAAAAGATGGCAGGAAGGGCAGTGTTACTAGCAGGGCCACCAGGAACAGGAAAGACTGCTCTCGCCTTGGCTATAGCACAGGAGCTGGGAAACAAGGTGCCTTTCTGCC
>SRR13950739.5379756 5379756 length=150
GTTTTATTCCCGTTATCCCCCCAACACCAGCCGCAGCAGATATAAAAAAAGATAAAAAACCTGCATATTTTTGTCAAAGCCCTTCCCTTTTTTCCTTTTACAGTAAGTGTGGGTCTCAGGAGTGAGTCCAGCTGACTGGAACCCAGTGAG

Can build a star reference....

Checking output
 cat SRR13950739_pass_1.tsv | awk ' {if ( $3 > 90 && $4 > 30) print }' | sort -k 3 -n

>SRR13950739.46471272 46471272 length=150
GCTAGCGGCCGGCCTGCAAACATGCAGATCTTCGTGAAGACGTTGACGGGGAAGACCATCACCCTCGAGGTTGAGCCCAGCGACACTATCGAAAATGTCAAGGCTAAGATCCAAGATAAGGAAGGAATTCCCCCTGACCAGCAGCGTCTG

This interesting sequence is similar to L. calcarifer (100% ubiquitinn protein)

building STAR aligner index
module load star/2.7.6a

srun -p high -t 12:00:00 --nodes=1 --mem=32GB STAR --runThreadN 4 --runMode genomeGenerate --genomeDir /home/maccamp/lates-wgs/genome/ --genomeFastaFiles GCF_001640805.2_TLL_Latcal_v3_genomic.fna --sjdbGTFfile GCF_001640805.2_TLL_Latcal_v3_genomic.gtf

Need to build job array. Align to genome, then remove non-aligned sequences. 

Aligning with:

--outReadsUnmapped Fastx

do-align-star.slurm

```{sh, eval=FALSE}
#!/bin/bash
#
#SBATCH --time=4-00
#SBATCH --mem=10G # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --output=slurmout/star-%A-%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/star-%A-%a.err # File to which STDERR will be written


start=`date +%s`
hostname
#aklog
export baseP=/home/$USER/lates-wgs/outputs/800/
export seqP=${baseP}/split
export outP=${baseP}/align
export cwd=${baseP}/scripts
export tmpP=$cwd/tmp

if [ ! -d "${outP}" ]; then
   mkdir ${outP}
fi

if [ ! -d "${tmpP}" ]; then
   mkdir ${tmpP}
fi

name=`head -n ${SLURM_ARRAY_TASK_ID} allsamples.txt | tail -1`

module load star/2.7.6a 

call="STAR \
--runThreadN 4 \
--genomeDir /home/maccamp/lates-wgs/genome/ \
--readFilesIn $seqP/${name}_pass_1.fastq.gz $seqP/${name}_pass_2.fastq.gz  \
--readFilesCommand zcat \
--outFilterType BySJout \
--outFilterMultimapNmax 20 \
--alignSJoverhangMin 8 \
--alignSJDBoverhangMin 1 \
--outFilterMismatchNmax 999 
--outFilterMismatchNoverLmax 0.1 \
--alignIntronMin 20 \
--alignIntronMax 1000000 \
--alignMatesGapMax 1000000 \
--outSAMattributes NH HI NM MD \
--outSAMtype BAM SortedByCoordinate \
--outReadsUnmapped Fastx \
--outFileNamePrefix $outP/${name}- \
--quantMode GeneCounts"

echo $call
eval $call
```

Trialing with 3 samples
`sbatch -p med -J star-$USER --array=1-3 do-align-star.slurm `

Works, extending to 29
`sbatch -p med -J star-$USER --array=1-29 do-align-star.slurm `

SRR13950739 is healthy liver
SRR13950740 is healthy liver
SRR13950741 is sick liver

Diamond search of unaligned reads

```{sh, eval=FALSE}
srun -p high -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q align/SRR13950739-Unmapped.out.mate1 -o SRR13950739-unaligned.tsv > SRR13950739_pass_1.out 2> SRR13950739_pass_1.err &
srun -p high -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q align/SRR13950740-Unmapped.out.mate1 -o SRR13950740-unaligned.tsv > SRR13950740_pass_1.out 2> SRR13950740_pass_1.err &
srun -p high -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q align/SRR13950741-Unmapped.out.mate1 -o SRR13950741-unaligned.tsv > SRR13950741_pass_1.out 2> SRR13950741_pass_1.err &
srun -p med -t 1:00:00 --nodes=1 --mem=32G diamond blastx -d parvoviridae -q align/SRR13950741-Unmapped.out.mate1 -o SRR13950741-unaligned-parvo.tsv > SRR13950741_pass_1.out 2> SRR13950741_pass_1.err &

```
An interesting sequence, appears to be a fish protein kinase
>SRR13950741.4529774 0:N:  00
TTTGGATTTGCAAAAAGGGTGAAGGGCCGAACCTGGACGCTTTGCGGAACCCCAGAGTACCTGGCACCAGAGATCATCCTCAGTAAGGGTTACAATAAAGCAGTGGATTGGTGGGCTCTGGGAGTGCTGATATATGAGAGATCGGAAGAG

Find scale drop seqs (YP_009163*) at 100% sim and 50 AA align.

Installing megahit via conda

```{sh,eval=FALSE}
srun -p med -t 4:00:00 --nodes=1 --mem=32G megahit -1 align/SRR13950739-Unmapped.out.mate1 -2 align/SRR13950739-Unmapped.out.mate2 -o megahit-SRR13950739 --out-prefix SRR13950739 > SRR13950739-megahit.out 2> SRR13950739-megahit.err &
srun -p med -t 4:00:00 --nodes=1 --mem=32G megahit -1 align/SRR13950740-Unmapped.out.mate1 -2 align/SRR13950740-Unmapped.out.mate2 -o megahit-SRR13950740 --out-prefix SRR13950740 > SRR13950740-megahit.out 2> SRR13950740-megahit.err &
srun -p med -t 4:00:00 --nodes=1 --mem=32G megahit -1 align/SRR13950741-Unmapped.out.mate1 -2 align/SRR13950741-Unmapped.out.mate2 -o megahit-SRR13950741 --out-prefix SRR13950741 > SRR13950741-megahit.out 2> SRR13950741-megahit.err &

srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q megahit-SRR13950739/SRR13950739.contigs.fa -o SRR13950739-megahit.tsv > SRR13950739_pass_1.out 2> SRR13950739_pass_1.err &
srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q megahit-SRR13950740/SRR13950740.contigs.fa -o SRR13950740-megahit.tsv > SRR13950740_pass_1.out 2> SRR13950740_pass_1.err &
srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q megahit-SRR13950741/SRR13950741.contigs.fa -o SRR13950741-megahit.tsv > SRR13950741_pass_1.out 2> SRR13950741_pass_1.err &
```

Pulling 17 interesting sequences
(base) maccamp@farm:~/lates-wgs/outputs/800$ cat SRR13950741-megahit.tsv | awk '{ if ( $3>95 && $4 > 100) print}' | sort -k 3 -n | cut -f 1 | sort | uniq | while read line; do grep "$line " -A 1 megahit-SRR13950741/SRR13950741.contigs.fa ; done;
>k141_1000 flag=1 multi=3.0000 len=505
ATTCCTTACACATCTAAATTGTTGTTTCAGCAATTGGCGAGTATCGGAATAAAAATACTGATAGATTAAATATTAAAACATGGATAATATATTTCAATCTATAAACAATCGTAACTTTACGGCGTTTAAAAAAGCTATTCGTACTTTAACACAAGAAGATGTACCTCCTAATGTTCAGTATCATGTTATGTACATGATTATAATAAACATTGATAAATGTCTTATTTGGATAACAGAATGGTTTAGCGTTGTACCAACTCAATACACACAATCAGAGATATTAAACTGTAATAATGCACCAGATGGGGTATTACTAAATTACATTATAGAACATGGCTCATATGAGAACGTAAAACAAGCTGTAAAATTAGGATGTGATCCATTTCTAAATGGAGATGGAACCCCACCGATATTTTCTATGTACGACAAGTACCCTGATAAATTAAATGACCTTGTATTTGACAATATGACTGTCAATCCAAATACATTAAATAATTGCAAATTA
>k141_1066 flag=1 multi=4.0000 len=634
CTTTTGTGTACGTCAATCGTTTATGCAATTGAACCTGATATTTCTAAAATATTTCAGAATTCTCAGTGTAAACCACGTAGCACAAAAATAAATGTGTACAGTCTTGCGGGTAGCGATGTTAGCATAATGTATAAACCTGCATGCATTTATGTTGACAAATGTGGCGGATGTTGTAATGATGAAGCGTTAGCATGCAAACCCATTGAGAAAACAACTGTCAATGTTACTGTGCTTTCTATAGGCAACAGAAACGCTCAATTTCAACAATTCCCAGTTGTTACGCACACAAAGTGTAATTGTTTACCAAAACCATCACGGCGCGGACCCAGATAAAAACAATGAGAGAAGAAAGAATAAAAGCGTTTTTCCAATTTGTCAAATGTGCCATCCTCAACAACAACGTTAATGCTCAACAGGTGTTGGCCGAAGCTCTCCGACTCGACGTAAAAATGTTTGCGCCTTTGATTTTATGTGAATTGTTGTTTGATTGTAAAATATGTGAACAAATTAAAACATATAAACAATGTTTCTATTTACTATGTGAAAACGATTTGCAGGCTACAAAATATCTGTTAAAAGGCATAGAATTTGTTATTTGGCTATATCCTGAAACTCTCGTACCACAGGCACCGTT
>k141_1558 flag=1 multi=2.0000 len=752
ATTGACTGCACTCCTGTGTTTTTTCGACATGTTTTGGATTCATGTGTGACACAAGGATTACTGTCATTGGACTGTGCTCATATGATACAAACATCTGGAACGCAAAACCTAAACATTCTGACCAGCACCATTGTTTGTAAAGATTATCACAAGTTTCTTGTGATATTACGCAATGAAGATCGTAGTTGGTTCTTCAATCACTGGTATGTGTTATTGCGAAAGGAAAATCTGACAATCGATGAGTCTTTCGATATGTATGACGGTCTGAAACCAGAGGAATTGCATGTGACGCTTCATTATAAGAATGTGTTACAAATATTTGTAACGCATGGTTTGTTGACGTCCTCTGAACAGTCTGTGCTAGAAGCGCTGACTCCAGATTACATACGTCTATGTAGGTATGTTCGTGTTAAAGGTCCCAACGCGTCCAAATTGGCAAACAAGCTATTGAGAAGCTACTATGAAACACATCATTTGACGCTAGAAGAGGACGATGATGATGATTGGTGCGAATAGTTAGTTTTACTGGAACGTTACAGTCATTGAAAAAAGATGGATTTATGGGGACTGTTACGTCTCGGTAAAACGATGTATGACAATTTATGTACAGCGACTGATTATAACACATCGGACAAACAGTATGTTACCGAATCTACTGTTTCTCGAGGCGAGGCTGAATGTAAACGCGCAATCGAAGAACTCACTGGACACAAATTTATAAAAACACGATTTCTAAAAAATCCAGCGACGGG
>k141_1623 flag=1 multi=3.0000 len=389
GTTGACATGTAACTGTTATTAAGTGTACACTACCGAAAGTCCCGGTAACGCTATGGTTTGTTCCCACTGAAAGTAATGTTTATACTTGTGCGTAACGACTTCATCTTTGTTGTTTGTCAGTACAAAAGGCCTTTTGTCACGAACTAGCGTCACAGTCCCATGTATATTGTTTTCTTCATCTTTAAACGTTACTTGAACCTTATCGTGTGATACTATCCTGGCTTCGCTGTAAATATACAATGGATGAAAATAATTGCGTACACATTGTGGCATGTTTGTTAAAGCAACTGTTTTTGCGAATCTGACGAGAGTCATGACGCTATCGATATTGTCTATTGCTTTTTTGTACAAGCATGCTACTAACACAGCAAAGCAAAATGCTCCAAATG
>k141_1764 flag=1 multi=2.0000 len=356
GGAAATATCAGAAAATTTTACACCTTCTGAATAAGGTTTATTACCATGAGAGCGAACATTCTGTCCTTTGGTTACAATAACGCGATCTTCTGATAATTCCACATGTCCTCTGACTTTGTCTAAAGATTGATGATCTAACAAAGCAATTACTCTTGCGTCAATATTGTCTTTTTCATTTGGTTTTTCTACATGAACTGATGCACTGGGATGATTGCGACTTTGTTTGTTAGTATGGTTTTCAACAGCAATGCGCTGATGCGCATCCAATATGGCTTGCACGTCAATATATTGTCGACGTACTGATGATCTGGGTTTAGGCGGAGATCCTTTGGGTATGGACACACTGACTGGGTTAA
>k141_1809 flag=1 multi=4.0000 len=404
GTCGAGTTCTTCCTGTGATATATCAGAATAGGCTTTTGGCTCTTCTACGTGTTGCTGTTGACCGTATCGTTGACCAACAATAACGCTATTAGCTTTAATTTGCAAGCTTGGTTTAGTTCCGCAATAAACGCTGCTAATGTCGCACGTAACCATACATTTAAACTTTGCGGTTCCTAAAGTTTCAGGACTGAGGGGTACTACTTCTTGATTGATGGAATCTATTTGATGAAATGATGTAGTAATTTTAGTTGGGTCTTTAAAAGATGTTTTTAAACGAACGTATAAAACAGGTGATGATGTTTCACAAGGTACCCCACCCACCATTTTACGGTAAAGAGTGGTCATGCTTTCGGTGTAGACATTGATTTTAGCTTTGTTTAACATTTTAGCAATTTCAGGACTGG
>k141_2015 flag=1 multi=4.0000 len=706
GTTGAAACAAATAATCGTGGTGTGCATAGTGGCAATGGCAGCGGTATTTGCAGATGATGATCCTTGTACTAATGTAAAAACTCAGCTAAATACACTGTTTAATCAAATTAAAACAGAGTATGACACTAATCTAAAAACCTATTACCAGTCGATTGCTCCTAGTGCATTTGATCCTTTCAATAATACAAATTATCTGTATTCTGTGCAAGGAAATGATTATAAATGTTACACCATTTTCGAAACACTAAGTTTTCTAATGGGAGATGTTTATCCCAGAGCTACTACTAACGAGTCGGTGCGTCTTTCGCTTGCCAAAGTAGCTACTAGCTCAACGCAAGGAGCTATGGTTATGAATTTGTGCAGACAACAGTTGGGTTGCGGTCCTCCGCCTTTTGACGCTAAGACTCTATATGATGATCGCGCAGAGTATGGCGCAGATGATATTATGGCCACTCTCGATACAGCGCTAGCAAAATTTAAATTGGTGCTGGAGAGTGAAAATGTTGTTTAAACTAACAAATTTGCAGCTATGTATTTATTACACAAATATGTTACAAAATCTTGTACAGTTCGTACAGAAAGTAAATCCGGGTGAGTCCACATGGACAATATACGAGCCCACGAAACAGACATGCGTTGTTTACAGCCTACTGTGCGGAAAAGAGATTGTTCGTGTTCTGTGGGATTTCGAGATGCTGATTTAATA
>k141_2655 flag=1 multi=4.0000 len=312
GCAACAGAAGAAAATTTTAGTGGAGATCAATCATCTATAGATAGCAACATTGCATCTACTGAGGCGATTGTGACATCAAGTCACATTAAATTGTCTCATAATGACATAACCACTGCAGATATGTCAACAGTCGGTTCAGACATTAAACGCTCAAATGTCATCGGTAACTATATGCAATTTTCTGATCCAAATAGTATACACGATATGATTAAAGGTGTTATGATTGGCTCCGGTATTTCTGTTGCGACCGTCGTTGTAATATTACTTGTAATACAATTTATTTACCAATGCGATAAAATACGCAAAATCCGC
>k141_3074 flag=1 multi=4.0000 len=1055
TTGTTTTTAACTTATATAAATTTCAATTTGCTGTTTCAACGCGGTAAGAAATGGAGTGTAACATGCAATGTCCAATAAACGATCTTTGTACAGACGGTGAAAATTCACACGCTTTAAGCCATCGTAATATTCCGCATATTCATTGTAAACATCTACTATGATTGAAGTGCCATGTTTTGCAGCAATAAAAAGTCTGGCAATTTCCGTAGATGCATTGTTGACTATTAGCCAGTTATAGTATGTTTTTTCTAAACACGCAAGTCCCTCTAGTGTGCCGATGCATTTTTTGAAAAGGACATGGTCGGATTGACGTACTGATTCGATTACGTATTGTTTAATCACATTTGCTTGTTCAACACATGTGCAATTGAAACAAGTATTCAAAGCAAATGTATATGTATATTTCAGCAAAACACGTCCATGTTCTGTGGCATCAGCAACCCTGCACGCATATTGTCCCCAGCATTCATCTGTACGACATTGTGAGAAGTCGTGGTTATAGTTGTAGTATTGTACAACATCATTTCTCAAATAGTCATACATTTCAATTGCATGGGTGATCTCTTGGTTTACAGATGCCATCTTGTAAGCAAGATATTATTTTGATCAAATGTAACAATAATTAATAATCAATTTTTTAATTATATGTAATGTAGATGACTGTAGAATCTTGATTGCATTCAGGTTGTGTTTCACCTTTATATGTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAAATGTGTTTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAATACGGTTTCACATTTACCGAATTTGTATTTCCATTTGAAATACTAACTACATTAGAACCACCATTGCTTTTTACTACAAAATTAGAGTTTTGTGGTAACCTAACATCTTGATGGCCAGGTTGTTTGATAACAGTCTCACCAGTAACGGGATCAGTTGAGATAATCATAGTTAAGCTCGGTGTCATCGAGCTGGGCCAGGGCTCTTTGGGCACTTTCCTTCACCTGGGTCAGGTAGAC
>k141_4043 flag=1 multi=2.0000 len=566
GCCTTGGTTAAAATTTTCAGGAATCGCATACCATTCTGAAGTATCTACGTATTTATAGGGTTCCTGATCTTTTAGCGCTTTTTTCATTTTTCGACGTTTATTTTTACTCATAACCACGGTCCATCCTGACTCCTCTGACTGAGCCTCTTCCGTATGTACGTCTTGCTCGCTCCAAATCAAACTGGGGTTTTCCGAGATGTCCATTTACGCTATTGTGAATGTTGACCGTAAGTTCAAACAAACTATCACCTGTTGCTAACCACGCTTCTAAATCGATATAATCTAATTTTTGTTTGTAATTATTTTTACAAGTTATGCAAGGAATGAGGGCGTGTAGGTTTTTTAATATACACTTCCAAGAGTCACGATCGCTAGGGCTGAGTGTGTGGGGTGCGCTAGCAGCTGCGGTGTGCAATACATACCACAAACGAGGTCCAAATGATTTAGGATCAAAAGAATATGTCATATTTTTTACATTGCAAATATTACCATTTCACTGTTGTCTTTGATATTTAACGAAAATGAAAAATCATTTGTCACATTGTCACACCCCACTACAACTTCTA
>k141_4389 flag=1 multi=3.0000 len=452
CTTCAAAATCAAACGATTTGTCAGATATTATGATCCAATTACTAGAAAAACATGTCGATTGGCAAATGTGGTTAAATAACAAATGTATCGATGTAAATTTTAAAGATCATTCGTTGTCTGTCGTGTCACGAAAAGCATAGCCTCTAATAGTATTATTGCTTCGTTTACAATTGCTGTACGCACTAAATTCGTAAATAACGCGCTGTTGGGTTGCATACTGGGTATTGCCTGGGCAAAATGCAGCCAACCACTCAGTAAACGTTTTAAACAATTCAGCATATGTGATAACACTGTTTTCTTGTTTAATGACCATTGTTTCAATAAACGTTTCCATGTGGTTGTTTTCGCGTTGATATTCTTTGGTAGCTTCCAGCACTTTTTCTGGCATTGTGTACACGTCTGAAGGTTCTCCGCGTGTAATCATCCAAACAAACACAAGCAACCAAGCCAGA
>k141_4517 flag=1 multi=4.0000 len=401
TTTGTTTCTTCAAGTGGCGCGGAATAGATATGACACATTCACACCTATGCAAAAAGAAAATATCATGCAAATAAACGCGTTTCTTAATTCATTAATTTATACAACAAATAACAGCTAAAACCCACTCCACCCCAAAACGTCAATGCTTTCGTTATCCCCAAGTCTAACGTACAACATGCATTTACTAATTCAGGCCAACCGCCATGTTCGCAATACCAAGCGCTTTCATTTAATATATCAGTAAAAGTAGCACAGCAGACAGCGGGATCTTCCTTTAAAACAGCGCAGCTGCGAACACAATAAACCAATAGCGCAATATATCTCGGCCAATTGTAAATATTGTCTGCAAACATAGACGTGGCCACAGACTTATATTGCGTATGAGAAATATTAAATTGAAG
>k141_4634 flag=1 multi=3.0000 len=528
GTTTGAATGGACAGAAGCACCAAATATTAAATTGGCTACTAATGTCTACAAAACGGAAAAATCTTCTTTTTCTACAACTGTACATAATATGACTGTTGTAAACAATACATCTATAATATTACACAATTCTTCTGAACTTATTTCAAGGATTGTAATCTTACCTGTTTCAAACGAAAGCATTGTATGTCAAACACATGACAACTCAAATTTAACATGTGAAGTATATTCTAGTAATGTAACAGATAATATAATTGTATCGTCAAATACTGTCTCGTTGACAACCGATGAAACCCATTATTCCACAATTTCTACATTGACATCTACTCAACAATCAACAATTAATGAAACCCATTATTCTACAACTCCTTTACCGTTAACTCAACACTCAAATAGGCATACACTATCAACCAAACGATCAAACTATTTTCCATCGACAACTGCAATTTCAATTGAACGATCTACATCATCATCTACTGATATTTCTCATTTGACCACCGATGAACCACATTTGTCATCGACCAAACCAAC
>k141_54 flag=1 multi=4.0000 len=485
AAAAGAACAGCTTAAAAGTGGCGCTCAATTGCATGCGCCCGATATAAAATTTTCAGGTCATGCTGTGACCGTAGCATTGGATTATATGAAGACCAATTATTGTATTAACATCGGTAAAGATATCAATGCAGCGTCAATATAGCAAAAGCAAAACAGTTACAGATAAACTATTAAAACTCATCAAGATTACCGATTGGTCTACCTACACCATTCCGGAATATTACATTAAAACATGTACCGTTTTGAGATTGCTAAAATCTGTGATTCCATCTGTGCGGATTCAGCAGCAGGTCAAAGTCATTTTGCAGCATCACCTATGTATGTTTCTGTTTCTACTAGGCTACGCTTCTGCGGAGCATGCAGTTGCAAAGGGTCATAATGTTGTAAAACCAGAGGATGTCGATTTTGTACTGAAAAATAGACAAATACATCCACATTCTATAGTAACTCCTCCTGTACCACGAAAAATTAAAACTGAAATAGTC
>k141_691 flag=1 multi=2.0000 len=316
TTGGTTACTGTATTTTAGATGCAGGTAAATAATTTTGTTATCTATATCAGATACTCTAAACTCCAACAATGTTCGGGTCTTGTAAAAATCATGTGCATTTGTTTTATCCATTGTTTTATCTGTATTTTCAGCTACCGTACATTGCAAATCTTCACGGCTAATTGTTATTTTCTTGACAACTGTGCTATCAAAATGTTTGTTGATAAGCTGAAAATTAAACGGGATTGCAATCTCTTCTTCCGAATCGTAAAATGCTAGCGTAGGCGTATATTCTTTATTATAAGGTGGATTGATTATACTCCTTAAAACACGCTCC
>k141_882 flag=1 multi=5.0000 len=1209
CTGGCATACTGTCCGACAAATAAAATTTCATCGCGATCACTATCATGTACGACGTAAATAAAGGGTCGAGATGCTATAAATTGTAATGTTGGTTTTGTCTTTGCGCTAAAAAGAGTACATGTTAGTACCGTAGACGCATATGCTTCAGTTGACGTTTCATCTACTTCGATAACGGCGGTATGTTTTGCGGCAGAAACGGTCAATTTGTCAATTGTATTTTCGGTAAATCCAGACATGTCGCACTGTCTGTCATCGAAAATCACAAGTTCTGGAAATGTGCGTTTTAAGTCAAACGTGCTCTCTATTTTAAATTTAGGGATGCCAATTGTTACATATTTCTTGGTAAGTCGCGTTCTGTTTTTTACCAATCCATAAATATCAGTCGGCATATCGACACCTTCATCAGGTAAGATAATGAACATGTTACATTTTGATTTGTATGGAATTTCAAGCATGTCTACACCTTCTGTAACTTGCCCAACATTGTACCAACTATCATCCATCATCATCTCTACAATGTGACTAGTTCCATCGATGTGATTAAATTTATGAAGATCTGTATCTTCAAACGGGCGTTTCCAAAACTCTTTCAGGTAGACAGCATTGACTACAACTAATTTAGAATGGGGATCTAAAGATGACAAAATGCTTGTAATTTTTCCTTCTGTTTTTTGTCGTACCCAATTATTAATCTTTCCCATTGTATCTGCTACATTAATAAAGTCAACATTTTTAATGTCTGTCAATATTGCTGGATAAGCATTGTCAATAAATATACGTGTGGAACTATATTTGCTTAACGGCAAGCGTTTGACATCTTCCAAGAATTCTCGCAATTGTGAATCGCATAAACCGCGTGTCCCAATATTTACTGTACGTAAAGTGTCGCTGATGCCAACAGGTGACACACATACATTTTTGCCAGGAACAAATATTTTTGCAAACACCTTGTTAACAAACGCCATTGTTTCAGATTGTAAAATAGTCTTCGTATTCAAGAAGAGCCTCTTTTGATATATCTCTGGTTTTTGGCCAAAATTTTTTCACGACCGCAAACACAACCACAACAATCCCAAATGAGACAATGACAAAGTATAACCCTAACATTGCCACAGAAGACGAATAATATATCAATTTTTCTTCTGTTTGACATTTACCCCATGGCCATTGGATGTCGTAATATCTAAATAGACTCTGTTCAAGTCCACA
>k141_90 flag=1 multi=2.0000 len=335
TACTATGGCTCACGCTATACTGAAATATTTTCCTTCAAACAAAAACTCTTGTATTTCACCACTGGGAATTGACTATCTGTTAGAAATGTTGTTGGAAGGTGCTGCTGGAAACACCAGAACTCAATTACTACACGCATTACAAAACAAAATACCCCAATCGGTCGTACAATATATTTCTGCTAGAATGTTTGTTAACACAACTTATTTTGAGAAATGGACAAGCGGTATAGATTGCGTTGAACAAATTGACTTTAAGTTGCCCGAATCCATCGACACTGTTAATAAATGGGTTAGAGAATCTACTAAAGGTAAAATATCATCTCTGTTTAATGAAA

Neat, all these are scale drop seqs!!

(base) maccamp@farm:~/lates-wgs/outputs/800$ cat SRR13950741-megahit.tsv | awk '{ if ( $3>80 && $4 > 100) print}' | sort -k 3 -n | cut -f 1 | sort | uniq | while read line; do grep "$line " -A 1 megahit-SRR13950741/SRR13950741.contigs.fa ; done;
>k141_1000 flag=1 multi=3.0000 len=505
ATTCCTTACACATCTAAATTGTTGTTTCAGCAATTGGCGAGTATCGGAATAAAAATACTGATAGATTAAATATTAAAACATGGATAATATATTTCAATCTATAAACAATCGTAACTTTACGGCGTTTAAAAAAGCTATTCGTACTTTAACACAAGAAGATGTACCTCCTAATGTTCAGTATCATGTTATGTACATGATTATAATAAACATTGATAAATGTCTTATTTGGATAACAGAATGGTTTAGCGTTGTACCAACTCAATACACACAATCAGAGATATTAAACTGTAATAATGCACCAGATGGGGTATTACTAAATTACATTATAGAACATGGCTCATATGAGAACGTAAAACAAGCTGTAAAATTAGGATGTGATCCATTTCTAAATGGAGATGGAACCCCACCGATATTTTCTATGTACGACAAGTACCCTGATAAATTAAATGACCTTGTATTTGACAATATGACTGTCAATCCAAATACATTAAATAATTGCAAATTA
>k141_1066 flag=1 multi=4.0000 len=634
CTTTTGTGTACGTCAATCGTTTATGCAATTGAACCTGATATTTCTAAAATATTTCAGAATTCTCAGTGTAAACCACGTAGCACAAAAATAAATGTGTACAGTCTTGCGGGTAGCGATGTTAGCATAATGTATAAACCTGCATGCATTTATGTTGACAAATGTGGCGGATGTTGTAATGATGAAGCGTTAGCATGCAAACCCATTGAGAAAACAACTGTCAATGTTACTGTGCTTTCTATAGGCAACAGAAACGCTCAATTTCAACAATTCCCAGTTGTTACGCACACAAAGTGTAATTGTTTACCAAAACCATCACGGCGCGGACCCAGATAAAAACAATGAGAGAAGAAAGAATAAAAGCGTTTTTCCAATTTGTCAAATGTGCCATCCTCAACAACAACGTTAATGCTCAACAGGTGTTGGCCGAAGCTCTCCGACTCGACGTAAAAATGTTTGCGCCTTTGATTTTATGTGAATTGTTGTTTGATTGTAAAATATGTGAACAAATTAAAACATATAAACAATGTTTCTATTTACTATGTGAAAACGATTTGCAGGCTACAAAATATCTGTTAAAAGGCATAGAATTTGTTATTTGGCTATATCCTGAAACTCTCGTACCACAGGCACCGTT
>k141_1558 flag=1 multi=2.0000 len=752
ATTGACTGCACTCCTGTGTTTTTTCGACATGTTTTGGATTCATGTGTGACACAAGGATTACTGTCATTGGACTGTGCTCATATGATACAAACATCTGGAACGCAAAACCTAAACATTCTGACCAGCACCATTGTTTGTAAAGATTATCACAAGTTTCTTGTGATATTACGCAATGAAGATCGTAGTTGGTTCTTCAATCACTGGTATGTGTTATTGCGAAAGGAAAATCTGACAATCGATGAGTCTTTCGATATGTATGACGGTCTGAAACCAGAGGAATTGCATGTGACGCTTCATTATAAGAATGTGTTACAAATATTTGTAACGCATGGTTTGTTGACGTCCTCTGAACAGTCTGTGCTAGAAGCGCTGACTCCAGATTACATACGTCTATGTAGGTATGTTCGTGTTAAAGGTCCCAACGCGTCCAAATTGGCAAACAAGCTATTGAGAAGCTACTATGAAACACATCATTTGACGCTAGAAGAGGACGATGATGATGATTGGTGCGAATAGTTAGTTTTACTGGAACGTTACAGTCATTGAAAAAAGATGGATTTATGGGGACTGTTACGTCTCGGTAAAACGATGTATGACAATTTATGTACAGCGACTGATTATAACACATCGGACAAACAGTATGTTACCGAATCTACTGTTTCTCGAGGCGAGGCTGAATGTAAACGCGCAATCGAAGAACTCACTGGACACAAATTTATAAAAACACGATTTCTAAAAAATCCAGCGACGGG
>k141_1623 flag=1 multi=3.0000 len=389
GTTGACATGTAACTGTTATTAAGTGTACACTACCGAAAGTCCCGGTAACGCTATGGTTTGTTCCCACTGAAAGTAATGTTTATACTTGTGCGTAACGACTTCATCTTTGTTGTTTGTCAGTACAAAAGGCCTTTTGTCACGAACTAGCGTCACAGTCCCATGTATATTGTTTTCTTCATCTTTAAACGTTACTTGAACCTTATCGTGTGATACTATCCTGGCTTCGCTGTAAATATACAATGGATGAAAATAATTGCGTACACATTGTGGCATGTTTGTTAAAGCAACTGTTTTTGCGAATCTGACGAGAGTCATGACGCTATCGATATTGTCTATTGCTTTTTTGTACAAGCATGCTACTAACACAGCAAAGCAAAATGCTCCAAATG
>k141_1764 flag=1 multi=2.0000 len=356
GGAAATATCAGAAAATTTTACACCTTCTGAATAAGGTTTATTACCATGAGAGCGAACATTCTGTCCTTTGGTTACAATAACGCGATCTTCTGATAATTCCACATGTCCTCTGACTTTGTCTAAAGATTGATGATCTAACAAAGCAATTACTCTTGCGTCAATATTGTCTTTTTCATTTGGTTTTTCTACATGAACTGATGCACTGGGATGATTGCGACTTTGTTTGTTAGTATGGTTTTCAACAGCAATGCGCTGATGCGCATCCAATATGGCTTGCACGTCAATATATTGTCGACGTACTGATGATCTGGGTTTAGGCGGAGATCCTTTGGGTATGGACACACTGACTGGGTTAA
>k141_1809 flag=1 multi=4.0000 len=404
GTCGAGTTCTTCCTGTGATATATCAGAATAGGCTTTTGGCTCTTCTACGTGTTGCTGTTGACCGTATCGTTGACCAACAATAACGCTATTAGCTTTAATTTGCAAGCTTGGTTTAGTTCCGCAATAAACGCTGCTAATGTCGCACGTAACCATACATTTAAACTTTGCGGTTCCTAAAGTTTCAGGACTGAGGGGTACTACTTCTTGATTGATGGAATCTATTTGATGAAATGATGTAGTAATTTTAGTTGGGTCTTTAAAAGATGTTTTTAAACGAACGTATAAAACAGGTGATGATGTTTCACAAGGTACCCCACCCACCATTTTACGGTAAAGAGTGGTCATGCTTTCGGTGTAGACATTGATTTTAGCTTTGTTTAACATTTTAGCAATTTCAGGACTGG
>k141_2015 flag=1 multi=4.0000 len=706
GTTGAAACAAATAATCGTGGTGTGCATAGTGGCAATGGCAGCGGTATTTGCAGATGATGATCCTTGTACTAATGTAAAAACTCAGCTAAATACACTGTTTAATCAAATTAAAACAGAGTATGACACTAATCTAAAAACCTATTACCAGTCGATTGCTCCTAGTGCATTTGATCCTTTCAATAATACAAATTATCTGTATTCTGTGCAAGGAAATGATTATAAATGTTACACCATTTTCGAAACACTAAGTTTTCTAATGGGAGATGTTTATCCCAGAGCTACTACTAACGAGTCGGTGCGTCTTTCGCTTGCCAAAGTAGCTACTAGCTCAACGCAAGGAGCTATGGTTATGAATTTGTGCAGACAACAGTTGGGTTGCGGTCCTCCGCCTTTTGACGCTAAGACTCTATATGATGATCGCGCAGAGTATGGCGCAGATGATATTATGGCCACTCTCGATACAGCGCTAGCAAAATTTAAATTGGTGCTGGAGAGTGAAAATGTTGTTTAAACTAACAAATTTGCAGCTATGTATTTATTACACAAATATGTTACAAAATCTTGTACAGTTCGTACAGAAAGTAAATCCGGGTGAGTCCACATGGACAATATACGAGCCCACGAAACAGACATGCGTTGTTTACAGCCTACTGTGCGGAAAAGAGATTGTTCGTGTTCTGTGGGATTTCGAGATGCTGATTTAATA
>k141_2655 flag=1 multi=4.0000 len=312
GCAACAGAAGAAAATTTTAGTGGAGATCAATCATCTATAGATAGCAACATTGCATCTACTGAGGCGATTGTGACATCAAGTCACATTAAATTGTCTCATAATGACATAACCACTGCAGATATGTCAACAGTCGGTTCAGACATTAAACGCTCAAATGTCATCGGTAACTATATGCAATTTTCTGATCCAAATAGTATACACGATATGATTAAAGGTGTTATGATTGGCTCCGGTATTTCTGTTGCGACCGTCGTTGTAATATTACTTGTAATACAATTTATTTACCAATGCGATAAAATACGCAAAATCCGC
>k141_3074 flag=1 multi=4.0000 len=1055
TTGTTTTTAACTTATATAAATTTCAATTTGCTGTTTCAACGCGGTAAGAAATGGAGTGTAACATGCAATGTCCAATAAACGATCTTTGTACAGACGGTGAAAATTCACACGCTTTAAGCCATCGTAATATTCCGCATATTCATTGTAAACATCTACTATGATTGAAGTGCCATGTTTTGCAGCAATAAAAAGTCTGGCAATTTCCGTAGATGCATTGTTGACTATTAGCCAGTTATAGTATGTTTTTTCTAAACACGCAAGTCCCTCTAGTGTGCCGATGCATTTTTTGAAAAGGACATGGTCGGATTGACGTACTGATTCGATTACGTATTGTTTAATCACATTTGCTTGTTCAACACATGTGCAATTGAAACAAGTATTCAAAGCAAATGTATATGTATATTTCAGCAAAACACGTCCATGTTCTGTGGCATCAGCAACCCTGCACGCATATTGTCCCCAGCATTCATCTGTACGACATTGTGAGAAGTCGTGGTTATAGTTGTAGTATTGTACAACATCATTTCTCAAATAGTCATACATTTCAATTGCATGGGTGATCTCTTGGTTTACAGATGCCATCTTGTAAGCAAGATATTATTTTGATCAAATGTAACAATAATTAATAATCAATTTTTTAATTATATGTAATGTAGATGACTGTAGAATCTTGATTGCATTCAGGTTGTGTTTCACCTTTATATGTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAAATGTGTTTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAATACGGTTTCACATTTACCGAATTTGTATTTCCATTTGAAATACTAACTACATTAGAACCACCATTGCTTTTTACTACAAAATTAGAGTTTTGTGGTAACCTAACATCTTGATGGCCAGGTTGTTTGATAACAGTCTCACCAGTAACGGGATCAGTTGAGATAATCATAGTTAAGCTCGGTGTCATCGAGCTGGGCCAGGGCTCTTTGGGCACTTTCCTTCACCTGGGTCAGGTAGAC
>k141_4043 flag=1 multi=2.0000 len=566
GCCTTGGTTAAAATTTTCAGGAATCGCATACCATTCTGAAGTATCTACGTATTTATAGGGTTCCTGATCTTTTAGCGCTTTTTTCATTTTTCGACGTTTATTTTTACTCATAACCACGGTCCATCCTGACTCCTCTGACTGAGCCTCTTCCGTATGTACGTCTTGCTCGCTCCAAATCAAACTGGGGTTTTCCGAGATGTCCATTTACGCTATTGTGAATGTTGACCGTAAGTTCAAACAAACTATCACCTGTTGCTAACCACGCTTCTAAATCGATATAATCTAATTTTTGTTTGTAATTATTTTTACAAGTTATGCAAGGAATGAGGGCGTGTAGGTTTTTTAATATACACTTCCAAGAGTCACGATCGCTAGGGCTGAGTGTGTGGGGTGCGCTAGCAGCTGCGGTGTGCAATACATACCACAAACGAGGTCCAAATGATTTAGGATCAAAAGAATATGTCATATTTTTTACATTGCAAATATTACCATTTCACTGTTGTCTTTGATATTTAACGAAAATGAAAAATCATTTGTCACATTGTCACACCCCACTACAACTTCTA
>k141_4337 flag=1 multi=3.0000 len=497
GACACGCAACGAGGAGACATCACCTTCCCGGAAAATAAACACAAACGACCTACAGGTGAAGTATGGCTCGTACCAAGCAGACCGCTCGTAAATCCACCGGAGGAAAGGCGCCTAGGAAGCAGCTAGCCACCAAAGCTGCTAGGAAAAGCGCGCCCTCAACCGGCGGTGTGAAGAAGCCCCATAGATACAGGCCTGGTACTGTTGCTCTGCGTGAGATTCGTCGGTACCAGAAATCCACTGAGCTGCTCATCAGGAAGCTGCCCTTCCAGCGCCTTGTCAGGGAAATCGCCCAAGATTTCAAGACAGATCTGCGTTTCCAGAGTGCAGCTATTGGCGCTCTCCAGGAAGCCAGTGAGGCATACTTGGTTGGACTGTTTGAGGACACCAACCTGTGCGCCATCCATGCAAAGAGGGTTACCATCATGCCCAAGGACATTCAGCTGGCCAGGCGAATTAGAGGAGAGCGTGCATAAATGATTGGATTTTATCTTTTTAGT
>k141_4389 flag=1 multi=3.0000 len=452
CTTCAAAATCAAACGATTTGTCAGATATTATGATCCAATTACTAGAAAAACATGTCGATTGGCAAATGTGGTTAAATAACAAATGTATCGATGTAAATTTTAAAGATCATTCGTTGTCTGTCGTGTCACGAAAAGCATAGCCTCTAATAGTATTATTGCTTCGTTTACAATTGCTGTACGCACTAAATTCGTAAATAACGCGCTGTTGGGTTGCATACTGGGTATTGCCTGGGCAAAATGCAGCCAACCACTCAGTAAACGTTTTAAACAATTCAGCATATGTGATAACACTGTTTTCTTGTTTAATGACCATTGTTTCAATAAACGTTTCCATGTGGTTGTTTTCGCGTTGATATTCTTTGGTAGCTTCCAGCACTTTTTCTGGCATTGTGTACACGTCTGAAGGTTCTCCGCGTGTAATCATCCAAACAAACACAAGCAACCAAGCCAGA
>k141_4517 flag=1 multi=4.0000 len=401
TTTGTTTCTTCAAGTGGCGCGGAATAGATATGACACATTCACACCTATGCAAAAAGAAAATATCATGCAAATAAACGCGTTTCTTAATTCATTAATTTATACAACAAATAACAGCTAAAACCCACTCCACCCCAAAACGTCAATGCTTTCGTTATCCCCAAGTCTAACGTACAACATGCATTTACTAATTCAGGCCAACCGCCATGTTCGCAATACCAAGCGCTTTCATTTAATATATCAGTAAAAGTAGCACAGCAGACAGCGGGATCTTCCTTTAAAACAGCGCAGCTGCGAACACAATAAACCAATAGCGCAATATATCTCGGCCAATTGTAAATATTGTCTGCAAACATAGACGTGGCCACAGACTTATATTGCGTATGAGAAATATTAAATTGAAG
>k141_4634 flag=1 multi=3.0000 len=528
GTTTGAATGGACAGAAGCACCAAATATTAAATTGGCTACTAATGTCTACAAAACGGAAAAATCTTCTTTTTCTACAACTGTACATAATATGACTGTTGTAAACAATACATCTATAATATTACACAATTCTTCTGAACTTATTTCAAGGATTGTAATCTTACCTGTTTCAAACGAAAGCATTGTATGTCAAACACATGACAACTCAAATTTAACATGTGAAGTATATTCTAGTAATGTAACAGATAATATAATTGTATCGTCAAATACTGTCTCGTTGACAACCGATGAAACCCATTATTCCACAATTTCTACATTGACATCTACTCAACAATCAACAATTAATGAAACCCATTATTCTACAACTCCTTTACCGTTAACTCAACACTCAAATAGGCATACACTATCAACCAAACGATCAAACTATTTTCCATCGACAACTGCAATTTCAATTGAACGATCTACATCATCATCTACTGATATTTCTCATTTGACCACCGATGAACCACATTTGTCATCGACCAAACCAAC
>k141_54 flag=1 multi=4.0000 len=485
AAAAGAACAGCTTAAAAGTGGCGCTCAATTGCATGCGCCCGATATAAAATTTTCAGGTCATGCTGTGACCGTAGCATTGGATTATATGAAGACCAATTATTGTATTAACATCGGTAAAGATATCAATGCAGCGTCAATATAGCAAAAGCAAAACAGTTACAGATAAACTATTAAAACTCATCAAGATTACCGATTGGTCTACCTACACCATTCCGGAATATTACATTAAAACATGTACCGTTTTGAGATTGCTAAAATCTGTGATTCCATCTGTGCGGATTCAGCAGCAGGTCAAAGTCATTTTGCAGCATCACCTATGTATGTTTCTGTTTCTACTAGGCTACGCTTCTGCGGAGCATGCAGTTGCAAAGGGTCATAATGTTGTAAAACCAGAGGATGTCGATTTTGTACTGAAAAATAGACAAATACATCCACATTCTATAGTAACTCCTCCTGTACCACGAAAAATTAAAACTGAAATAGTC
>k141_691 flag=1 multi=2.0000 len=316
TTGGTTACTGTATTTTAGATGCAGGTAAATAATTTTGTTATCTATATCAGATACTCTAAACTCCAACAATGTTCGGGTCTTGTAAAAATCATGTGCATTTGTTTTATCCATTGTTTTATCTGTATTTTCAGCTACCGTACATTGCAAATCTTCACGGCTAATTGTTATTTTCTTGACAACTGTGCTATCAAAATGTTTGTTGATAAGCTGAAAATTAAACGGGATTGCAATCTCTTCTTCCGAATCGTAAAATGCTAGCGTAGGCGTATATTCTTTATTATAAGGTGGATTGATTATACTCCTTAAAACACGCTCC
>k141_882 flag=1 multi=5.0000 len=1209
CTGGCATACTGTCCGACAAATAAAATTTCATCGCGATCACTATCATGTACGACGTAAATAAAGGGTCGAGATGCTATAAATTGTAATGTTGGTTTTGTCTTTGCGCTAAAAAGAGTACATGTTAGTACCGTAGACGCATATGCTTCAGTTGACGTTTCATCTACTTCGATAACGGCGGTATGTTTTGCGGCAGAAACGGTCAATTTGTCAATTGTATTTTCGGTAAATCCAGACATGTCGCACTGTCTGTCATCGAAAATCACAAGTTCTGGAAATGTGCGTTTTAAGTCAAACGTGCTCTCTATTTTAAATTTAGGGATGCCAATTGTTACATATTTCTTGGTAAGTCGCGTTCTGTTTTTTACCAATCCATAAATATCAGTCGGCATATCGACACCTTCATCAGGTAAGATAATGAACATGTTACATTTTGATTTGTATGGAATTTCAAGCATGTCTACACCTTCTGTAACTTGCCCAACATTGTACCAACTATCATCCATCATCATCTCTACAATGTGACTAGTTCCATCGATGTGATTAAATTTATGAAGATCTGTATCTTCAAACGGGCGTTTCCAAAACTCTTTCAGGTAGACAGCATTGACTACAACTAATTTAGAATGGGGATCTAAAGATGACAAAATGCTTGTAATTTTTCCTTCTGTTTTTTGTCGTACCCAATTATTAATCTTTCCCATTGTATCTGCTACATTAATAAAGTCAACATTTTTAATGTCTGTCAATATTGCTGGATAAGCATTGTCAATAAATATACGTGTGGAACTATATTTGCTTAACGGCAAGCGTTTGACATCTTCCAAGAATTCTCGCAATTGTGAATCGCATAAACCGCGTGTCCCAATATTTACTGTACGTAAAGTGTCGCTGATGCCAACAGGTGACACACATACATTTTTGCCAGGAACAAATATTTTTGCAAACACCTTGTTAACAAACGCCATTGTTTCAGATTGTAAAATAGTCTTCGTATTCAAGAAGAGCCTCTTTTGATATATCTCTGGTTTTTGGCCAAAATTTTTTCACGACCGCAAACACAACCACAACAATCCCAAATGAGACAATGACAAAGTATAACCCTAACATTGCCACAGAAGACGAATAATATATCAATTTTTCTTCTGTTTGACATTTACCCCATGGCCATTGGATGTCGTAATATCTAAATAGACTCTGTTCAAGTCCACA
>k141_90 flag=1 multi=2.0000 len=335
TACTATGGCTCACGCTATACTGAAATATTTTCCTTCAAACAAAAACTCTTGTATTTCACCACTGGGAATTGACTATCTGTTAGAAATGTTGTTGGAAGGTGCTGCTGGAAACACCAGAACTCAATTACTACACGCATTACAAAACAAAATACCCCAATCGGTCGTACAATATATTTCTGCTAGAATGTTTGTTAACACAACTTATTTTGAGAAATGGACAAGCGGTATAGATTGCGTTGAACAAATTGACTTTAAGTTGCCCGAATCCATCGACACTGTTAATAAATGGGTTAGAGAATCTACTAAAGGTAAAATATCATCTCTGTTTAATGAAA


These two are lower scoring aligns, one looks to be scale drop virus, one looks to be something else entirely (Fish histone mRNA)
>k141_3074 flag=1 multi=4.0000 len=1055
TTGTTTTTAACTTATATAAATTTCAATTTGCTGTTTCAACGCGGTAAGAAATGGAGTGTAACATGCAATGTCCAATAAACGATCTTTGTACAGACGGTGAAAATTCACACGCTTTAAGCCATCGTAATATTCCGCATATTCATTGTAAACATCTACTATGATTGAAGTGCCATGTTTTGCAGCAATAAAAAGTCTGGCAATTTCCGTAGATGCATTGTTGACTATTAGCCAGTTATAGTATGTTTTTTCTAAACACGCAAGTCCCTCTAGTGTGCCGATGCATTTTTTGAAAAGGACATGGTCGGATTGACGTACTGATTCGATTACGTATTGTTTAATCACATTTGCTTGTTCAACACATGTGCAATTGAAACAAGTATTCAAAGCAAATGTATATGTATATTTCAGCAAAACACGTCCATGTTCTGTGGCATCAGCAACCCTGCACGCATATTGTCCCCAGCATTCATCTGTACGACATTGTGAGAAGTCGTGGTTATAGTTGTAGTATTGTACAACATCATTTCTCAAATAGTCATACATTTCAATTGCATGGGTGATCTCTTGGTTTACAGATGCCATCTTGTAAGCAAGATATTATTTTGATCAAATGTAACAATAATTAATAATCAATTTTTTAATTATATGTAATGTAGATGACTGTAGAATCTTGATTGCATTCAGGTTGTGTTTCACCTTTATATGTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAAATGTGTTTTTGCCACGGATGTTCTTGTCCTTCAAACGTTTGCCACGGATGTTTTTGCCAATACGGTTTCACATTTACCGAATTTGTATTTCCATTTGAAATACTAACTACATTAGAACCACCATTGCTTTTTACTACAAAATTAGAGTTTTGTGGTAACCTAACATCTTGATGGCCAGGTTGTTTGATAACAGTCTCACCAGTAACGGGATCAGTTGAGATAATCATAGTTAAGCTCGGTGTCATCGAGCTGGGCCAGGGCTCTTTGGGCACTTTCCTTCACCTGGGTCAGGTAGAC
>k141_4337 flag=1 multi=3.0000 len=497
GACACGCAACGAGGAGACATCACCTTCCCGGAAAATAAACACAAACGACCTACAGGTGAAGTATGGCTCGTACCAAGCAGACCGCTCGTAAATCCACCGGAGGAAAGGCGCCTAGGAAGCAGCTAGCCACCAAAGCTGCTAGGAAAAGCGCGCCCTCAACCGGCGGTGTGAAGAAGCCCCATAGATACAGGCCTGGTACTGTTGCTCTGCGTGAGATTCGTCGGTACCAGAAATCCACTGAGCTGCTCATCAGGAAGCTGCCCTTCCAGCGCCTTGTCAGGGAAATCGCCCAAGATTTCAAGACAGATCTGCGTTTCCAGAGTGCAGCTATTGGCGCTCTCCAGGAAGCCAGTGAGGCATACTTGGTTGGACTGTTTGAGGACACCAACCTGTGCGCCATCCATGCAAAGAGGGTTACCATCATGCCCAAGGACATTCAGCTGGCCAGGCGAATTAGAGGAGAGCGTGCATAAATGATTGGATTTTATCTTTTTAGT

## Things to check

(1) Adapters in base RNA data
(2) Trimming of input RNA
(3) STAR alignment settings 
(4) Content of refseq database


Some other database

Accession: PRJNA352735 ID: 352735
Lates calcarcifer hepatic transcriptome Raw sequence reads

Taxonomy: Lates calcarifer (barramundi perch)
Project data type: Raw sequence reads
Scope: Multiisolate
hepatic transcriptome of barramundi caught in the N. Johnstone River, in Barratta Creek, and in the Wenlock Catchment, in North Queensland, Australia
https://www.sciencedirect.com/science/article/pii/S0141113616303154?via%3Dihub.  
. Illumina's TruSeq stranded mRNA sample preparation was used and the manufacturer's protocol was followed. The samples were run on an Illumina HiSeq 2000 with 100 base pair reads. The sample reads were deposited in the CSIRO data access portal and at NCBI at PRJNA352735 (SRP093513). Data are also available via the CSIRO data access portal at http://doi.org/10.4225/08/583766535a240.

SRR5032847
SRR5032848
SRR5032849
SRR5032850
SRR5032851
SRR5032852
SRR5032853
SRR5032854
SRR5032855
SRR5032856
SRR5032857
SRR5032858
SRR5032859
SRR5032860
SRR5032861
SRR5032862
SRR5032863


https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-11/SRR005/032/SRR5032863/SRR5032863.sralite.1

in liver-raw
 cat list.txt  | while read line; do wget https://sra-pub-run-odp.s3.amazonaws.com/sra/$line/$line; done;
 
```{sh, eval=FALSE}
mkdir split-liver
for f in raw-liver/SRR*; do echo $f; fastq-dump --outdir split-liver --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f; done;    
```



```{sh, eval=FALSE}
#!/bin/bash
#
#SBATCH --time=4-00
#SBATCH --mem=10G # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --output=slurmout/star-%A-%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/star-%A-%a.err # File to which STDERR will be written


start=`date +%s`
hostname
#aklog
export baseP=/home/$USER/lates-wgs/outputs/800/
export seqP=${baseP}/split-liver
export outP=${baseP}/align-liver
export cwd=${baseP}/scripts
export tmpP=$cwd/tmp

if [ ! -d "${outP}" ]; then
   mkdir ${outP}
fi

if [ ! -d "${tmpP}" ]; then
   mkdir ${tmpP}
fi

name=`head -n ${SLURM_ARRAY_TASK_ID} liver-samples.txt | tail -1`

module load star/2.7.6a 

call="STAR \
--runThreadN 4 \
--genomeDir /home/maccamp/lates-wgs/genome/ \
--readFilesIn $seqP/${name}_pass.fastq.gz   \
--readFilesCommand zcat \
--outFilterType BySJout \
--outFilterMultimapNmax 20 \
--alignSJoverhangMin 8 \
--alignSJDBoverhangMin 1 \
--outFilterMismatchNmax 999 
--outFilterMismatchNoverLmax 0.1 \
--alignIntronMin 20 \
--alignIntronMax 1000000 \
--alignMatesGapMax 1000000 \
--outSAMattributes NH HI NM MD \
--outSAMtype BAM SortedByCoordinate \
--outReadsUnmapped Fastx \
--outFileNamePrefix $outP/${name}- \
--quantMode GeneCounts"

echo $call
eval $call
```

sbatch -p med -J star-$USER --array=1-17 do-align-liver.slurm

Set up megahit:

sbatch -p med -J megahit-$USER --array=1-17 assemble-liver.slurm

assemble-liver.slurm

```{sh, eval=FALSE}
#!/bin/bash
#
#SBATCH --time=4:00:00
#SBATCH --mem=32G # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --output=slurmout/star-%A-%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/star-%A-%a.err # File to which STDERR will be written


start=`date +%s`
hostname
#aklog
export baseP=/home/$USER/lates-wgs/outputs/800/
export seqP=${baseP}/align-liver
export outP=${baseP}/assemble-liver
export cwd=${baseP}/scripts
export tmpP=$cwd/tmp

if [ ! -d "${outP}" ]; then
   mkdir ${outP}
fi

if [ ! -d "${tmpP}" ]; then
   mkdir ${tmpP}
fi

name=`head -n ${SLURM_ARRAY_TASK_ID} liver-samples.txt | tail -1`


call="megahit \
-r $seqP/${name}-Unmapped.out.mate1 \
-o $outP/${name}
--out-prefix ${name}"

echo $call
eval $call
```


srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q assemble-liver/SRR5032847/SRR5032847.contigs.fa -o diamond-liver/SRR5032847.tsv > diamond-liver/SRR5032847.out 2> diamond-liver/SRR5032847.err &


srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d parvoviridae -q assemble-liver/SRR5032847/SRR5032847.contigs.fa -o diamond-liver/SRR5032847-parvoviridae.tsv > diamond-liver/SRR5032847.out 2> diamond-liver/SRR5032847.err &

cat *mate1 > all-unaligned.fastq
srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d parvoviridae -q align-liver/all-unaligned.fastq -o diamond-liver/all-unaligned-parvoviridae.tsv > diamond-liver/all.out 2> diamond-liver/all.err &
srun -p med -t 4:00:00 --nodes=1 --mem=32G diamond blastx -d ncbi-refseq -q align-liver/all-unaligned.fastq -o diamond-liver/all-unaligned-refseq.tsv > diamond-liver/all.out 2> diamond-liver/all.err &

(base) maccamp@farm:~/lates-wgs/outputs/800$ cat diamond-liver/SRR5032847.tsv |awk '{ if ( $3>90 && $4 > 100) print}' | cut -f 1 | sort | uniq  | while read line; do grep "$line " -A 1 assemble-liver/SRR5032847/SRR5032847.contigs.fa ; done;
