---
title: "900-lakdiva"
output: html_document
date: "2024-05-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

Sequenced 150 bp pe on a novaseq

Lane	Sample Name	Paired Reads	Data Yield (bp)
150bp Paired End - Flowcell ID: 22KM77LT3
7	12-1A	83,967,562	25.36 Gb
12-2A	111,926,702	33.80 Gb
12-3A	83,917,724	25.34 Gb
12-4A	81,371,582	24.57 Gb
12-5A	89,870,104	27.14 Gb
13-1A	88,117,390	26.61 Gb
13-2A	110,779,590	33.46 Gb
13-3A	99,227,904	29.97 Gb
14-1A	97,398,835	29.41 Gb
14-2A	117,971,280	35.63 Gb
Total	964,548,673	291.29 Gb

```{r}
library(tidyverse)
library(dartR)
library(ggtree)
library(phangorn)
library(tanggle)
library(phytools)
library(ggpubr)
library(ggrepel)
```

Validating checksums


Include with meta, align,
 bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Also, these data are 150 bp pe sequencing, most of the data is 100 bp pe
Trim off 50 bp, can use seqtk

then:    

_1_ doIbs PCA/admixture.     

_2_ Call SNPs with japonicus for rooted phylo (Species Tree)


```{r}
m71<-read_csv("meta/71-Indo-split.csv")
m71 %>% filter(Region=="Sri Lanka")
m70<-m71 %>% filter(Lineage != "japonicus")
m71 %>% select(Path) %>% write_tsv(file="bamlists/71.bamlist", col_names = FALSE)
m71 %>% select(Run) %>% write_tsv(file="bamlists/71.names", col_names = FALSE)

ind<-m71 %>% filter(Lineage %in% c("IND","lakdiva"))
ind %>% select(Path) %>% write_tsv(file="bamlists/ind.bamlist", col_names = FALSE)
ind %>% select(Run) %>% write_tsv(file="bamlists/ind.names", col_names = FALSE)


```

```{r}
m71 %>% filter(Region=="Sri Lanka") %>% write_csv("meta/sl-meta.csv")
```

## Trim
e.g. seqtk trimfq -e 50 data/new-data/NS01_RA.fastq > data/new-data-trim/NS01_RA.fastq

module load seqtk/1.3 
module load parallel
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 901.1-trim.sh

```{r}
coms<-m71 %>% filter(Region=="Sri Lanka") %>% mutate(Command1=paste0("seqtk trimfq -e 50 data/lakdiva/", Run, "_R1.fastq.gz | gzip > data/lakdiva-trim/", Run, "_R1.fastq.gz")) %>% mutate(Command2=paste0("seqtk trimfq -e 50 data/lakdiva/", Run, "_R2.fastq.gz | gzip > data/lakdiva-trim/", Run, "_R2.fastq.gz")) %>% select(Command1, Command2)

c<-c(coms$Command1, coms$Command2)
c %>% as_tibble() %>% write_tsv("900.1-trim.sh", col_names = FALSE)
```

Align.... Our new fish have much higher coverage than the old fish.

Set up to downsample to ~7X 


```{r}
down <- m70 %>% filter (Lineage == "lakdiva" ) %>% mutate(Frac=7/Coverage) %>% 
  mutate(Command = paste0("samtools view -bs ", Frac, " ", Path, " > ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam"))

down %>% select(Command) %>% write_tsv(file="900.1-downsample-sl.sh", col_names = FALSE)
```

module load parallel
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 900.1-downsample-sl.sh

don't forget to index!!!


Questions:

Placement of lakdiva vs. other lineages?    
Genetic structure of lakdiva present?    
Are the homozygous for inversion genotypes?

Generate PCA and GL file

```{r}
m70<-m71 %>% filter(Lineage != "japonicus") 
m70 %>% select(Path) %>% write_tsv(file="bamlists/70.bamlist", col_names = FALSE)
m70 %>% mutate(Path2=ifelse(Lineage=="lakdiva",paste0("data/downsample/", Run, ".reduced.bam"),Path)) %>%
  select(Path2) %>% write_tsv(file="bamlists/70-down.bamlist", col_names = FALSE)


m70 %>% select(Run) %>% write_tsv(file="bamlists/70.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 32:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/70.bamlist -rf genome/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 63 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/900/70-ibs-90 > outputs/900/70-ibs-90.out 2> outputs/900/70-ibs-90.err &
	

srun -p high -t 8:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/70.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 63 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/900/70-ibs-90-lca01 > outputs/900/70-ibs-90-lca01.out 2> outputs/900/70-ibs-90-lca01.err &

#dropping inversion chroms 5564715
srun -p high -t 32:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/70.bamlist -rf meta/lates-lgs-no-inv.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 63 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/900/70-ibs-90-noinv > outputs/900/70-ibs-90-noinv.out 2> outputs/900/70-ibs-90-noinv.err &



srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-lca01.beagle.gz -K 2 -o outputs/900/70-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-lca01.beagle.gz -K 3 -o outputs/900/70-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-lca01.beagle.gz -K 4 -o outputs/900/70-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-lca01.beagle.gz -K 5 -o outputs/900/70-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-lca01.beagle.gz -K 6 -o outputs/900/70-K6 &




srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90.beagle.gz -K 2 -o outputs/900/70-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90.beagle.gz -K 3 -o outputs/900/70-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90.beagle.gz -K 4 -o outputs/900/70-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90.beagle.gz -K 5 -o outputs/900/70-K5 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90.beagle.gz -K 6 -o outputs/900/70-K6 &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-noinv.beagle.gz -K 2 -o outputs/900/70-K2-noinv &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-noinv.beagle.gz -K 3 -o outputs/900/70-K3-noinv &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-noinv.beagle.gz -K 4 -o outputs/900/70-K4-noinv &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-noinv.beagle.gz -K 5 -o outputs/900/70-K5-noinv &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/70-ibs-90-noinv.beagle.gz -K 6 -o outputs/900/70-K6-noinv &


# Generate GLS without doIBS
srun -p high -t 32:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 2  \
-bam bamlists/60.bamlist -rf genome/lates-lgs.txt \
-anc  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 54 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 \
-doMajorMinor 1 -doPost 1 -doMaf 1 -SNP_pval 1e-6 \
-minMaf 0.05 \
-out outputs/900/70-gls-90 > outputs/900/70-gls-90.out 2> outputs/900/70-gls-90.err &

#Just ind, 23

srun -p high -t 12:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=4 $HOME/angsd/angsd -P 4  \
-bam bamlists/ind.bamlist -rf genome/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 21 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/900/in-ibs-90 > outputs/900/ind-ibs-90.out 2> outputs/900/ind-ibs-90.err &
	
```


Visualize covMat



```{r}
meta<-m70
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                              "Indonesia-K","Indonesia-SU","Indonesia-SJ","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast","Sri Lanka"))


min(meta$Coverage)
max(meta$Coverage)
mean(meta$Coverage)
round(median(meta$Coverage),2)
```

```{r}
m <- as.matrix(read.table("outputs/900/70-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


For figure  

```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta) 

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region, shape=Lineage), size=3, alpha=0.75) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("A") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(21,23,25,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "none")

pc12
```
```{r}
text13<-covs  %>% select(Run, Region, V1, V3, Lineage) %>% filter(Lineage %in% c("IND","lakdiva")) %>% 
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region, shape=Lineage), size=3, alpha=0.75) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("B") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(21,23,25,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.3))
ggsave("outputs/900/combined-wgspcs.pdf", width=13, height=6)
ggsave("outputs/900/combined-wgspcs.jpeg", width=13, height=6)

```

Plot Admix


```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```

```{r}
PlotAdmix("outputs/900/70-K2.qopt", meta)
```

```{r}
PlotAdmix("outputs/900/70-K2-noinv.qopt", meta)
```
```{r}
PlotAdmix("outputs/900/70-K3.qopt", meta)
```


```{r}
PlotAdmix("outputs/900/70-K3-noinv.qopt", meta)
```

```{r}
PlotAdmix("outputs/900/70-K4.qopt", meta)
```

```{r}
PlotAdmix("outputs/900/70-K4-noinv.qopt", meta)
```

```{r}
PlotAdmix("outputs/900/70-K5.qopt", meta)
```

```{r}
PlotAdmix("outputs/900/70-K6.qopt", meta)
```

## Just IND fish


```{r}
m <- as.matrix(read.table("outputs/900/in-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


For figure  

```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta %>% filter(Lineage %in% c("lakdiva","IND"))) 

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region, shape=Lineage), size=3, alpha=0.75) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("A") +
  theme(plot.title = element_text(size=18, face="bold")) +
 scale_shape_manual(values=c(23,25)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
 theme(legend.position = "none")

pc12
```
```{r}
text13<-covs  %>% select(Run, Region, V1, V3, Lineage) %>% filter(Lineage %in% c("IND","lakdiva")) %>% 
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region, shape=Lineage), size=3, alpha=0.75) +
#  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("B") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(23,25)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.3))
ggsave("outputs/900/ind-wgspcs.pdf", width=13, height=6)
ggsave("outputs/900/ind-wgspcs.jpeg", width=13, height=6)

```

# Phylogeny with japonicus

At 95% threshold 
```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 68 -bam bamlists/71.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/900/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/900/snps-wgs.out 2> outputs/900/snps-wgs.err &

```

1795511 variants in plink file
NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1

Create pruned vcf
```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#renaming vcf   locally
bcftools view -r NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1 outputs/900/plink.vcf.gz | bcftools reheader --samples bamlists/71.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.20 -w 10000  > outputs/900/pruned.vcf
```










locally
```{sh, eval=FALSE}
conda activate py2
~/github/mccloud-rrt/vcf2phylip.py -i pruned.vcf 

conda deactivate
```

12398

```{sh, eval=FALSE}
iqtree2 -s pruned.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo -o DRR391968
iqtree2 -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 10000 --redo -o DRR391968
raxmlHPC-PTHREADS -T 3 -m ASC_GTRGAMMA --asc-corr lewis -p 2 -s  pruned.min4.phy.varsites.phy  -n asc -o DRR391968 
```

# SNP Calls without japonicus

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 67 -bam bamlists/70.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/900/snps-wgs-cal  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/900/snps-wgs-cal.out 2> outputs/900/snps-wgs-cal.err &
```



```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 67 -bam bamlists/70-down.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/900/snps-wgs-cal-down  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/900/snps-wgs-down.out 2> outputs/900/snps-wgs-down.err &
```

Don't forget to filter inversion chroms!

Or not, whatever

1594493 snps in 

snps-wgs-cal-down 

```{sh, eval=FALSE}
plink --tped snps-wgs-cal-down.tped --tfam snps-wgs-cal-down.tfam  --out snps-wgs-cal-down-binary --recode --allow-extra-chr --noweb
plink --ped snps-wgs-cal-down-binary.ped --map snps-wgs-cal-down-binary.map --recode vcf --allow-extra-chr -out snps-wgs-cal-down-plink
bgzip snps-wgs-cal-down-plink.vcf 
tabix snps-wgs-cal-down-plink.vcf.gz

#renaming vcf in 900 locally
bcftools view -r NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1 outputs/900/snps-wgs-cal-down-plink.vcf.gz| bcftools reheader --samples bamlists/70.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.20 -w 10000  > outputs/900/snps-wgs-cal-down-plink-pruned.vcf

bcftools view -r NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1 outputs/900/snps-wgs-cal-down-plink.vcf.gz| bcftools reheader --samples bamlists/70.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/900/newhybrids.vcf


bcftools view -r NC_066835.1:1-17500000 outputs/900/snps-wgs-cal-down-plink.vcf.gz| bcftools reheader --samples bamlists/70.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/900/newhybrids-lca03inv.vcf

bcftools view -r NC_066837.1:5247652- outputs/900/snps-wgs-cal-down-plink.vcf.gz| bcftools reheader --samples bamlists/70.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/900/newhybrids-lca05inv.vcf
```

## Inversion zone

```{r}
m<-m70
#vcf <-read.vcfR("outputs/900/newhybrids.vcf")
vcf<-read.vcfR("outputs/900/newhybrids-lca05inv.vcf")
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(m$Region)
```

Downsample
```{r}

#Checking status of Thai fish
gl2<-gl[gl$pop %in% c("Vietnam","Cambodia","Thailand","Indonesia-SU","Indonesia-K","Indonesia-SJ", "Philippines",
                     "Bangladesh","India Eastern Coast","Sri Lanka", "India Western Coast"),]

glx<-gl2
glx <- gl.compliance.check(glx) 


gl3<-gl.filter.maf(glx, threshold = 0.05)
gl3<-gl.filter.allna(gl3)
gl3
```

```{r, eval=FALSE}
hyb <- gl.nhybrids(gl3,
p0=c("India Western Coast"),
method="AvgPIC",
threshold = 0,
nhyb.directory="~/github/newhybrids", # Specify as necessary
outpath="~/github/lates-wgs/outputs/900",  # Specify as necessary, usually getwd() [= workspace]
BurnIn=10000,
sweeps=10000,
verbose=3,
plot=FALSE)

```


```{r}
out<-read_csv("outputs/900/aa-PofZ.csv",col_names = c("Sequence ID","Region","NoName","P0","P1","F1","F2","F1xP0","F1xP1"), skip=1)
out %>% arrange(Region)
```

```{r}
long<-out %>% pivot_longer(cols = 4:9, names_to = c("Class"), values_to = c("Prob"))
long$Class<-gsub("NA","P0",long$Class)

long$Region<-factor(long$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                           "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","Sri Lanka", "India Western Coast"))

long$Class<-factor(long$Class, levels=c("P0","P1","F1","F2","F1xP0","F1xP1"))
l2<-long %>% arrange(Region)
l2$`Sequence ID`<-as.factor(l2$`Sequence ID`)
```

```{r}
ggplot(l2) + 
  geom_tile(aes(x=`Sequence ID`, y=Class, fill=Prob)) +
  scale_fill_viridis_c(direction=-1) +
  facet_grid(.~Region, scales="free_x", space="free" ) +
  theme_bw()+
  theme(panel.grid=element_blank()) +
  theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1)) +
  xlab("Sequence ID")


ggsave("outputs/900/trial-new-hybrids-lca05-inversion.pdf", width=12, height=6)
```

Hey! That matches our expectations pretty well!
All F1s.

## Genome-wide

```{r}
m<-m70
vcf <-read.vcfR("outputs/900/newhybrids.vcf")
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(m$Region)
```

Downsample
```{r}

#Checking status of Thai fish
gl2<-gl[gl$pop %in% c("Vietnam","Cambodia","Thailand","Indonesia-SU","Indonesia-K","Indonesia-SJ", "Philippines",
                     "Bangladesh","India Eastern Coast","Sri Lanka", "India Western Coast"),]

glx<-gl2
glx <- gl.compliance.check(glx) 


gl3<-gl.filter.maf(glx, threshold = 0.2)
gl3<-gl.filter.allna(gl3)
gl3
```

```{r, eval=FALSE}
hyb <- gl.nhybrids(gl3,
p0=c("India Western Coast"),
method="AvgPIC",
threshold = 0,
nhyb.directory="~/github/newhybrids", # Specify as necessary
outpath="~/github/lates-wgs/outputs/900",  # Specify as necessary, usually getwd() [= workspace]
BurnIn=10000,
sweeps=10000,
verbose=3,
plot=FALSE)

```


```{r}
out<-read_csv("outputs/900/aa-PofZ.csv",col_names = c("Sequence ID","Region","NoName","P0","P1","F1","F2","F1xP0","F1xP1"), skip=1)
out %>% arrange(Region)
```

```{r}
long<-out %>% pivot_longer(cols = 4:9, names_to = c("Class"), values_to = c("Prob"))
long$Class<-gsub("NA","P0",long$Class)

long$Region<-factor(long$Region, levels=c("Queensland","Northern Territory","Papua New Guinea",
                                           "Indonesia-K","Indonesia-SJ","Indonesia-SU","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","Sri Lanka", "India Western Coast"))

long$Class<-factor(long$Class, levels=c("P0","P1","F1","F2","F1xP0","F1xP1"))
l2<-long %>% arrange(Region)
l2$`Sequence ID`<-as.factor(l2$`Sequence ID`)
```

```{r}
ggplot(l2) + 
  geom_tile(aes(x=`Sequence ID`, y=Class, fill=Prob)) +
  scale_fill_viridis_c(direction=-1) +
  facet_grid(.~Region, scales="free_x", space="free" ) +
  theme_bw()+
  theme(panel.grid=element_blank()) +
  theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1)) +
  xlab("Sequence ID")


ggsave("outputs/900/trial-new-hybrids.pdf", width=12, height=6)
```


```{sh, eval=FALSE}
conda activate py2
~/github/mccloud-rrt/vcf2phylip.py -i snps-wgs-cal-down-plink-pruned.vcf

conda deactivate
```

70 20658
70 9941 with varsites
```{sh, eval=FALSE}
iqtree2 -s snps-wgs-cal-down-plink-pruned.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -s snps-wgs-cal-down-plink-pruned.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 10000 --redo 
```


the tree

```{r}
tree<-read.tree("outputs/900/snps-wgs-cal-down-plink-pruned.min4.phy.varsites.phy.contree")
t1<-midpoint.root(tree)
t2<-ggtree(t1) + geom_nodelab(aes(label=node))

t2
```

```{r}
#t3<-as.polytomy(t1, feature='node.label', fun=function(x) as.numeric(x) < 50)

t<-t1

t<-ggtree(t, layout = 'circular')

t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```

```{r}
u<-t %<+% meta + #bind_rows(m58, rm) +
  geom_point(data=e,  fill="gray50", cex=3, alpha=1, pch=22) +
  geom_point(data=d,  fill="black", cex=3, alpha=1, pch=22) +
  geom_tippoint(aes(x=x+0.002,fill=Region, shape=Lineage), cex=3) + 
  geom_tiplab(aes(label=Region, x=x+0.01), align = FALSE, size=3) +
  #geom_text(data=t$data %>% filter(isTip==TRUE) %>% filter(label %in% Run), aes(label=label, x=x+0.01), size=3) +
  scale_shape_manual(values=c(21,23,25,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(0,.15) +
  scale_fill_viridis_d(option="H") +
  theme(legend.position = "none") +
  geom_treescale(x = 0)

u

ggsave("outputs/900/concatenated-tree-70inds.jpeg", width=9, height=9)
```


Network version

```{r}
dat<-read.dna(file="outputs/900/snps-wgs-cal-down-plink-pruned.min4.phy")
write.nexus.data(dat, file="outputs/900/snps-wgs-cal-down-plink-pruned.min4.nex")
```

```{r}
net<-read.nexus.networx("outputs/900/pruned-down-noinv.network")
```

```{r}
g<-ggsplitnet(net)  
g$data<-left_join(g$data, meta, by=c("label"="Run")) 

n<-g + 
  geom_tippoint(aes(shape=Lineage, fill=Region), cex=5, alpha=1.00) +
  geom_tiplab2(data=g$data %>% filter(Lineage=='lakdiva')) +
  geom_tiplab2(data=g$data %>% filter(Lineage!='lakdiva'), aes(label=Region)) +
  scale_shape_manual(values = c(21,23,25,24)) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  xlim(-.06,.04) +
  ylim(-.05,.05)

n
ggsave("outputs/900/wgs-network.jpeg", width=12, height=10)
ggsave("outputs/900/wgs-network.pdf", width=12, height=10)

```


## Only lakdiva

```{r}
lak<-m71 %>% filter(Lineage %in% c("lakdiva"))
lak %>% select(Path) %>% write_tsv(file="bamlists/lak.bamlist", col_names = FALSE)
lak %>% select(Run) %>% write_tsv(file="bamlists/lak.names", col_names = FALSE)
```

10 samples.  

```{sh, eval=FALSE}
srun -p high -t 32:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/lak.bamlist -rf genome/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 8 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/900/lak-ibs-80 > outputs/900/lak-ibs-80.out 2> outputs/900/lak-ibs-80.err &

```

	-> Number of sites retained after filtering: 4784503 



```{r}
m <- as.matrix(read.table("outputs/900/lak-ibs-80.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


For figure  

```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(lak)

text12<-covs %>% select(Run, Region, V1, V2) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Locality, shape=Lineage), size=3, alpha=0.75) +
#  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("A") +
  theme(plot.title = element_text(size=18, face="bold")) +
 scale_shape_manual(values=c(25)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
 theme(legend.position = "none")

pc12
```
```{r}
text13<-covs  %>% select(Run, Region, V1, V3, Lineage) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Locality, shape=Lineage), size=3, alpha=0.75) +
#  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="magma") +
  ggtitle("B") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(25)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.3))
ggsave("outputs/900/lak-wgspcs.pdf", width=13, height=6)
ggsave("outputs/900/lak-wgspcs.jpeg", width=13, height=6)

```


admixture.   

```{sh, eval=FALSE}
srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/lak-ibs-80.beagle.gz -K 2 -o outputs/900/lak-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/lak-ibs-80.beagle.gz -K 3 -o outputs/900/lak-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/lak-ibs-80.beagle.gz -K 4 -o outputs/900/lak-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/lak-ibs-80.beagle.gz -K 5 -o outputs/900/lak-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/900/lak-ibs-80.beagle.gz -K 6 -o outputs/900/lak-K6 &
```


```{r}
PlotAdmix("outputs/900/lak-K2.qopt", lak)
```

## Silly trial


```{r}
m<-read_csv("~/Downloads/m80-sorted.csv")

m$pop<-gsub("0","F",m$Pheno)
m$pop<-gsub("1","M",m$pop)
m$pop<-as.factor(m$pop)

#(base) âžœ  Downloads bcftools view -r NC_061078.1:15021030-15313438 reordered.vcf.gz | bcftools +fill-tags > sub.vcf 
#bcftools view -r NC_061078.1:15021030-15313438 reordered.vcf.gz | bcftools +fill-tags | bcftools view -i 'MAF<0.35' > sub.vcf
vcf <-read.vcfR("~/Downloads/sub.vcf")
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(m$pop)
```

Downsample
```{r}
gl$ploidy<-as.integer(rep(2,80))
glx<-gl
glx <- gl.compliance.check(glx) 



gl3<-gl.filter.maf(glx, threshold = 0.15)

gl3<-gl.filter.allna(gl3)
gl3
```

```{r, eval=FALSE}
hyb <- gl.nhybrids(gl3,
method="AvgPIC",
p0="F",
nhyb.directory="~/github/newhybrids", # Specify as necessary
outpath="~/Downloads",  # Specify as necessary, usually getwd() [= workspace]
BurnIn=50000,
sweeps=900000,
verbose=3,
plot=FALSE)

```



```{r}
out<-read_csv("~/Downloads/aa-PofZ.csv",col_names = c("Sequence ID","Region","NoName","P0","P1","F1","F2","F1xP0","F1xP1"), skip=1)
out %>% filter(Region=="M")
```

```{r}
long<-out %>% pivot_longer(cols = 4:9, names_to = c("Class"), values_to = c("Prob"))

long$Class<-factor(long$Class, levels=c("P0","P1","F1","F2","F1xP0","F1xP1"))
l2<-long %>% arrange(Region)
l2$`Sequence ID`<-as.factor(l2$`Sequence ID`)
```

```{r}
ggplot(l2) + 
  geom_tile(aes(x=`Sequence ID`, y=Class, fill=Prob)) +
  scale_fill_viridis_c(direction=-1) +
  facet_grid(.~Region, scales="free_x", space="free" ) +
  theme_bw()+
  theme(panel.grid=element_blank()) +
  theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1)) +
  xlab("Sequence ID")
```

Assigment of females to p0

```{r}
out %>% group_by(Region) %>% summarize(Count=n())
out %>% filter(Region=="F") %>% filter(P0>0.9) %>% summarize(Count=n())
out %>% filter(Region=="M") %>% filter(F1>0.9) %>% summarize(Count=n())

```

34/38 females
22/42 males

```{r}
out %>% filter(Region=="F") %>% filter(P0<0.9)
```

Pop out as males
901071	F	NoName	0.0000	0	1.00000	0.00000	0.0000	0
701931	F	NoName	0.0000	0	0.99988	0.00012	0.0000	0
902671	F	NoName	0.0000	0	1.00000	0.00000	0.0000	0
110381	F	NoName	0.0086	0	0.00000	0.00000	0.9914	0

Individual heterozygosities:

```{r}
vcf <-read.vcfR("~/Downloads/top-sites.vcf")
gl<-vcfR2genlight(vcf)
gl$pop<-as.factor(m$pop)
```

```{r}
snp<-import.snpR.data(gl)
```

```{r}
hs<-calc_hs(snp)
hs <- get.snpR.stats(hs, stats = "hs")
hs
```

```{r}
table<-hs$sample %>% as_tibble()
table<-table %>% mutate(Call=ifelse(hs == 0, "Female", "Male")) 

table%>% select(pop, Call) %>% group_by(pop,Call) %>% summarize(Count=n())
```
