---
title: "900-lakdiva"
output: html_document
date: "2024-05-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

Sequenced 150 bp pe on a novaseq

Lane	Sample Name	Paired Reads	Data Yield (bp)
150bp Paired End - Flowcell ID: 22KM77LT3
7	12-1A	83,967,562	25.36 Gb
12-2A	111,926,702	33.80 Gb
12-3A	83,917,724	25.34 Gb
12-4A	81,371,582	24.57 Gb
12-5A	89,870,104	27.14 Gb
13-1A	88,117,390	26.61 Gb
13-2A	110,779,590	33.46 Gb
13-3A	99,227,904	29.97 Gb
14-1A	97,398,835	29.41 Gb
14-2A	117,971,280	35.63 Gb
Total	964,548,673	291.29 Gb

```{r}
library(tidyverse)
```

Validating checksums


Include with meta, align,
 bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Also, these data are 150 bp pe sequencing, most of the data is 100 bp pe
Trim off 50 bp, can use seqtk

then:    

_1_ doIbs PCA/admixture.     

_2_ Call SNPs with japonicus for rooted phylo (Species Tree)


```{r}
m71<-read_csv("meta/71-Indo-split.csv")
m71 %>% filter(Region=="Sri Lanka")

m71 %>% select(Path) %>% write_tsv(file="bamlists/71.bamlist", col_names = FALSE)
m71 %>% select(Run) %>% write_tsv(file="bamlists/71.names", col_names = FALSE)

```

## Trim
e.g. seqtk trimfq -e 50 data/new-data/NS01_RA.fastq > data/new-data-trim/NS01_RA.fastq

module load seqtk/1.3 
module load parallel
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 900.1-trim.sh

```{r}
coms<-m71 %>% filter(Region=="Sri Lanka") %>% mutate(Command1=paste0("seqtk trimfq -e 50 data/lakdiva/", Run, "-R1.fastq.gz | gzip > data/lakdiva-trim/", Run, "-R1.fastq.gz")) %>% mutate(Command2=paste0("seqtk trimfq -e 50 data/lakdiva/", Run, "-R2.fastq.gz | gzip > data/lakdiva-trim/", Run, "-R2.fastq.gz")) %>% select(Command1, Command2)

c<-c(coms$Command1, coms$Command2)
c %>% as_tibble() %>% write_tsv("900.1-trim.sh", col_names = FALSE)
```

Align.... Our new fish have much higher coverage than the old fish.
Questions:

Placement of lakdiva vs. other lineages?    
Genetic structure of lakdiva present?    
Are the homozygous for inversion genotypes?

Generate PCA and GL file

```{r}
m70<-m71 %>% filter(Lineage != "japonicus") 
m70 %>% select(Path) %>% write_tsv(file="bamlists/70.bamlist", col_names = FALSE)
m70 %>% select(Run) %>% write_tsv(file="bamlists/70.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 16:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/70.bamlist -rf genome/lates-lgs.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 63 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1000/70-ibs-90 > outputs/1000/70-ibs-90.out 2> outputs/1000/70-ibs-90.err &
	

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1000/70-ibs-90.beagle.gz -K 2 -o outputs/1000/70-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1000/70-ibs-90.beagle.gz -K 3 -o outputs/1000/70-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1000/70-ibs-90.beagle.gz -K 4 -o outputs/1000/70-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1000/70-ibs-90.beagle.gz -K 5 -o outputs/1000/70-K5 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/1000/70-ibs-90.beagle.gz -K 6 -o outputs/1000/70-K6 &

# Generate GLS without doIBS
srun -p high -t 16:00:00 --mem=32G --nodes=1 $HOME/angsd/angsd -P 2  \
-bam bamlists/60.bamlist -rf genome/lates-lgs.txt \
-anc  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 54 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 \
-doMajorMinor 1 -doPost 1 -doMaf 1 -SNP_pval 1e-6 \
-minMaf 0.05 \
-out outputs/1000/70-gls-90 > outputs/1000/70-gls-90.out 2> outputs/1000/70-gls-90.err &

```


# Phylogeny with japonicus

```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 68 -bam bamlists/71.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/1000/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/1000/snps-wgs.out 2> outputs/1000/snps-wgs.err &

```


Create pruned vcf
```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#renaming vcf in 403 locally
bcftools reheader --samples bamlists/61-sample-names.txt -o outputs/403/renamed.vcf.gz outputs/403/plink.vcf.gz

bcftools +prune -l 0.20 -w 10000 outputs/403/renamed.vcf.gz > outputs/403/pruned.vcf
bcftools +prune -l 0.30 -w 10000 outputs/403/renamed.vcf.gz > outputs/403/pruned-03.vcf
```

in 403 locally   

in 403
```{sh, eval=FALSE}
~/github/mccloud-rrt/vcf2phylip.py -i pruned.vcf 
~/github/mccloud-rrt/vcf2phylip.py -i pruned-03.vcf 

```

```{sh, eval=FALSE}
iqtree -s pruned.min4.phy -st DNA -m GTR+ASC -bb 1000 --redo
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo -o DRR391968
iqtree -s pruned.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 10000 --redo -o DRR391968
raxmlHPC-PTHREADS -T 3 -m ASC_GTRGAMMA --asc-corr lewis -p 2 -s  pruned.min4.phy.varsites.phy  -n asc -o DRR391968 
```