---
title: "901-hatchery-strains"
output: html_document
date: "2024-07-31"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(dartR)
library(ggtree)
library(phangorn)
library(tanggle)
library(phytools)
library(ggpubr)
library(ggrepel)
```

Validating checksums
md5sum -c checksums.md5

all OK!     

Mainstream-1 Mainstream-2 Mainstream-3 Mainstream-4 Mainstream-5

OceanPick-1 OceanPick-11 OceanPick-15 OceanPick-16 OceanPick-17 OceanPick-20 OceanPick-3 OceanPick-5 OceanPick-8

These fish were sequenced across multiple lanes, symbolically linking all files.. 
working command

 cat Mainstream-1_*R1.fastq.gz > Mainstream1_R1.fastq.gz 

Include with meta, align,
 bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Also, these data are 150 bp pe sequencing, most of the data is 100 bp pe
Trim off 50 bp, can use seqtk

perl -pe 's/_\d+.+_R/_R/g' 

paste files.txt newnames.txt | awk '{print "mv " $1 " " $2}' > commands

```{r}
m81<-read_csv("meta/81-Indo-split.csv")
m81 %>% filter(Region %in% c("Mainstream","OceanPick"))
```
## combine
 cat Mainstream-1_*R1.fastq.gz > Mainstream1_R1.fastq.gz 

```{r}
merge<-m81 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% 
  mutate(Command=paste0("cat data/hatcheries/",BioSample, "_*R1.fastq.gz > data/hatcheries/", Run, "_R1.fastq.gz")) %>% select(Command) %>%
  write_tsv("901.1-merge.sh", col_names = FALSE)
```

module load parallel

srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=15 parallel -j 15 < 901.1-merge.sh


## Trim   

```{r}
coms<-m81 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% mutate(Command1=paste0("seqtk trimfq -e 50 data/hatcheries/", BioSample, "_R1.fastq.gz | gzip > data/hatch-trim/", BioSample, "_R1.fastq.gz")) %>% mutate(Command2=paste0("seqtk trimfq -e 50 data/hatcheries/", BioSample, "_R2.fastq.gz | gzip > data/lakdiva-trim/", Run, "_R2.fastq.gz")) %>% select(Command1, Command2)

c<-c(coms$Command1, coms$Command2)
c %>% as_tibble() %>% write_tsv("901.1-trim.sh", col_names = FALSE)
```

module load seqtk/1.3 
module load parallel

srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 901.1-trim.sh

Set up to downsample to ~7X 


```{r}
down <- m81 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% mutate(Frac=7/Coverage) %>% 
  mutate(Command = paste0("samtools view -bs ", Frac, " ", Path, " > ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam"))

down %>% select(Command) %>% write_tsv(file="901.1-downsample-sl.sh", col_names = FALSE)
```

module load parallel
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 900.1-downsample-sl.sh

don't forget to index!!!
