---
title: "901-hatchery-strains"
output: html_document
date: "2024-07-31"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(dartR)
library(ggtree)
library(phangorn)
library(tanggle)
library(phytools)
library(ggpubr)
library(ggrepel)
```

Validating checksums
md5sum -c checksums.md5

all OK!     

Mainstream-1 Mainstream-2 Mainstream-3 Mainstream-4 Mainstream-5

OceanPick-1 OceanPick-11 OceanPick-15 OceanPick-16 OceanPick-17 OceanPick-20 OceanPick-3 OceanPick-5 OceanPick-8

These fish were sequenced across multiple lanes, symbolically linking all files.. 
working command

 cat Mainstream-1_*R1.fastq.gz > Mainstream1_R1.fastq.gz 

Include with meta, align,
 bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

Also, these data are 150 bp pe sequencing, most of the data is 100 bp pe
Trim off 50 bp, can use seqtk

perl -pe 's/_\d+.+_R/_R/g' 

paste files.txt newnames.txt | awk '{print "mv " $1 " " $2}' > commands

```{r}
m86<-read_csv("meta/86-Indo-split.csv")
m86 %>% filter(Region %in% c("Mainstream","OceanPick"))
```
## combine
 cat Mainstream-1_*R1.fastq.gz > Mainstream1_R1.fastq.gz 

```{r}
merge<-m86 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% 
  mutate(Command=paste0("cat data/hatcheries/",BioSample, "_*R1.fastq.gz > data/hatcheries/", Run, "_R1.fastq.gz")) %>% select(Command) %>%
  write_tsv("901.1-merge.sh", col_names = FALSE)

merge<-m86 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% 
  mutate(Command=paste0("cat data/hatcheries/",BioSample, "_*R2.fastq.gz > data/hatcheries/", Run, "_R2.fastq.gz")) %>% select(Command) %>%
  write_tsv("901.1-merge2.sh", col_names = FALSE)
```

module load parallel

srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=15 parallel -j 15 < 901.1-merge.sh
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=15 parallel -j 15 < 901.1-merge2.sh


## Trim   

```{r}
coms<-m86 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% mutate(Command1=paste0("seqtk trimfq -e 50 data/hatcheries/", Run, "_R1.fastq.gz | gzip > data/hatch-trim/", Run, "_R1.fastq.gz")) %>% mutate(Command2=paste0("seqtk trimfq -e 50 data/hatcheries/", Run, "_R2.fastq.gz | gzip > data/hatch-trim/", Run, "_R2.fastq.gz")) %>% select(Command1, Command2)

c<-c(coms$Command1, coms$Command2)
c %>% as_tibble() %>% write_tsv("901.1-trim.sh", col_names = FALSE)
```

module load seqtk/1.3 
module load parallel

srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=15 parallel -j 15 < 901.1-trim.sh

Set up to downsample to ~7X 

align.    

 bash ../../doAlign-zipped.sh samples.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

downsample, don't forget to index!!!

```{r}
down <- m86 %>% filter(Region %in% c("Mainstream","OceanPick")) %>% mutate(Frac=7/Coverage) %>% 
  mutate(Command = paste0("samtools view -bs ", Frac, " ", "data/hatch-trim/",Run,".sort.flt.bam", " > ",   "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;"," samtools index ", "/home/maccamp/lates-wgs/data/downsample/", Run, ".reduced.bam ;")) 

down %>% select(Command) %>% write_tsv(file="901.1-downsample-sl.sh", col_names = FALSE)

```

module load parallel
srun -p high --time 12:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < 901.1-downsample-sl.sh



# Generate PCA and GL file

```{r}
m85<-m86 %>% filter(Lineage != "japonicus") 
m85 %>% select(Path) %>% write_tsv(file="bamlists/85-down.bamlist", col_names = FALSE)

m85 %>% select(Run) %>% write_tsv(file="bamlists/85.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 8:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/85-down.bamlist -r NC_066835.1:1-17500000 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 77 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/901/85-ibs-90-lca03 > outputs/901/85-ibs-90-lca03.out 2> outputs/901/85-ibs-90-lca03.err &

srun -p high -t 8:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/85-down.bamlist -r NC_066837.1:5247652- \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 77 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/901/85-ibs-90-lca05 > outputs/901/85-ibs-90-lca05.out 2> outputs/901/85-ibs-90-lca05.err &

srun -p high -t 8:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/85-down.bamlist -r NC_066852.13500000- \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 77 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/901/85-ibs-90-lca20 > outputs/901/85-ibs-90-lca20.out 2> outputs/901/85-ibs-90-lca20.err &

# NC_066833.1 only
srun -p high -t 32:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/85-down.bamlist -r NC_066833.1 \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 77 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/901/85-ibs-90-noinv > outputs/901/85-ibs-90-noinv.out 2> outputs/901/85-ibs-90-noinv.err &

#	


#dropping inversion chroms 
srun -p high -t 32:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  \
-bam bamlists/85-down.bamlist -rf meta/lates-lgs-no-inv.txt \
-ref  genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 77 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/901/85-ibs-90-noinv > outputs/901/85-ibs-90-noinv.out 2> outputs/901/85-ibs-90-noinv.err &


srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/901/85-ibs-90-noinv.beagle.gz -K 2 -o outputs/901/85-K2 &
srun -p high -t 07:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/901/85-ibs-90-noinv.beagle.gz -K 3 -o outputs/901/85-K3 &
srun -p high -t 07:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/901/85-ibs-90-noinv.beagle.gz -K 4 -o outputs/901/85-K4 &
srun -p high -t 08:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/901/85-ibs-90-noinv.beagle.gz -K 5 -o outputs/901/85-K5 &
srun -p high -t 09:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/901/85-ibs-90-noinv.beagle.gz -K 6 -o outputs/901/85-K6 &

## Calls
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 81 -bam bamlists/85-down.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf genome/lates-lgs.txt -out outputs/901/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/901/snps-wgs.out 2> outputs/901/snps-wgs.err &
```



## PCA
 
```{r}
meta<-m85 
meta$Region<-factor(meta$Region, levels=c("Queensland","Northern Territory","Papua New Guinea","Mainstream","OceanPick",
                                              "Indonesia-K","Indonesia-SU","Indonesia-SJ","Philippines",
                                              "Vietnam","Cambodia","Thailand", "Bangladesh",
                                              "India Eastern Coast","India Western Coast","Sri Lanka"))


min(meta$Coverage)
max(meta$Coverage)
mean(meta$Coverage)
round(median(meta$Coverage),2)
```

```{r}
m <- as.matrix(read.table("outputs/901/85-ibs-90-noinv.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```



```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta) 
 
text12<-covs %>% select(Run, Region, Lineage, V1, V2) %>% filter(Lineage %in% c("Mainstream","OceanPick","lakdiva")) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Region, shape=Lineage), size=3, alpha=0.75) +
  geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("A") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(21,23,25,22,25,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "none")

pc12
```


```{r}
text13<-covs  %>% select(Run, Region, V1, V3, Lineage)  %>% filter(Lineage %in% c("Mainstream","OceanPick","lakdiva")) %>%
  group_by(Region) %>% summarize(Count=n(), x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Region, shape=Lineage), size=3, alpha=0.75) +
  geom_text_repel(data=text13, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("B") +
  theme(plot.title = element_text(size=18, face="bold")) +
  scale_shape_manual(values=c(21,23,25,22,25,24)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) 

pc13
```

```{r}
ggarrange(pc12,pc13,ncol=2,widths=c(1,1.3))
ggsave("outputs/901/combined-wgspcs.pdf", width=13, height=6)
ggsave("outputs/901/combined-wgspcs.jpeg", width=13, height=6)

```


## Admix


```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```

```{r}
PlotAdmix("outputs/901/85-K2.qopt", meta)
```

```{r}
PlotAdmix("outputs/901/85-K3.qopt", meta)
```


```{r}
PlotAdmix("outputs/901/85-K4.qopt", meta)
```


```{r}
PlotAdmix("outputs/901/85-K5.qopt", meta)
```

# Calls

```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out snps-wgs-binary --recode --allow-extra-chr --noweb
plink --ped snps-wgs-binary.ped --map snps-wgs-binary.map --recode vcf --allow-extra-chr -out snps-wgs-plink
bgzip snps-wgs-plink.vcf 
tabix snps-wgs-plink.vcf.gz

#renaming vcf in 901 locally
bcftools view -r NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1 outputs/901/snps-wgs-plink.vcf.gz| bcftools reheader --samples bamlists/85.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.20 -w 10000  > outputs/901/snps-wgs-pruned.vcf

bcftools view -r NC_066833.1,NC_066834.1,NC_066836.1,NC_066838.1,NC_066839.1,NC_066840.1,NC_066841.1,NC_066842.1,NC_066843.1,NC_066844.1,NC_066845.1,NC_066846.1,NC_066847.1,NC_066848.1,NC_066849.1,NC_066850.1,NC_066851.1,NC_066853.1,NC_066854.1,NC_066855.1,NC_066856.1 outputs/901/snps-wgs-plink.vcf.gz| bcftools reheader --samples bamlists/85.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/901/newhybrids.vcf


bcftools view -r NC_066835.1:1-17500000 outputs/901/snps-wgs-plink.vcf.gz| bcftools reheader --samples bamlists/85.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/901/newhybrids-lca03inv.vcf

bcftools view -r NC_066837.1:5247652- outputs/901/snps-wgs-plink.vcf.gz| bcftools reheader --samples bamlists/85.names   | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.30 -w 50kb --nsites-per-win 1 > outputs/901/newhybrids-lca05inv.vcf
```


